<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<title>Spatial Transcriptomics: All in One View</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" type="text/css" href="assets/styles.css">
<script src="assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="manifest" href="site.webmanifest">
<link rel="mask-icon" href="safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" content="#ffffff">
</head>
<body>
    <header id="top" class="navbar navbar-expand-md navbar-light bg-white top-nav incubator"><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-6">
      <div class="large-logo">
        <img alt="Carpentries Incubator" src="assets/images/incubator-logo.svg"><abbr class="badge badge-light" title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught." style="background-color: #FF4955; border-radius: 5px">
          <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link" style="color: #000">
            <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
            Pre-Alpha
          </a>
          <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
        </abbr>
        
      </div>
    </div>
    <div class="selector-container">
      
      
      <div class="dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Learner View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
<li><button class="dropdown-item" type="button" onclick="window.location.href='instructor/aio.html';">Instructor View</button></li>
        </ul>
</div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl navbar-light bg-white bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Carpentries Incubator" src="assets/images/incubator-logo-sm.svg">
</div>
    <div class="lesson-title-md">
      Spatial Transcriptomics
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
<li class="nav-item">
          <span class="lesson-title">
            Spatial Transcriptomics
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="reference.html#glossary">Glossary</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="profiles.html">Learner Profiles</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown">
<li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul>
</li>
      </ul>
</div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a class="btn btn-primary" href="aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div>
<!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Spatial Transcriptomics
</div>

<aside class="col-md-12 lesson-progress"><div style="width: %" class="percentage">
    %
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: %" aria-valuenow="" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Learner View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="instructor/aio.html">Instructor View</a>
                      </div>
                    </div>
                  </div>
<!--/div.accordion-item-->
                </div>
<!--/div.accordion-flush-->
              </div>
<!--div.sidenav-view-selector -->
            </div>
<!--/div.col -->
      
            <hr>
</div>
<!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Setup</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="introduction.html">1. Spatially Resolved Transcriptomics in Life Sciences Research</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="data-and-study-design.html">2. Data and Study Design</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="data-preprocessing.html">3. Data Preprocessing</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="remove-low-quality-spots.html">4. Remove Low-quality Spots</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="apply-normalization-methods.html">5. Normalization in Spatial Transcriptomics</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="feature-selection-dimensionality-reduction-clustering.html">6. Feature Selection, Dimensionality Reduction, and Spot Clustering</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="deconvolve-cell-types-in-a-spot.html">7. Deconvolution in Spatial Transcriptomics</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="differential-expression-testing.html">8. Differential Expression Testing</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width">
<div class="accordion accordion-flush resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul>
<li>
                        <a href="key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="reference.html#glossary">Glossary</a>
                      </li>
                      <li>
                        <a href="profiles.html">Learner Profiles</a>
                      </li>
                      <li><a href="reference.html">Reference</a></li>
                    </ul>
</div>
                </div>
              </div>
            </div>
            <hr class="half-width resources">
<a href="aio.html">See all in one page</a>
            

            <hr class="d-none d-sm-block d-md-none">
<div class="d-grid gap-1">
            
            </div>
          </div>
<!-- /div.accordion -->
        </div>
<!-- /div.sidebar-inner -->
      </nav>
</div>
<!-- /div.sidebar -->
  </div>
<!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-extra.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <main id="main-content" class="main-content"><div class="container lesson-content">
        
        
<section id="aio-introduction"><p>Content from <a href="introduction.html">Spatially Resolved Transcriptomics in Life Sciences Research</a></p>
<hr>
<p>Last updated on 2024-06-13 |
        
        <a href="https://github.com/smcclatchy/spatial-transcriptomics/edit/main/episodes/introduction.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is spatial transcriptomics?</li>
<li>What research questions or problems can spatial transcriptomics
address?</li>
<li>How do the technologies work?</li>
<li>Which technology will we learn about in this lesson?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Describe why and how spatial transcriptomics can be used in
research.</li>
<li>Describe how spatial transcriptomics technology works.</li>
<li>Describe how spatial transcriptomics addresses the limitations of
single-cell or bulk RNA sequencing technologies.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section id="spatial-transcriptomics-in-biomedical-research"><h2 class="section-heading">Spatial transcriptomics in biomedical research<a class="anchor" aria-label="anchor" href="#spatial-transcriptomics-in-biomedical-research"></a>
</h2>
<hr class="half-width">
<p>Investigating the organization of cells and tissues is fundamental to
life sciences research. Cells and tissues situated in different regions
of an organ can possess diverse functions and cell types. These cells in
turn are influenced by varying tissue microenvironments, receiving and
processing distinct information from that microenvironment. Co-located
cells can communicate directly with one another through chemical and
mechanical signals, responding to these signals with changes in their
own state. Thus, knowing the spatial organization of cells in a tissue
can reveal cell and tissue function.</p>
<figure><img src="https://upload.wikimedia.org/wikipedia/commons/c/c3/Skeletal_muscle_-_cross_section%2C_nerve_bundle.jpg" alt="A cross-section of human skeletal muscle showing muscle cells and a nerve nearby. Stained with hematoxylin and eosin." class="figure mx-auto d-block"><div class="figcaption">A cross-section of skeletal muscle tissue
showing muscle cells and a small nerve.</div>
</figure><p><a href="https://commons.wikimedia.org/wiki/File:Skeletal_muscle_-_cross_section,_nerve_bundle.jpg" class="external-link">Department
of Histology, Jagiellonian University Medical College</a>
<a href="https://creativecommons.org/licenses/by-sa/3.0/deed.en" rel="license" class="external-link">CC
BY-SA 3.0 DEED</a></p>
<p>Spatially resolved transcriptomics describes spatial organization and
cell signals, specifically gene expression signals. Spatial patterns of
gene expression determine how genes are regulated within a tissue system
and how those tissues and their component cells function. Spatial
transcriptomic (ST) methods map cell position in a tissue, clarifying
the physical relationships between cells and cellular structures. ST
simultaneously measures gene expression, delivering valuable information
about cell phenotype, state, and cell and tissue organization and
function. The combination of cellular expression and position sheds
light on signals a cell sends or receives through cell-to-cell
interactions. Spatial information localizes cell signaling while
delivering comprehensive gene expression profiling within tissues.</p>
<figure><img src="https://upload.wikimedia.org/wikipedia/commons/0/04/Notchccr.svg" alt="alt text for accessibility purposes" class="figure mx-auto d-block"><div class="figcaption">Signaling between adjacent cells. The Notch
protein functions as a receptor for ligands that activate or inhibit
such receptors. Receptor-ligand interactions ground cell signaling and
communication, often requiring close proximity between cells.</div>
</figure><p><a href="https://commons.wikimedia.org/wiki/File:Notchccr.svg" class="external-link">Fred
the Oyster</a> Public domain, via Wikimedia Commons
<a href="https://creativecommons.org/licenses/by-sa/4.0/" rel="license" class="external-link">CC
BY-SA 4.0 DEED</a></p>
<p>Spatial transcriptomics addresses a key obstacle in single-cell and
bulk RNA sequencing studies: their loss of spatial information. Spatial
organization and structure determine function in most tissues and
organs, so capturing both spatial and expression information is critical
for understanding tissue function in neuroscience, immuno-oncology,
developmental biology, and most other fields.</p>
</section><section id="spatial-transcriptomics-technologies"><h2 class="section-heading">Spatial transcriptomics technologies<a class="anchor" aria-label="anchor" href="#spatial-transcriptomics-technologies"></a>
</h2>
<hr class="half-width">
<p>Spatial transcriptomics technologies broadly fall within two groups:
imaging-based and sequencing-based methods. Both imaging- and
sequence-based datasets are available through the <a href="https://biccn.org/" class="external-link">The BRAIN Initiative - Cell Census
Network</a>. Sequencing-based datasets are featured in the <a href="https://data.humancellatlas.org/" class="external-link">Human Cell Atlas</a>. These
technologies vary in ability to profile entire transcriptomes, deliver
single-cell resolution, and detect genes efficiently.</p>
<div class="section level3">
<h3 id="imaging-based-technologies">Imaging-based technologies<a class="anchor" aria-label="anchor" href="#imaging-based-technologies"></a>
</h3>
<p>Imaging-based technologies read transcriptomes in place using
microscopy at single-cell or even single-molecule resolution. They
identify messenger RNA (mRNA) species fluorescence in situ hybridization
(FISH), i.e., by hybridizing mRNA to gene-specific fluorescent
probes.</p>
<figure><img src="fig/FISH_Overview.png" alt="a general schematic showing fluorescence in situ hybridization" class="figure mx-auto d-block"><div class="figcaption">Overview of fluorescence in situ hybridization
(FISH).</div>
</figure><p><a href="https://commons.wikimedia.org/wiki/File:Spatial_Transcriptomics_Overview.png" class="external-link">
Adapted from Spatial Transcriptomics Overview by SlifertheRyeDragon.</a>
Image created with Biorender.com. Public domain, via Wikimedia Commons
<a href="https://creativecommons.org/licenses/by-sa/4.0/" rel="license" class="external-link">CC
BY-SA 4.0 DEED</a></p>
<p>RNA can be visualized in place in the original tissue environment by
hybridizing labeled probes to their specific targets. Current FISH
methods employ multiple hybridization rounds, with risk of error for
each transcript growing exponentially with each round. FISH methods are
limited in the size of tissue that they can profile and most are
applicable only to fresh-frozen (FF) tissue. They can also be
time-consuming and expensive due to microscopic imaging they require.
Since they target specific genes, they can only detect genes that are in
the probe set. They have high spatial resolution though, even delivering
single-molecule resolution in single-molecule FISH (smFISH).</p>
<p>Conventional FISH methods have few distinct color channels that limit
the number of genes that can be simultaneously analyzed. Multiplexed
error-robust FISH (MERFISH) overcomes this problem, greatly increasing
the number of RNA species that can be simultaneously imaged in single
cells by using binary code gene labeling in multiple rounds of
hybridization.</p>
<figure><img src="https://upload.wikimedia.org/wikipedia/commons/thumb/a/ad/MERFISH_Diagram.png/1024px-MERFISH_Diagram.png" alt="alt text for accessibility purposes" class="figure mx-auto d-block"><div class="figcaption">Schematic representation of multiplexed
error-robust FISH (MERFISH). Binary codes assigned to mRNA species of
interest, where “1” represents a short fluorescent DNA probe. b,
Consecutive hybridization rounds, bleaching in between is implied, but
not shown for clarity. At the end of the sixth round, it is possible to
tell different mRNAs apart due to the decoded combinations of “1” and
“0”.</div>
</figure><p><a href="https://commons.wikimedia.org/wiki/File:MERFISH_Diagram.png" class="external-link">SlifertheRyeDragon</a>,
<a href="https://creativecommons.org/licenses/by-sa/4.0" class="external-link">CC BY-SA
4.0</a>, via Wikimedia Commons</p>
<p>A second imaging-based method, in situ sequencing, amplifies and
sequences mRNAs directly within a block or section of fresh-frozen (FF)
or formalin-fixed paraffin embedded (FFPE) tissue.</p>
<p><img src="fig/ISS_Overview.png" alt="a general schematic showing in situ sequencing" class="figure"><a href="https://commons.wikimedia.org/wiki/File:Spatial_Transcriptomics_Overview.png" class="external-link">
Adapted from Spatial Transcriptomics Overview by SlifertheRyeDragon.</a>
Image created with Biorender.com. Public domain, via Wikimedia Commons
<a href="https://creativecommons.org/licenses/by-sa/4.0/" rel="license" class="external-link">CC
BY-SA 4.0 DEED</a></p>
<p>Messenger RNA (mRNA) is reverse transcribed to complementary DNA
(cDNA) within tissue sections. A “padlock” probe binds to the cDNA,
which is then circularized. Following circularization, the cDNA is
amplified by rolling-circle amplification (RCA), then sequenced by
ligation for decoding. Probes profile one or two bases at a time using
different fluorophores, eventually revealing the identity of the cDNA
through imaging. Since it requires imaging, in situ sequencing is an
imaging-based method even though it involves sequencing. In situ
sequencing can accommodate larger tissue sections than can FISH, though
FISH methods are more efficient at detecting mRNA of genes in the probe
set. Like FISH, in situ sequencing requires considerable imaging time on
a microscope but delivers high spatial resolution. Both methods require
a priori knowledge of target mRNA.</p>
</div>
<div class="section level3">
<h3 id="sequencing-based-technologies">Sequencing-based technologies<a class="anchor" aria-label="anchor" href="#sequencing-based-technologies"></a>
</h3>
<p>Sequencing-based methods capture, sequence, and count mRNA using
next-generation sequencing while retaining positional information. This
is distinct from in situ sequencing because next-generation sequencing
is employed. Sequencing-based methods may be unbiased, in which they
capture the entire transcriptome, or probe-based, in which they
typically capture the majority of protein-coding genes. Sequencing-based
methods retain spatial information through laser-capture microdissection
(LCM), microfluidics, or through ligation of mRNAs to arrays of barcoded
probes that record position.</p>
<p>LCM-based methods employ lasers to cut a tissue slice or fuse tissue
to a membrane followed by sequencing of individual cells.</p>
<p><img src="fig/LCM_Overview.png" alt="a general schematic showing laser-capture microdissection " class="figure"><a href="https://commons.wikimedia.org/wiki/File:Spatial_Transcriptomics_Overview.png" class="external-link">
Adapted from Spatial Transcriptomics Overview by SlifertheRyeDragon.</a>
Image created with Biorender.com. Public domain, via Wikimedia Commons
<a href="https://creativecommons.org/licenses/by-sa/4.0/" rel="license" class="external-link">CC
BY-SA 4.0 DEED</a></p>
<p>LCM techniques process tissue sections for transcriptomic profiling
by isolating regions of interest. They are useful for profiling
transcriptomes as a first pass and for identifying RNA isoforms, but
their blunt approach to capturing spatial expression data limits spatial
resolution and requires many samples for sequencing. Since they focus on
regions of interest, it is often not possible to obtain a picture of
spatial expression across a whole tissue. LCM is an older technology
that has long been used with FFPE tissues. Modern LCM-based approaches
include Nanostring’s GeoMx DSP and STRP-seq.</p>
<p>Microfluidics places a chip with multiple barcode-containing channels
onto a tissue section followed by a second chip with channels
perpendicular to the first. The barcodes are then ligated to each other
to create an array of unique barcodes on the tissue. This “deterministic
barcoding” is employed in DBiT-seq. DBiT-seq can be used with FFPE
tissues. This approach is helpful to avoid diffusion of mRNA away from
capture areas, though a disadvantage is that cells often sit astride
multiple capture areas.</p>
<p><img src="fig/Workflow-of-DBiT-seq-on-FFPE-samples-a-Scheme-of-DBiT-seq-on-FFPE-samples.png" alt="a general schematic showing a microfluidics workflow with DBit-seq on formalin-fixed paraffin embedded (FFPE) tissue" class="figure">
Adapted from
<a href="https://www.researchgate.net/publication/346261659_Spatial_transcriptome_sequencing_of_FFPE_tissues_at_cellular_level" class="external-link">Liu
Y, Enninful A, Deng Y, &amp; Fan R (2020). Spatial transcriptome
sequencing of FFPE tissues at cellular level. Preprint.</a>
<a href="https://creativecommons.org/licenses/by-sa/4.0/" rel="license" class="external-link">CC
BY-SA 4.0 DEED</a></p>
<p>Other array-based methods capture mRNA with spatially-barcoded probes
and sequence them. They can profile larger tissue sections than can FISH
or in situ sequencing and they don’t rely on time-consuming microscopic
imaging. Spatial resolution is lower, however.</p>
<p>In this lesson we will use data from positionally barcoded
arrays.</p>
<figure><img src="https://upload.wikimedia.org/wikipedia/commons/1/14/Spatial_transcriptomics_ii.png" alt="A graphic showing printed spots on a glass slide that are identified by a barcode and that contain primers to capture mRNA from the tissue laid on top of them" class="figure mx-auto d-block"><div class="figcaption">A sequencing-based spatial transcriptomics
method using printed spots on a slide.</div>
</figure><p><a href="https://commons.wikimedia.org/wiki/User:Jasquatch" class="external-link">James
Chell</a>,
<a href="https://commons.wikimedia.org/wiki/File:Spatial_transcriptomics_ii.png" class="external-link">Spatial
transcriptomics ii</a>,
<a href="https://creativecommons.org/licenses/by-sa/4.0/legalcode" rel="license" class="external-link">CC
BY-SA 4.0</a></p>
<table class="table">
<colgroup>
<col width="27%">
<col width="18%">
<col width="18%">
<col width="18%">
<col width="18%">
</colgroup>
<thead><tr class="header">
<th>Technology</th>
<th align="center">Gene detection efficiency</th>
<th align="center">Transcriptome-wide profiling</th>
<th align="center">Spatial resolution</th>
<th align="center">Tissue area</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>FISH</td>
<td align="center">+</td>
<td align="center">-</td>
<td align="center">+</td>
<td align="center">-</td>
</tr>
<tr class="even">
<td>In situ sequencing</td>
<td align="center">-</td>
<td align="center">-</td>
<td align="center">+</td>
<td align="center">+</td>
</tr>
<tr class="odd">
<td>LCM-based</td>
<td align="center">+</td>
<td align="center">+</td>
<td align="center">-</td>
<td align="center">-</td>
</tr>
<tr class="even">
<td>Microfluidics</td>
<td align="center">-</td>
<td align="center">+</td>
<td align="center">-</td>
<td align="center">+</td>
</tr>
<tr class="odd">
<td>Array-based</td>
<td align="center">-</td>
<td align="center">+</td>
<td align="center">-</td>
<td align="center">+</td>
</tr>
</tbody>
</table>
<p>Table 1. Relative strengths and weaknesses of spatial transcriptomics
technologies by general category.</p>
<div id="callout1" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout<a class="anchor" aria-label="anchor" href="#callout1"></a>
</h3>
<div class="callout-content">
<p>This introduction to the technologies is intended to help you
navigate a complex technological landscape so that you can learn more
about existing technologies. It is not intended to be comprehensive.</p>
</div>
</div>
</div>
<p>The diversity in spatial transcriptomics technologies is enormous and
rapidly developing. If you would like to learn more about spatial
transcriptomics technologies, please see the
<a href="./reference.html"><strong>list of references located
here</strong></a>.</p>
<div id="discussion-which-technology-is-right-for-your-research" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="discussion-which-technology-is-right-for-your-research" class="callout-inner">
<h3 class="callout-title">Discussion: Which technology is right for your
research?<a class="anchor" aria-label="anchor" href="#discussion-which-technology-is-right-for-your-research"></a>
</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li><p>Would an imaging-based or a sequencing-based solution be
preferable for your research? Why?</p></li>
<li><p>From the descriptions above, which technology do you think would
best suit your research? Would you use fresh-frozen (FF) or
formalin-fixed tissues embedded in paraffin (FFPE)? Even if your
institution does not offer service using a specific technology, describe
which best suits your research and why you think it’s best
suited.</p></li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">Show me the solution</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>Which technology you use depends on your experimental aim. If you are
testing hypotheses about specific genes, you can profile those genes at
high resolution and sensitivity with an imaging-based method. If instead
you aim to generate hypotheses, next generation sequencing-based methods
that profile many genes in an unbiased manner would be best.</p>
</div>
</div>
</div>
</div>
</div>
</section><section id="x-genomics-visium-technology"><h2 class="section-heading">10X Genomics Visium technology<a class="anchor" aria-label="anchor" href="#x-genomics-visium-technology"></a>
</h2>
<hr class="half-width">
<p>In this lesson we will use data from an array-based method called
Visium that is offered by <a href="https://www.10xgenomics.com/platforms/visium" class="external-link">10X Genomics</a>.
Visium is an upgrade and commercialization of the spatial
transcriptomics method described in 2016 in <a href="https://doi.org/10.1126/science.aaf2403" class="external-link">Science, 353(6294)</a>
and illustrated in general in the figure above. A more specific
schematic is given below.</p>
<figure><img src="fig/visium-technology.png" alt="a general schematic showing the Visium technology" class="figure mx-auto d-block"><div class="figcaption">Overview of Visium technology with fresh-frozen
(FF) or formalin-fixed paraffin embedded (FFPE) tissue. Source: <a href="https://www.10xgenomics.com/platforms/visium" class="external-link">10x Genomics
Visium</a>
</div>
</figure><p>Thin tissue sections are placed atop spots printed with spatial
barcodes. For fresh-frozen (FF) tissues, tissue fixing and
permeabilization discharges mRNA to bind with spatially barcoded probes
that indicate the position on the slide. The assay is sensitive to
permeabilization time, which is often optimized in a separate
experimental procedure. Captured mRNA is then reverse transcribed to
cDNA and sequencing libraries created from these. The formalin-fixed
paraffin embedded (FFPE) assay utilizes probe pairs that identify each
gene in the probe set and capture target genes by hybridizing to them.
Permeabilization discharges the captured mRNA to spatially barcoded
probes on the slide, but does not require a separate optimization step
as in the FF protocol. The capture mRNA is then extended to include
complements of the spatial barcodes. Sequencing libraries are then
formed from the captured and barcoded mRNA.</p>
<p>Spatial transcriptomics combines two key modes: histological imaging
and gene expression profiling. Histological imaging captures tissue
morphology with standard staining protocols while expression profiling
is captured by sequencing spatially barcoded cDNA.</p>
<p>Sequencing-based datasets have grown faster than have imaging-based
datasets, with Visium dominating published datasets. Unlike most other
sequencing-based technologies, Visium accommodates both FF or FFPE
tissue. Each spot provides average gene expression measurements from
between one to a few tens of cells, approaching single-cell resolution.
Average gene expression measurements are combined with histological
images that couple molecular detail and tissue morphology and
structure.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>Spatial transcriptomics provides the location of cells relative to
neighboring cells and cell structures.</li>
<li>A cell’s location is useful data for describing its phenotype,
state, and cell and tissue function.</li>
<li>Spatial transcriptomics addresses a key obstacle in single-cell
studies: the loss of spatial information when tissues are
dissociated.</li>
<li>The main goal of spatial transcriptomics studies is to integrate
expression with spatial information.</li>
</ul>
</div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</section></section><section id="aio-data-and-study-design"><p>Content from <a href="data-and-study-design.html">Data and Study Design</a></p>
<hr>
<p>Last updated on 2024-06-13 |
        
        <a href="https://github.com/smcclatchy/spatial-transcriptomics/edit/main/episodes/data-and-study-design.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>Which data will we explore in this course?</li>
<li>How was the study that generated the data designed?</li>
<li>What are some critical design elements for rigorous, reproducible
spatial transcriptomics experiments?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Describe a spatial transcriptomics experiment.</li>
<li>Identify important elements for good experimental design.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section id="the-data"><h2 class="section-heading">The Data<a class="anchor" aria-label="anchor" href="#the-data"></a>
</h2>
<hr class="half-width">
<p>Recall that tissue is laid on a glass slide containing spots with
primers to capture mRNA. The graphic below details a Visium slide with
four capture areas. Each capture area has arrays of barcoded spots
containing oligonucleotides. The oligonucleotides each contain a
poly(dT) sequence for capture of polyadenylated molecules, a unique
molecular identifier (UMI) to identify duplicate molecules, a spatial
barcode shared by all oligonucleotides within the same spot, and a
partial read for library preparation and sequencing.</p>
<figure><img src="fig/visium-slide.png" alt="A graphic showing printed spots on a glass slide that are identified by a barcode and that contain oligonucleotides to capture messenger RNA from the tissue laid on top of them" class="figure mx-auto d-block"><div class="figcaption">Visium spatial gene expression slide</div>
</figure><p>In spatial transcriptomics the barcode indicates the x-y coordinates
of the spot. Barcodes are generic identifiers that identify different
things in different technologies. A barcode in single-cell
transcriptomics, for example, refers to a single cell, not to a spot on
a slide. When you see barcodes in ST data, think “spot”, not “single
cell”. In fact, one spot can capture mRNA from many cells. This is a
feature of ST experiments that is distinct from single-cell
transcriptomics experiments. As a result, many single-cell methods won’t
work with ST data. Later we will look at methods to <a href="deconvolve-cell-types-in-a-spot.html">“deconvolve” cell types per
spot</a> to determine the number and types of cells in each spot. Spots
can contain zero, one, or many cells.</p>
<p>The graphic below shows a Visium workflow for fresh-frozen
tissues.</p>
<p><img src="fig/fresh-frozen-workflow.png" alt="A Visium spatial transcriptomics workflow with fresh-frozen tissue" class="figure">
Graphic from
<a href="https://www.10xgenomics.com/library/6f2b8a" class="external-link"><i>Grant
application resources for Visium products</i></a> at 10X Genomics</p>
<p>Count data for each mRNA are mapped back to spots on the slide to
indicate the tissue position of gene expression. An image of the tissue
overlaid on the array of spots pinpoints spatial gene expression in the
tissue.</p>
<figure><img src="fig/Spatial_transcriptomics_ii_lower_third.png" alt="A graphic showing printed spots on a glass slide that are identified by a barcode and that contain primers to capture messenger RNA from the tissue laid on top of them" class="figure mx-auto d-block"><div class="figcaption">Sequencing data is mapped back to spots on the
slide and compared to an image of the tissue to localize
expression</div>
</figure><p>Adapted from
<a href="https://commons.wikimedia.org/wiki/User:Jasquatch" class="external-link">James
Chell</a>,
<a href="https://commons.wikimedia.org/wiki/File:Spatial_transcriptomics_ii.png" class="external-link">Spatial
transcriptomics ii</a>,
<a href="https://creativecommons.org/licenses/by-sa/4.0/legalcode" rel="license" class="external-link">CC
BY-SA 4.0</a></p>
<p>Data from the 10X Genomics Visium platform contain gene identifiers
in rows and barcode identifiers in columns. In the graphic below, row 1
of column 1 contains the mRNA counts for gene 1 at barcode (spot) 1.</p>
<figure><img src="fig/spatial-data.png" alt="An example of spatial transcriptomics data showing genes in rows and barcodes (spots) in columns" class="figure mx-auto d-block"><div class="figcaption">Spatial transcriptomics data include genes in
rows and barcodes in columns</div>
</figure><div id="challenge-1-row-and-column-sums" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-row-and-column-sums" class="callout-inner">
<h3 class="callout-title">Challenge 1: Row and column sums<a class="anchor" aria-label="anchor" href="#challenge-1-row-and-column-sums"></a>
</h3>
<div class="callout-content">
<p>What does the sum of a single row signify?<br>
What does the sum of a single column signify?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">Show me the solution</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>The row sum is the total expression of one gene across all spots on
the slide.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sum</span><span class="op">(</span><span class="st">'data[1, ]'</span><span class="op">)</span></span></code></pre>
</div>
<p>The column sum is the total expression of all genes for one spot on
the slide.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sum</span><span class="op">(</span><span class="st">'data[ , 1]'</span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>
</div>
</div>
</section><section id="study-design"><h2 class="section-heading">Study design<a class="anchor" aria-label="anchor" href="#study-design"></a>
</h2>
<hr class="half-width">
<p>We will use data from <a href="https://doi.org/10.1038/s41593-020-00787-0" class="external-link">Transcriptome-scale
spatial gene expression in the human dorsolateral prefrontal cortex by
Maynard et al, Nat Neurosci 24, 425–436 (2021).</a> These data come from
sections of the dorsolateral prefrontal cortex that contain six cortical
layers plus white matter.</p>
<figure><img src="fig/tissue-block-with-layers.png" alt="A human brain showing a section of dorsolateral prefrontal cortex extracted. A block of tissue containing six cortical layers and an underlying layer of white matter is excised from the section." class="figure mx-auto d-block"><div class="figcaption">Tissue blocks were excised from human
dorsolateral prefrontal cortex. Tissue blocks include six cortical
layers and underlying white matter (wm).</div>
</figure><p>Adapted from
<a href="https://doi.org/10.1038/s41593-020-00787-0" class="external-link">Maynard et al, Nat
Neurosci 24, 425–436 (2021)</a>.
<a href="https://www.biorender.com" class="external-link">Created with BioRender.com</a>.</p>
<p>Two pairs of spatially adjacent replicates were taken from three
neurotypical donors. The second pair of replicates was taken from 300
microns posterior to the first pair of replicates.</p>
<p><img src="fig/experimental-design.png" alt="Three Visium slides showing four spatial capture areas each. Each slide contains directly adjacent serial tissue sections for one subject. The second pair of samples contains tissue sections that are 300 microns posterior to the first pair of samples." class="figure">
Adapted from
<a href="https://doi.org/10.1038/s41593-020-00787-0" class="external-link">Maynard et al, Nat
Neurosci 24, 425–436 (2021)</a>.
<a href="https://www.biorender.com" class="external-link">Created with BioRender.com</a>.</p>
<div id="challenge-2-what-do-you-notice" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-what-do-you-notice" class="callout-inner">
<h3 class="callout-title">Challenge 2: What do you notice?<a class="anchor" aria-label="anchor" href="#challenge-2-what-do-you-notice"></a>
</h3>
<div class="callout-content">
<p>What do you notice about the experimental design that might create
issues during data analysis? Why might the authors have done the
experiment this way? Is there anything that can be done about this?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2">Show me the solution</h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>Human brain tissues start to deteriorate at death. Brain banks that
provide tissue for studies requiring intact mRNA will quickly remove the
brain and rapidly weigh, examine, dissect and freeze it to optimize mRNA
integrity. For this study, dorsolateral prefrontal cortex samples were
embedded in a medium (see Methods section) and then cryosectioned.
Sections were then placed on chilled Visium slides, fixed and
stained.</p>
<p>This whole process might have depended on donor availability. All
three donors were neurotypical, and it’s not clear how they died
(e.g. an accident, a terminal illness, old age). Suffice it to say that
it might not have been possible to predict when samples would be
available, so it might not have been possible to randomize the samples
from each donor to different Visium slides. The article doesn’t clarify
any of this, however, there undoubtedly are complications in accessing
human brain tissues.</p>
<p>Sometimes you might have to confound variables in your study due to
sample availability or other factors. The key thing is to know that
confounding has occurred.</p>
</div>
</div>
</div>
</div>
</section><section id="important-considerations-for-rigorous-reproducible-experiments"><h2 class="section-heading">Important considerations for rigorous, reproducible experiments<a class="anchor" aria-label="anchor" href="#important-considerations-for-rigorous-reproducible-experiments"></a>
</h2>
<hr class="half-width">
<p>Good experimental design plays a critical role in obtaining reliable
and meaningful results and is an essential feature of rigorous,
reproducible experiments. Designed experiments aim to describe and
explain variability under experimental conditions. Variability is
natural in the real world. A medication given to a group of patients
will affect each of them differently. A specific diet given to a cage of
mice will affect each mouse differently. Ideally if something is
measured many times, each measurement will give exactly the same result
and will represent the true value. This ideal doesn’t exist in the real
world. Variability is a feature of natural systems and also a natural
part of every experiment we undertake.</p>
<div class="section level4">
<h4 id="replication">Replication<a class="anchor" aria-label="anchor" href="#replication"></a>
</h4>
<p>To figure out whether a difference in responses is real or inherently
random, replication applies the same treatment to multiple experimental
units. The variability of the responses within a set of replicates
provides a measure against which we can compare differences among
different treatments. “Experimental error” describes the variability in
the responses. Random variation (a.k.a random error or noise) reflects
imprecision, but not inaccuracy. Larger sample sizes reduce this
imprecision.</p>
<p>In addition to random (experimental) error, systematic error or bias
occurs when there are deviations in measurements or observations that
consistently either overestimate or underestimate the true value. As an
example, a scale might be calibrated so that mass measurements are
consistently too high or too low. Unlike random error, systematic error
is consistent in one direction, is predictable and follows a pattern.
Larger sample sizes don’t correct for systematic bias; equipment or
measurement calibration does. Technical replicates define this
systematic bias by running the same sample through the machine or
measurement protocol multiple times to characterize the variation caused
by equipment or protocols.</p>
<p>A biological replicate measures different biological samples in
parallel to estimate the variation caused by the unique biology of the
samples. The sample or group of samples are derived from the same
biological source, such as cells, tissues, organisms, or individuals.
Biological replicates assess the variability and reproducibility of
experimental results. The greater the number of biological replicates,
the greater the precision (the closeness of two or more measurements to
each other). Having a large enough sample size to ensure high precision
is necessary to ensure reproducible results. Note that increasing the
number of technical replicates will not help to characterize biological
variability! It is used to characterize systematic error, not
experimental error.</p>
<div id="challenge-3-which-kind-of-error" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3-which-kind-of-error" class="callout-inner">
<h3 class="callout-title">Challenge 3: Which kind of error?<a class="anchor" aria-label="anchor" href="#challenge-3-which-kind-of-error"></a>
</h3>
<div class="callout-content">
<p>A study used to determine the effect of a drug on weight loss could
have the following sources of experimental error. Classify the following
sources as either biological, systematic, or random error.<br>
1). A scale is broken and provides inconsistent readings.<br>
2). A scale is calibrated wrongly and consistently measures mice 1 gram
heavier.<br>
3). A mouse has an unusually high weight compared to its experimental
group (i.e., it is an outlier).<br>
4). Strong atmospheric low pressure and accompanying storms affect
instrument readings, animal behavior, and indoor relative humidity.</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3">Solution to Challenge 3</h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">
<p>1). random, because the scale is broken and provides any kind of
random reading it comes up with (inconsistent reading)<br>
2). systematic<br>
3). biological<br>
4). random or systematic; you argue which and explain why</p>
</div>
</div>
</div>
</div>
<div id="challenge-4-how-many-technical-and-biological-replicates" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-4-how-many-technical-and-biological-replicates" class="callout-inner">
<h3 class="callout-title">Challenge 4: How many technical and biological
replicates?<a class="anchor" aria-label="anchor" href="#challenge-4-how-many-technical-and-biological-replicates"></a>
</h3>
<div class="callout-content">
<p>In each scenario described below, identify how many technical and how
many biological replicates are represented. What conclusions can be
drawn about experimental error in each scenario?</p>
<p>1). One person is weighed on a scale five times.<br>
2). Five people are weighed on a scale one time each.<br>
3). Five people are weighed on a scale three times each.<br>
4). A cell line is equally divided into four samples. Two samples
receive a drug treatment, and the other two samples receive a different
treatment. The response of each sample is measured three times to
produce twelve total observations. In addition to the number of
replicates, can you identify how many experimental units there
are?<br>
5). A cell line is equally divided into two samples. One sample receives
a drug treatment, and the other sample receives a different treatment.
Each sample is then further divided into two subsamples, each of which
is measured three times to produce twelve total observations. In
addition to the number of replicates, can you identify how many
experimental units there are?</p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4">Solution to Challenge 4</h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" aria-labelledby="headingSolution4" data-bs-parent="#accordionSolution4">
<div class="accordion-body">
<p>1). One biological sample (not replicated) with five technical
replicates. The only conclusion to be drawn from the measurements would
be better characterization of systematic error in measuring. It would
help to describe variation produced by the instrument itself, the scale.
The measurements would not generalize to other people.<br>
2). Five biological replicates with one technical measurement (not
replicated). The conclusion would be a single snapshot of the weight of
each person, which would not capture systematic error or variation in
measurement of the scale. There are five biological replicates, which
would increase precision, however, there is considerable other variation
that is unaccounted for.<br>
3). Five biological replicates with three technical replicates each. The
three technical replicates would help to characterize systematic error,
while the five biological replicates would help to characterize
biological variability.<br>
4). Four biological replicates with three technical replicates each. The
three technical replicates would help to characterize systematic error,
while the four biological replicates would help to characterize
biological variability. Since the treatments are applied to each of the
four samples, there are four experimental units.<br>
5). Two biological replicates with three technical replicates each.
Since the treatments are applied to only the two original samples, there
are only two experimental units.</p>
</div>
</div>
</div>
</div>
</div>
<div class="section level4">
<h4 id="randomization">Randomization<a class="anchor" aria-label="anchor" href="#randomization"></a>
</h4>
<p>Randomization minimizes bias, moderates experimental error (a.k.a.
noise), and ensures that our comparisons between treatment groups are
valid. Randomized studies assign experimental units to treatment groups
randomly by pulling a number out of a hat or using a computer’s random
number generator. The main purpose for randomization comes later during
statistical analysis, where we compare the data we have with the data
distribution we might have obtained by random chance. Random assignment
(<em>allocation</em>) of experimental units to treatment groups prevents
the subjective bias that might be introduced by an experimenter who
selects, even in good faith and with good intention, which experimental
units should get which treatment.</p>
<p>Randomization also accounts for or cancels out effects of “nuisance”
variables like the time or day of the experiment, the investigator or
technician, equipment calibration, exposure to light or ventilation in
animal rooms, or other variables that are not being studied but that do
influence the responses. Randomization balances out the effects of
nuisance variables between treatment groups by giving an equal
probability for an experimental unit to be assigned to any treatment
group.</p>
<div id="challenge-5-treatment-and-control-samples" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-5-treatment-and-control-samples" class="callout-inner">
<h3 class="callout-title">Challenge 5: Treatment and control
samples<a class="anchor" aria-label="anchor" href="#challenge-5-treatment-and-control-samples"></a>
</h3>
<div class="callout-content">
<p>You plan to place samples of treated tissue on one slide and samples
of the controls on another slide. What will happen when it is time for
data analysis? What could you have done differently? <img src="fig/treated-and-control-slides.png" alt="An experiment with treated samples on one slide and control samples on another." class="figure"></p>
</div>
</div>
</div>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5">Show me the solution</h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" aria-labelledby="headingSolution5" data-bs-parent="#accordionSolution5">
<div class="accordion-body">

</div>
</div>
</div>
</div>
<div id="challenge-6-time-points" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-6-time-points" class="callout-inner">
<h3 class="callout-title">Challenge 6: Time points<a class="anchor" aria-label="anchor" href="#challenge-6-time-points"></a>
</h3>
<div class="callout-content">
<p>Your study requires data collection at three time points: 5, 10, and
15 weeks. At the end of 5 weeks, you will run samples through the entire
Visium workflow. You will repeat this for the 10- and 15-week samples
when each of those time points is reached. What will happen when it is
time for data analysis? What could you have done differently?</p>
<figure><img src="fig/timepoints.png" alt="An experiment with three timepoints at 5, 10 and 15 weeks. At the end of the first 5 weeks, those samples are run through Visium. This is repeated at 10 and 15 weeks." class="figure mx-auto d-block"><div class="figcaption">Three time points in an experiment</div>
</figure>
</div>
</div>
</div>
<div id="accordionSolution6" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution6" aria-expanded="false" aria-controls="collapseSolution6">
  <h4 class="accordion-header" id="headingSolution6">Show me the solution</h4>
</button>
<div id="collapseSolution6" class="accordion-collapse collapse" aria-labelledby="headingSolution6" data-bs-parent="#accordionSolution6">
<div class="accordion-body">
<p>The issue is that time point is now confounded. A better approach
would be to start the 15-week samples, then 5 weeks later start the
10-week samples, then 5 weeks later start the 5-week samples. This way
you can run all of your samples at the same time. None of your samples
will have spent a long time in the freezer, so you won’t need to worry
about the variation that might cause. You won’t need to worry about the
time point confounding the results.</p>
</div>
</div>
</div>
</div>
<div id="challenge-7-the-efficient-technician" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-7-the-efficient-technician" class="callout-inner">
<h3 class="callout-title">Challenge 7: The efficient technician<a class="anchor" aria-label="anchor" href="#challenge-7-the-efficient-technician"></a>
</h3>
<div class="callout-content">
<p>Your technician colleague finds a way to simplify and expedite an
experiment. The experiment applies four different wheel-running
treatments to twenty different mice over the course of five days. Four
mice are treated individually each day for two hours each with a random
selection of the four treatments. Your clever colleague decides that a
simplified protocol would work just as well and save time. Run treatment
1 five times on day 1, treatment 2 five times on day 2, and so on. Some
overtime would be required each day but the experiment would be
completed in only four days, and then they can take Friday off! Does
this adjustment make sense to you?<br>
Can you foresee any problems with the experimental results?</p>
<figure><img src="fig/wheel-running-experiment.png" alt="Four different wheel running treatments applied to 5 mice each. Treatment 1 is applied on day 1, treatment 2 on day 2, and so on." class="figure mx-auto d-block"><div class="figcaption">Four different wheel running treatments each
applied once per day to five mice, for a total of 20 mice treated.</div>
</figure>
</div>
</div>
</div>
<div id="accordionSolution7" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution7" aria-expanded="false" aria-controls="collapseSolution7">
  <h4 class="accordion-header" id="headingSolution7">Solution to Challenge 6</h4>
</button>
<div id="collapseSolution7" class="accordion-collapse collapse" aria-labelledby="headingSolution7" data-bs-parent="#accordionSolution7">
<div class="accordion-body">
<p>Since each treatment is run on only one day, the day effectively
becomes the experimental unit (explain this). Each experimental unit
(day) has five samples (mice), but only one replication of each
treatment. There is no valid way to compare treatments as a result.
There is no way to separate the treatment effect from the day-to-day
differences in environment, equipment setup, personnel, and other
extraneous variables.</p>
</div>
</div>
</div>
</div>
</div>
<div class="section level4">
<h4 id="statistical-power">Statistical power<a class="anchor" aria-label="anchor" href="#statistical-power"></a>
</h4>
<p>Statistical power represents the probability of detecting a real
treatment effect. Review the following figure to explore the
relationships between effect size, sample size, and power. What is the
relationship between effect size and sample size? Between sample size
and power?</p>
<figure><img src="fig/data-and-study-design-rendered-power-curve-2-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Adapted from <a href="https://levibaguley.github.io/2020/06/22/how-to-create-power-curves-in-ggplot/" class="external-link">How
to Create Power Curves in ggplot</a> by Levi Baguley</p>
<p>Notice that to detect a standardized effect size of 0.5 at 80% power,
you would need a sample size of approximately 70. Larger effect sizes
require much smaller sample sizes. Very small effects such as .01 never
reach the 80% power threshold without enormous samples sizes in the
hundreds of thousands.</p>
<figure><img src="fig/Null-hypothesis.png" alt="A normal curve with a mean of zero showing the type 1 error rate in the far right tail and specificity in the left of the curve." class="figure mx-auto d-block"><div class="figcaption">The null hypothesis states that there is no
difference between treatment groups.</div>
</figure><figure><img src="fig/Alternative-hypothesis.png" alt="A normal curve with a mean of approximately 3 showing the type 2 error rate in the left of the curve and sensitivity (also known as statistical power) in the far right tail of the curve. The effect size is shown as the difference in means between the null and alternative hypotheses." class="figure mx-auto d-block"><div class="figcaption">The alternative hypothesis states that there is
a difference between treatment groups.</div>
</figure><p>The effect size is shown in the figure above as the difference in
means between the null and alternative hypotheses. Statistical power,
also known as sensitivity, is the power to detect this effect.</p>
<p>To learn more about statistical power, effect sizes and sample size
calculations, see <a href="https://www.nature.com/articles/nmeth.2738" class="external-link">Power and sample size
by Krzywinski &amp; Altman</a> , Nature Methods 10, pages 1139–1140
(2013).</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>Use <code>.md</code> files for episodes when you want static
content</li>
<li>Use <code>.Rmd</code> files for episodes when you need to generate
output</li>
<li>Run <code>sandpaper::check_lesson()</code> to identify any issues
with your lesson</li>
<li>Run <code>sandpaper::build_lesson()</code> to preview your lesson
locally</li>
</ul>
</div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</div>
</section></section><section id="aio-data-preprocessing"><p>Content from <a href="data-preprocessing.html">Data Preprocessing</a></p>
<hr>
<p>Last updated on 2024-06-13 |
        
        <a href="https://github.com/smcclatchy/spatial-transcriptomics/edit/main/episodes/data-preprocessing.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What data files should I expect from the Visium assay?</li>
<li>Which data preprocessing steps are required to prepare the raw data
files for further analysis?</li>
<li>What software will we use for data preprocessing?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Explain how to use markdown with the new lesson template</li>
<li>Demonstrate how to include pieces of code, figures, and nested
challenge blocks</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section id="introduction"><h2 class="section-heading">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<p>The <a href="https://www.10xgenomics.com/support/software/space-ranger" class="external-link">Space
Ranger</a> software is a popular, though by no means only, set of
pipelines for preprocessing of Visium data. We focus on it here. It
provides the following output:</p>
<table class="table">
<colgroup>
<col width="48%">
<col width="52%">
</colgroup>
<thead><tr class="header">
<th>File Name</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>web_summary.html</td>
<td>Run summary metrics and plots in HTML format</td>
</tr>
<tr class="even">
<td>cloupe.cloupe</td>
<td>Loupe Browser visualization and analysis file</td>
</tr>
<tr class="odd">
<td>spatial/</td>
<td>Folder containing outputs that capture the spatiality of the
data.</td>
</tr>
<tr class="even">
<td>spatial/aligned_fiducials.jpg</td>
<td>Aligned fiducials QC image</td>
</tr>
<tr class="odd">
<td>spatial/aligned_tissue_image.jpg</td>
<td>Aligned CytAssist and Microscope QC image. Present only for
CytAssist workflow</td>
</tr>
<tr class="even">
<td>spatial/barcode_fluorescence_intensity.csv</td>
<td>CSV file containing the mean and standard deviation of fluorescence
intensity for each spot and each channel. Present for the fluorescence
image input specified by –darkimage</td>
</tr>
<tr class="odd">
<td>spatial/cytassist_image.tiff</td>
<td>Input CytAssist image in original resolution that can be used to
re-run the pipeline. Present only for CytAssist workflow</td>
</tr>
<tr class="even">
<td>spatial/detected_tissue_image.jpg</td>
<td>Detected tissue QC image.</td>
</tr>
<tr class="odd">
<td>spatial/scalefactors_json.json</td>
<td>Scale conversion factors for spot diameter and coordinates at
various image resolutions</td>
</tr>
<tr class="even">
<td>spatial/spatial_enrichment.csv</td>
<td>Feature spatial autocorrelation analysis using Moran’s I in CSV
format</td>
</tr>
<tr class="odd">
<td>spatial/tissue_hires_image.png</td>
<td>Downsampled full resolution image. The image dimensions depend on
the input image and slide version</td>
</tr>
<tr class="even">
<td>spatial/tissue_lowres_image.png</td>
<td>Full resolution image downsampled to 600 pixels on the longest
dimension</td>
</tr>
<tr class="odd">
<td>spatial/tissue_positions.csv</td>
<td>CSV containing spot barcode; if the spot was called under (1) or out
(0) of tissue, the array position, image pixel position x, and image
pixel position y for the full resolution image</td>
</tr>
<tr class="even">
<td>analysis/</td>
<td>Folder containing secondary analysis data including graph-based
clustering and K-means clustering (K = 2-10); differential gene
expression between clusters; PCA, t-SNE, and UMAP dimensionality
reduction.</td>
</tr>
<tr class="odd">
<td>metrics_summary.csv</td>
<td>Run summary metrics in CSV format</td>
</tr>
<tr class="even">
<td>probe_set.csv</td>
<td>Copy of the input probe set reference CSV file. Present for Visium
FFPE and CytAssist workflow</td>
</tr>
<tr class="odd">
<td>possorted_genome_bam.bam</td>
<td>Indexed BAM file containing position-sorted reads aligned to the
genome and transcriptome, annotated with barcode information</td>
</tr>
<tr class="even">
<td>possorted_genome_bam.bam.bai</td>
<td>Index for possorted_genome_bam.bam. In cases where the reference
transcriptome is generated from a genome with very long chromosomes
(&gt;512 Mbp), Space Ranger v2.0+ generates a
possorted_genome_bam.bam.csi index file instead.</td>
</tr>
<tr class="odd">
<td>filtered_feature_bc_matrix/</td>
<td>Contains only tissue-associated barcodes in MEX format. Each element
of the matrix is the number of UMIs associated with a feature (row) and
a barcode (column). This file can be input into third-party packages and
allows users to wrangle the barcode-feature matrix (e.g. to filter
outlier spots, run dimensionality reduction, normalize gene
expression).</td>
</tr>
<tr class="even">
<td><strong>filtered_feature_bc_matrix.h5</strong></td>
<td><strong>Same information as filtered_feature_bc_matrix/ but in HDF5
format.</strong></td>
</tr>
<tr class="odd">
<td>raw_feature_bc_matrices/</td>
<td>Contains all detected barcodes in MEX format. Each element of the
matrix is the number of UMIs associated with a feature (row) and a
barcode (column).</td>
</tr>
<tr class="even">
<td><strong>raw_feature_bc_matrix.h5</strong></td>
<td><strong>Same information as raw_feature_bc_matrices/ in HDF5
format.</strong></td>
</tr>
<tr class="odd">
<td>﻿</td>
<td>raw_probe_bc_matrix.h5</td>
</tr>
<tr class="even">
<td>molecule_info.h5</td>
<td>Contains per-molecule information for all molecules that contain a
valid barcode, valid UMI, and were assigned with high confidence to a
gene or protein barcode. This file is required for additional analysis
spaceranger pipelines including aggr, targeted-compare and
targeted-depth.</td>
</tr>
</tbody>
</table>
<p>Fortunately, you will not need to look at all of these files. We
provide a brief description for you in case you are curious or need to
look at one of the files for technical reasons.</p>
<p>The two files that you will use are “raw_feature_bc_matrix.h5” and
“filtered_feature_bc_matrix.h5”. These files have an “h5” suffix, which
means that they are HDF5 files. <a href="https://www.hdfgroup.org/solutions/hdf5/" class="external-link">HDF5</a> is a compressed
file format for storing complex high-dimensional data. HDF5 stands for
“Hierarchical Data Formats, version 5”. There is an R package designed
to read and write HDF5 files called <a href="https://bioconductor.org/packages/release/bioc/html/rhdf5.html" class="external-link">rhdf5</a>.
This was one of the packages which you installed during the lesson
setup.</p>
<p>Briefly, HDF5 organizes data into directories within the compressed
file. There are three “files” within the HDF5 file:</p>
<table class="table">
<colgroup>
<col width="53%">
<col width="46%">
</colgroup>
<thead><tr class="header">
<th>File Name</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>features.csv</td>
<td>Contains the features (i.e. genes in this case) for each row in the
data matrix.</td>
</tr>
<tr class="even">
<td>barcodes.csv</td>
<td>Contains the probe barcodes for each spot on the tissue block.</td>
</tr>
<tr class="odd">
<td>matrix.mtx</td>
<td>Contains the counts for each gene in each spot. Features
(e.g. genes) are in rows and barcodes (e.g. spots) are in columns.</td>
</tr>
</tbody>
</table></section><section id="set-up-environment"><h2 class="section-heading">Set up Environment<a class="anchor" aria-label="anchor" href="#set-up-environment"></a>
</h2>
<hr class="half-width">
<p>Go to the “File” menu and select “Open Project…”. Open the
“spatialRNA” project which you created in the workshop Setup.</p>
<p>First, we will load in some utility functions to make our lives a bit
easier. The <a href="https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/source" class="external-link">source</a>
function reads an R file and runs the code in it. In this case, this
will load several useful functions.</p>
<p>We will then load the libraries that we need for this lesson.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span><span class="kw">library</span><span class="op">(</span><span class="va">hdf5r</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span><span class="kw">library</span><span class="op">(</span><span class="va">Seurat</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">source</span><span class="op">(</span><span class="st">"https://raw.githubusercontent.com/smcclatchy/spatial-transcriptomics/main/code/spatial_utils.R"</span><span class="op">)</span></span></code></pre>
</div>
<p>Note that the <a href="https://here.r-lib.org/" class="external-link">here library</a>
helps you to find your files by taking care of the absolute path.</p>
</section><section id="load-raw-and-filtered-spatial-expression-data"><h2 class="section-heading">Load Raw and Filtered Spatial Expression Data<a class="anchor" aria-label="anchor" href="#load-raw-and-filtered-spatial-expression-data"></a>
</h2>
<hr class="half-width">
<p>In this course, we will use the <a href="https://satijalab.org/seurat/" class="external-link">Seurat</a> R environment, which was
originally designed for analysis of single-cell RNA-seq data, but has
been extended for spatial transcriptomics data. The Seurat website
provides helpful <a href="https://satijalab.org/seurat/articles/get_started_v5_new" class="external-link">vignettes</a>
and a concise command <a href="https://satijalab.org/seurat/articles/essential_commands" class="external-link">cheat
sheet</a>. <a href="https://bioconductor.org/packages/release/bioc/html/SpatialExperiment.html" class="external-link">SpatialExperiment</a>
in R and <a href="https://scanpy.readthedocs.io/en/stable/" class="external-link">scanpy</a>
in python, amongst others, are also frequently used in analyzing spatial
transcriptomics data.</p>
<p>We will use the <a href="https://www.rdocumentation.org/packages/Seurat/versions/5.0.1/topics/Load10X_Spatial" class="external-link">Load10X_Spatial</a>
function from Seurat to read in the spatial transcription data. These
are the data which you downloaded in the setup section.</p>
<p>First, we will read in the raw data for sample 151673.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">raw_st</span> <span class="op">&lt;-</span> <span class="fu">Load10X_Spatial</span><span class="op">(</span>data.dir      <span class="op">=</span> <span class="st">"./data/151673"</span>, </span>
<span>                          filename      <span class="op">=</span> <span class="st">"151673_raw_feature_bc_matrix.h5"</span>,</span>
<span>                          filter.matrix <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre>
</div>
<p>If you did not see any error messages, then the data loaded in and
you should see an “raw_st” object in your “Environment” tab on the
right.</p>

<p>Let’s look at “raw_st”, which is a Seurat object.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">raw_st</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>An object of class Seurat 
33538 features across 4992 samples within 1 assay 
Active assay: Spatial (33538 features, 0 variable features)
 1 layer present: counts
 1 spatial field of view present: slice1</code></pre>
</div>
<p>The output says that we have 33,538 “features” and 4,992 “samples”
with one assay. “Feature” is a generic term for anything that we
measured. In this case, we measured gene expression, so each feature is
a gene. Each “sample” is one spot on the spatial slide. So this tissue
sample has 33,538 genes assayed across 4,992 spots.</p>
<p>An experiment may have more than one “assay.” For example, you may
run both RNA sequencing and chromatin accessibility in the same set of
samples. In this case, we have one assay – RNA-seq. Each assay will, in
turn, have one or more “layers.” Each layer stores a different form of
the data. Initially, our Seurat object has a single “counts” layer,
holding the raw, un-normalized RNA-seq counts for each spot. Subsequent
downstream analyses can populate other layers, including normalized
counts (conventionally stored in a “data” layer) or variance-stabilized
counts (conventionally stored in a “scale.data” layer).</p>
<p>There is also a single image called “slice1” attached to the Seurat
object.</p>
<p>Next, we will load in the filtered data. Use the code above and look
in a file browser to identify the “filtered” file for sample 151673.</p>
<div id="challenge-1-read-in-filtered-hdf5-file-for-sample-151673." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-read-in-filtered-hdf5-file-for-sample-151673." class="callout-inner">
<h3 class="callout-title">Challenge 1: Read in filtered HDF5 file for
sample 151673.<a class="anchor" aria-label="anchor" href="#challenge-1-read-in-filtered-hdf5-file-for-sample-151673."></a>
</h3>
<div class="callout-content">
<p>Open a file browser and navigate to “Desktop/spatialRNA/data/151673”.
Can you find an HDF5 file (with an “.h5” suffix) that haw the word
“filtered” in it?</p>
<p>If so, read that file in and assign it to a variable called
“filter_st”.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">Solution 1</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filter_st</span> <span class="op">&lt;-</span> <span class="fu">Load10X_Spatial</span><span class="op">(</span>data.dir <span class="op">=</span> <span class="st">"./data/151673"</span>, </span>
<span>                             filename <span class="op">=</span> <span class="st">"151673_filtered_feature_bc_matrix.h5"</span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>
</div>
</div>
<p>Once you have the filtered data loaded in, look at the object.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filter_st</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>An object of class Seurat 
33538 features across 3639 samples within 1 assay 
Active assay: Spatial (33538 features, 0 variable features)
 1 layer present: counts
 1 spatial field of view present: slice1</code></pre>
</div>
<p>The raw and filtered data both have the same number of genes(33,538).
But the two objects have different numbers of spots. The raw data has
4,992 spots and the filtered data has 4,384 spots.</p>
<p>Look at the H&amp;E slide below and notice the grey “fiducial” spots
around the border. These are used by the spatial transcriptomics
software to “register” the H&amp;E image and the spatially-barcoded
sequences.</p>
<figure><img src="fig/tissue_hires_image_151673.png" alt="H &amp; E slide of sample 151673" class="figure mx-auto d-block"><div class="figcaption">H &amp; E slide of sample 151673</div>
</figure><div id="challenge-2-how-does-the-computer-know-how-to-orient-the-image." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-how-does-the-computer-know-how-to-orient-the-image." class="callout-inner">
<h3 class="callout-title">Challenge 2: How does the computer know how to
orient the image?.<a class="anchor" aria-label="anchor" href="#challenge-2-how-does-the-computer-know-how-to-orient-the-image."></a>
</h3>
<div class="callout-content">
<p>Look carefully at the spots in the H&amp;E image above. Are the spots
symmetric? Is there anything different about the spots that might help a
computer to assign up/down and left/right to the image?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2">Solution 2</h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>Look at the spots in each corner. In the upper-left, you will see the
following patterns:</p>
<figure><img src="fig/tissue_diagram_with_registration_spots.png" alt="Diagram showing tissue and registration spots in corners" class="figure mx-auto d-block"><div class="figcaption">Slide Diagram</div>
</figure><p>The patterns in each corner allow the spatial transcriptomics
software to orient the slide.</p>
</div>
</div>
</div>
</div>
</section><section id="add-spot-metadata"><h2 class="section-heading">Add Spot Metadata<a class="anchor" aria-label="anchor" href="#add-spot-metadata"></a>
</h2>
<hr class="half-width">
<p>Next, we will read a file containing information about whether each
spot is in the background or the tissue. This file does not contain
column names, although the next version of <a href="https://www.10xgenomics.com/support/software/space-ranger/latest/analysis/outputs/spatial-outputs" class="external-link">SpaceRanger
2.0</a>, which is used to process the data at the sequencing core,
should add column names to this file.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tissue_position</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="st">"./data/151673/spatial/tissue_positions_list.csv"</span>,</span>
<span>                            col_names <span class="op">=</span> <span class="cn">FALSE</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>                     <span class="fu">column_to_rownames</span><span class="op">(</span><span class="st">'X1'</span><span class="op">)</span></span>
<span><span class="fu">colnames</span><span class="op">(</span><span class="va">tissue_position</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">"in_tissue"</span>, </span>
<span>                               <span class="st">"array_row"</span>, </span>
<span>                               <span class="st">"array_col"</span>, </span>
<span>                               <span class="st">"pxl_row_in_fullres"</span>, </span>
<span>                               <span class="st">"pxl_col_in_fullres"</span><span class="op">)</span></span></code></pre>
</div>
<p>It is important to note that the order of the spots differs between
the Seurat object and the tissue position file. We need to reorder the
tissue positions to match the Seurat object. We can extract the spot
barcodes using the <a href="https://satijalab.org/seurat/reference/cells" class="external-link">Cells()</a>
function. This is named for the earlier versions of Seurat, which
processed single cell transcriptomic data. In this case, we are getting
<strong>spot</strong> IDs, even though the function is called
“Cells”.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tissue_position</span> <span class="op">&lt;-</span> <span class="va">tissue_position</span><span class="op">[</span><span class="fu">Cells</span><span class="op">(</span><span class="va">raw_st</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="fu">stopifnot</span><span class="op">(</span><span class="fu">rownames</span><span class="op">(</span><span class="va">tissue_position</span><span class="op">)</span> <span class="op">==</span> <span class="fu">Cells</span><span class="op">(</span><span class="va">raw_st</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Now that we have aligned the barcodes between the Seurat object and
the tissue positions, we can add the tissue positions to the Seurat
object’s metadata.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">raw_st</span>    <span class="op">&lt;-</span> <span class="fu">AddMetaData</span><span class="op">(</span>object <span class="op">=</span> <span class="va">raw_st</span>,    metadata <span class="op">=</span> <span class="va">tissue_position</span><span class="op">)</span></span>
<span><span class="va">filter_st</span> <span class="op">&lt;-</span> <span class="fu">AddMetaData</span><span class="op">(</span>object <span class="op">=</span> <span class="va">filter_st</span>, metadata <span class="op">=</span> <span class="va">tissue_position</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we will plot the spot annotation, indicating spots that are in
the tissue in blue and background spots in red.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">SpatialPlot</span><span class="op">(</span><span class="va">raw_st</span>, group.by <span class="op">=</span> <span class="st">"in_tissue"</span>, alpha <span class="op">=</span> <span class="fl">0.3</span><span class="op">)</span></span></code></pre>
</div>
<figure style="text-align: center"><img src="fig/data-preprocessing-rendered-unnamed-chunk-9-1.png" alt="Histology slide with tissue and background spots labelled" class="figure mx-auto d-block"><figcaption>
Spots identified in Tissue and Background
</figcaption></figure><p>The 10x platform tags each molecule with a Unique Molecular
Identifier (UMI). This allows us to keep only one unique sequencing read
per molecule and to exclude those arising from PCR duplication. We
expect most of the UMI counts to be in the tissue spots. The Seurat
object metadata contains the UMI count in each spot in a column called
“nCount_Spatial”. Let’s plot the UMI counts in the tissue and background
spots.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">raw_st</span><span class="op">@</span><span class="va">meta.data</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="fu">as.logical</span><span class="op">(</span><span class="va">in_tissue</span><span class="op">)</span>, <span class="va">nCount_Spatial</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">'UMI Counts in Tissue and Background'</span>,</span>
<span>         x     <span class="op">=</span> <span class="st">'In Tissue?'</span>,</span>
<span>         y     <span class="op">=</span> <span class="st">'Counts'</span><span class="op">)</span></span></code></pre>
</div>
<figure style="text-align: center"><img src="fig/data-preprocessing-rendered-unnamed-chunk-10-1.png" alt="Boxplot showing lower trancript counts in background area of slide" class="figure mx-auto d-block"><figcaption>
UMI Counts in Tissue and Background
</figcaption></figure><p>As expected, we see most of the counts in the tissue spots.</p>
<p>We can also plot the number of genes detected in each spot. Seurat
calls genes “features”, so we will plot the “nFeature_Spatial” value.
This is stored in the metadata of the Seurat object.</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">raw_st</span><span class="op">@</span><span class="va">meta.data</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="fu">as.logical</span><span class="op">(</span><span class="va">in_tissue</span><span class="op">)</span>, <span class="va">nFeature_Spatial</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">'Number of Genes in Tissue and Background'</span>,</span>
<span>         x     <span class="op">=</span> <span class="st">'In Tissue?'</span>,</span>
<span>         y     <span class="op">=</span> <span class="st">'Number of Genes'</span><span class="op">)</span></span></code></pre>
</div>
<figure style="text-align: center"><img src="fig/data-preprocessing-rendered-unnamed-chunk-11-1.png" alt="Boxplot showing lower numbers of genes in background area of slide" class="figure mx-auto d-block"><figcaption>
Number of Genes in Tissue and Background
</figcaption></figure><div id="challenge-3-why-might-there-be-umi-counts-outside-of-the-tissue-boundaries" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3-why-might-there-be-umi-counts-outside-of-the-tissue-boundaries" class="callout-inner">
<h3 class="callout-title">Challenge 3: Why might there be UMI counts
outside of the tissue boundaries?<a class="anchor" aria-label="anchor" href="#challenge-3-why-might-there-be-umi-counts-outside-of-the-tissue-boundaries"></a>
</h3>
<div class="callout-content">
<p>We expect UMI counts in the spots which overlap with the tissue
section. What reasons can you think of that might lead UMI counts to
occur in the background spots?</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3">Solution 3</h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">
<ol style="list-style-type: decimal">
<li>When the tissue section is lysed, some transcripts may leak out of
the cells and into the background region of the slide.</li>
</ol>
</div>
</div>
</div>
</div>
<p>Up to this point, we have been working with the raw, unfiltered data
to show you how the spots are filtered. However, in most workflows, you
will work directly with the filtered file. From this point forward, we
will work with the filtered data object.</p>
<p>Let’s plot the spots in the tissue in the filtered object to verify
that it is only using spots in the tissue.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot1</span> <span class="op">&lt;-</span> <span class="fu">SpatialDimPlot</span><span class="op">(</span><span class="va">filter_st</span>, alpha <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>           <span class="fu">NoLegend</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">plot2</span> <span class="op">&lt;-</span> <span class="fu">SpatialDimPlot</span><span class="op">(</span><span class="va">filter_st</span><span class="op">)</span> <span class="op">+</span> </span>
<span>           <span class="fu">NoLegend</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">plot1</span> <span class="op">|</span> <span class="va">plot2</span></span></code></pre>
</div>
<figure><img src="fig/data-preprocessing-rendered-unnamed-chunk-12-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure></section><section id="plot-umi-and-gene-counts-across-tissue"><h2 class="section-heading">Plot UMI and Gene Counts across Tissue<a class="anchor" aria-label="anchor" href="#plot-umi-and-gene-counts-across-tissue"></a>
</h2>
<hr class="half-width">
<p>Next, we want to look at the distribution of UMI counts and numbers
of genes in each spot across the tissue. This can be helpful in
identifying technical issues with the sample processing.</p>
<p>It is useful to first think about what we expect. In the publication
associated with this data, the authors show the structure that they
expect in this tissue section of the human dorsolateral prefrontal
cortex (DLPFC). In the figure below, they show a series of layers, from
L1 to L6, arranged from the upper right to the lower left. In the lower
left corner, they expect to see “White Matter” (WM). So we expect to see
some series of layers arranged from the upper right to the lower
left.</p>
<!-- ![Maynard et al, Figure 1c](fig/Maynard_etal_Fig1c.png){alt="Figure 1c"} -->
<p>We will use Seurat’s <a href="https://satijalab.org/seurat/reference/spatialplot" class="external-link">SpatialFeaturePlot</a>
function to look at these values. We can color the spots based on the
spot metadata stored in the Seurat object. You can find these column
names by looking at the “meta.data” slot of the Seurat object.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">filter_st</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                      orig.ident nCount_Spatial nFeature_Spatial in_tissue
AAACAAGTATCTCCCA-1 SeuratProject           8458             3586         1
AAACAATCTACTAGCA-1 SeuratProject           1667             1150         1
AAACACCAATAACTGC-1 SeuratProject           3769             1960         1
AAACAGAGCGACTCCT-1 SeuratProject           5433             2424         1
AAACAGCTTTCAGAAG-1 SeuratProject           4278             2264         1
AAACAGGGTCTATATT-1 SeuratProject           4004             2178         1
                   array_row array_col pxl_row_in_fullres pxl_col_in_fullres
AAACAAGTATCTCCCA-1        50       102               8468               9791
AAACAATCTACTAGCA-1         3        43               2807               5769
AAACACCAATAACTGC-1        59        19               9505               4068
AAACAGAGCGACTCCT-1        14        94               4151               9271
AAACAGCTTTCAGAAG-1        43         9               7583               3393
AAACAGGGTCTATATT-1        47        13               8064               3665</code></pre>
</div>
<p>To plot the UMI counts, we will use the “nCount_Spatial” column in
the spot metadata.</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot1</span> <span class="op">&lt;-</span> <span class="fu">SpatialDimPlot</span><span class="op">(</span><span class="va">filter_st</span>, alpha <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>           <span class="fu">NoLegend</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">plot2</span> <span class="op">&lt;-</span> <span class="fu">SpatialFeaturePlot</span><span class="op">(</span><span class="va">filter_st</span>, features <span class="op">=</span> <span class="st">"nCount_Spatial"</span><span class="op">)</span></span>
<span><span class="va">plot1</span> <span class="op">|</span> <span class="va">plot2</span></span></code></pre>
</div>
<figure style="text-align: center"><img src="fig/data-preprocessing-rendered-unnamed-chunk-14-1.png" alt="Figure showing UMI counts in each spot with varying intensity across the tissue" class="figure mx-auto d-block"><figcaption>
UMI Counts in each Spot
</figcaption></figure><p>In this case, we see a band of higher counts running from upper left
to lower right. There are also bands of lower counts above and below
this band. The band in the upper right corner may be due to the fissure
in the tissue. It is less clear why the expression is low in the
lower-left corner.</p>
<p>We can also look at the number of genes detected in each spot using
“nFeature_Spatial”.</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot1</span> <span class="op">&lt;-</span> <span class="fu">SpatialDimPlot</span><span class="op">(</span><span class="va">filter_st</span>, alpha <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>           <span class="fu">NoLegend</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">plot2</span> <span class="op">&lt;-</span> <span class="fu">SpatialFeaturePlot</span><span class="op">(</span><span class="va">filter_st</span>, features <span class="op">=</span> <span class="st">"nFeature_Spatial"</span><span class="op">)</span></span>
<span><span class="va">plot1</span> <span class="op">|</span> <span class="va">plot2</span></span></code></pre>
</div>
<figure style="text-align: center"><img src="fig/data-preprocessing-rendered-unnamed-chunk-15-1.png" alt="Figure showing number of genes detected in each spot with varying intensity across the tissue" class="figure mx-auto d-block"><figcaption>
Number of Genes in each Spot
</figcaption></figure><p>It is difficult to lay out a broad set of rules that will work for
all types of tissues and samples. Some tissues may have homogeneous UMI
counts across the section, while others may show variation in UMI counts
due to tissue structure. For example, in cancer tissue sections, stromal
cells tend to have lower counts than tumor cells and this should be
evident in a UMI count plot. In the brain sample below, we might expect
some variation in UMI counts in different layers of the brain.</p>
</section><section id="removing-genes-with-low-expression"><h2 class="section-heading">Removing Genes with Low Expression<a class="anchor" aria-label="anchor" href="#removing-genes-with-low-expression"></a>
</h2>
<hr class="half-width">
<p>In order to build this lesson, we needed to reduce the size of the
data. To do this, we are going to filter out genes that have no
expression across the cells.</p>
<p>How many genes to we have before filtering?</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">paste</span><span class="op">(</span><span class="fu">nrow</span><span class="op">(</span><span class="va">filter_st</span><span class="op">)</span>, <span class="st">"genes."</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "33538 genes."</code></pre>
</div>
<p>Next, we will get the raw counts, calculate the sum of each gene’s
exprsssion across all spots, and filter the Seurat object to retain
genes with summed counts greater than zero.</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span>     <span class="op">&lt;-</span> <span class="fu">LayerData</span><span class="op">(</span><span class="va">filter_st</span>, <span class="st">'counts'</span><span class="op">)</span></span>
<span><span class="va">gene_sums</span>  <span class="op">&lt;-</span> <span class="fu">rowSums</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span></span>
<span><span class="va">keep_genes</span> <span class="op">&lt;-</span> <span class="fu">which</span><span class="op">(</span><span class="va">gene_sums</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="va">filter_st</span>  <span class="op">&lt;-</span> <span class="va">filter_st</span><span class="op">[</span><span class="va">keep_genes</span>,<span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning: Not validating Centroids objects
Not validating Centroids objects</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning: Not validating FOV objects
Not validating FOV objects
Not validating FOV objects
Not validating FOV objects
Not validating FOV objects
Not validating FOV objects</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning: Not validating Seurat objects</code></pre>
</div>
<p>How many genes to we have after filtering?</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">paste</span><span class="op">(</span><span class="fu">nrow</span><span class="op">(</span><span class="va">filter_st</span><span class="op">)</span>, <span class="st">"genes."</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "21842 genes."</code></pre>
</div>
<p>So we removed about 11,700 genes that had zero counts.</p>
</section><section id="conclusion"><h2 class="section-heading">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<hr class="half-width">
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>The 10x Space Ranger pipeline provides you with an unfiltered and a
filtered data file.</li>
<li>The HDF5 file ends with an “h5” extension and contains the barcodes,
features (genes), and counts matrix.</li>
<li>Seurat is one, of several, popular environments for analyzing
spatial transcriptomics data.</li>
<li>It is important to know something about the structure of the tissue
which you are analyzing.</li>
<li>Plotting total counts and genes in each spot may help to identify
quality control issues.</li>
</ul>
</div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</section></section><section id="aio-remove-low-quality-spots"><p>Content from <a href="remove-low-quality-spots.html">Remove Low-quality Spots</a></p>
<hr>
<p>Last updated on 2024-06-13 |
        
        <a href="https://github.com/smcclatchy/spatial-transcriptomics/edit/main/episodes/remove-low-quality-spots.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I remove low-quality spots?</li>
<li>What kinds of problems produce low-quality spots?</li>
<li>What happens if I skip quality control and proceed with
analysis?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand how to look for low quality spots.</li>
<li>Decide whether to retain or remove low quality spots.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section id="introduction"><h2 class="section-heading">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<p>Spatial transcriptomics involves a complex process that may involve
some technical failures. If the processing of the entire slide fails, it
should be obvious due to a large number of gene appearing in spots
outside of the tissue or low UMIs across the whole tissue.</p>
<p>However, there may also be variation in spot quality in a slide that
has largely high-quality spots. These artifacts are much rarer than in
single-cell transcriptomics because the process of tissue sectioning is
less disruptive than tissue dissociation. Because of this, we recommend
light spot filtering.</p>
<p>There are three metrics that we will use to identify and remove
low-quality spots:</p>
<ol style="list-style-type: decimal">
<li>Mitochondrial gene expression,</li>
<li>Total UMI counts,</li>
<li>Number of detected genes.</li>
</ol>
<p>During tissue processing, it is possible that some cells will be
lysed, spilling out the transcripts, but retaining the mitochondria.
These spots will appear with much higher mitochondrial gene expression.
We will also examine the total UMI counts and number of genes detected
in each spot because high counts may indicate spots with lysed cells
whose contents bled into other spots.</p>
<p>However, these metrics may be tissue-dependent. In some tissues,
there may be biological reasons for differential expression across the
tissue. For example, in a cancer sample, mitochondrial or total gene
expression may vary between stromal and tumor regions. It will be
important for you to familiarize yourself with the structure of the
tissue that you are analyzing in order to make rational judgments about
filtering.</p>
</section><section id="filtering-by-mitochondrial-gene-count"><h2 class="section-heading">Filtering by Mitochondrial Gene Count<a class="anchor" aria-label="anchor" href="#filtering-by-mitochondrial-gene-count"></a>
</h2>
<hr class="half-width">
<p>In single-cell RNA sequencing experiments, the tissue is digested and
the cells are dissociated. This mechanical disruption is stressful to
the cells and some of them are damaged in the process. Elevated levels
of mitochondrial genes often indicate cell death or damage because, when
a cell’s membrane is compromised, it loses most cytoplasmic content
while retaining mitochondrial RNA. Therefore, spots with high
mitochondrial RNA may represent damaged or dying cells, and their
exclusion helps focus the analysis on healthy, intact cells.</p>
<p>More details on this relationship can be found in the <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0888-1" class="external-link">literature
on mitochondrial DNA and cell death</a>.</p>
<p>However, in spatial transcriptomics, the tissue is either frozen or
formalin-fixed and there is much less mechanical disruption of the
tissue. Because of this, we are skeptical of the value of filtering
spots based on mitochondrial gene counts.</p>
<p>For completeness, we show how to obtain the mitochondrial genes,
calculate the percentage of counts produced by these genes in each spot,
and add this to the Seurat object metadata.</p>
<p>We will search the gene symbols in the feature metadata to identify
mitochondrial genes. We do not need to find all genes in these
categories, so we will search for genes with symbols that start with
“MT”.</p>
<p>The Seurat object is designed to be flexible and may contains several
data types. For example, it may contain both gene counts and open
chromatin peaks. In this analysis, the Seurat object only contains gene
counts. The different types of data are called “Layers” in Seurat and
may be accessed using the <a href="https://satijalab.github.io/seurat-object/reference/Layers.html" class="external-link">Layers</a>
function.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">Layers</span><span class="op">(</span><span class="va">filter_st</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "counts"</code></pre>
</div>
<p>This tells us that the “filter_st” object only contains one data
Layer called “counts”. We can access this using the <a href="https://satijalab.github.io/seurat-object/reference/Layers.html" class="external-link">LayerData</a>
function using “counts” as an argument.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu">LayerData</span><span class="op">(</span><span class="va">filter_st</span>, <span class="st">'counts'</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>6 x 5 sparse Matrix of class "dgCMatrix"
            AAACAAGTATCTCCCA-1 AAACAATCTACTAGCA-1 AAACACCAATAACTGC-1
MIR1302-2HG                  .                  .                  .
AL627309.1                   .                  .                  .
AL669831.5                   .                  .                  .
FAM87B                       .                  .                  .
LINC00115                    .                  .                  .
FAM41C                       .                  .                  .
            AAACAGAGCGACTCCT-1 AAACAGCTTTCAGAAG-1
MIR1302-2HG                  .                  .
AL627309.1                   .                  .
AL669831.5                   .                  .
FAM87B                       .                  .
LINC00115                    .                  .
FAM41C                       .                  .</code></pre>
</div>
<p>The output above may look odd to you since there are no numbers.
Notice that the text above the table says “sparse Matrix”. Many of the
counts in the file are likely to be zero. Due to the manner in which
numbers are stored in computer memory, a zero takes up as much space as
a number. If we had to store all of these zeros, it would consume a lot
of computer memory. A sparse matrix is a special data structure which
only stores the non-zero values. In the table above, each dot (.)
represents a position with zero counts.</p>

<p>If we look at another part of the “counts” matrix, we can see
numbers.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span><span class="op">[</span><span class="fl">20000</span><span class="op">:</span><span class="fl">20005</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>6 x 5 sparse Matrix of class "dgCMatrix"
       AAACAAGTATCTCCCA-1 AAACAATCTACTAGCA-1 AAACACCAATAACTGC-1
DNAJB1                  .                  .                  .
TECR                    3                  1                  .
NDUFB7                  3                  1                  1
ZNF333                  .                  .                  .
ADGRE2                  .                  .                  .
OR7C1                   .                  .                  .
       AAACAGAGCGACTCCT-1 AAACAGCTTTCAGAAG-1
DNAJB1                  1                  .
TECR                    2                  1
NDUFB7                  2                  4
ZNF333                  .                  .
ADGRE2                  .                  .
OR7C1                   .                  .</code></pre>
</div>
<p>As you can see in the table above, the gene symbols are stored in the
rownames of “counts”. We will find find mitochondrial genes by searching
for gene symbols which start with “MT”.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mito_pattern</span> <span class="op">&lt;-</span> <span class="st">'^[Mm][Tt]-'</span></span>
<span><span class="va">mito_genes</span>   <span class="op">&lt;-</span> <span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">[</span><span class="fu">grep</span><span class="op">(</span><span class="va">mito_pattern</span>, <span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">mito_genes</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code> [1] "MT-ND1"  "MT-ND2"  "MT-CO1"  "MT-CO2"  "MT-ATP8" "MT-ATP6" "MT-CO3" 
 [8] "MT-ND3"  "MT-ND4L" "MT-ND4"  "MT-ND5"  "MT-ND6"  "MT-CYB" </code></pre>
</div>
<p>We now have a set of mitochondrial genes. We will use these genes to
estimate the percentage of gene counts expressed by mitochondrial genes
in each cell and add this to the Seurat object. We will pass the
mitochondrial gene symbols into <a href="https://satijalab.org/seurat/reference/percentagefeatureset" class="external-link">PercentageFeatureSet</a>,
which will perform the calculation for us.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filter_st</span><span class="op">[[</span><span class="st">"percent.mt"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">PercentageFeatureSet</span><span class="op">(</span><span class="va">filter_st</span>, pattern <span class="op">=</span> <span class="va">mito_pattern</span><span class="op">)</span></span></code></pre>
</div>
<p>This syntax adds a new column called “percent.mt” to the spot
metadata.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">colnames</span><span class="op">(</span><span class="va">filter_st</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "orig.ident"         "nCount_Spatial"     "nFeature_Spatial"  
[4] "in_tissue"          "array_row"          "array_col"         
[7] "pxl_row_in_fullres" "pxl_col_in_fullres" "percent.mt"        </code></pre>
</div>

<p>Let’s look at histograms of the ribosomal and mitochondrial gene
percentages.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">hist</span><span class="op">(</span><span class="fu">FetchData</span><span class="op">(</span><span class="va">filter_st</span>, <span class="st">"percent.mt"</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>,   main <span class="op">=</span> <span class="st">"% Mitochondrial Genes"</span>,</span>
<span>     xlab <span class="op">=</span> <span class="st">"%"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/remove-low-quality-spots-rendered-unnamed-chunk-8-1.png" alt="mito" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In these plots, we are looking for spots which are outside of a
normal distribution. It is difficult to generalize how to select a
filtering threshold. Some tissue or cell types may have higher
mitochondrial gene expression. Further, heterogeneous tissues may have
subsets of cells with differing levels of mitochondrial gene
expression.</p>
<p>Let’s visually check whether the mitochondrial gene expression is
normally distributed.</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mito_expr</span> <span class="op">&lt;-</span> <span class="fu">FetchData</span><span class="op">(</span><span class="va">filter_st</span>, <span class="st">"percent.mt"</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span></span>
<span><span class="fu">qqnorm</span><span class="op">(</span><span class="va">mito_expr</span>, las <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu">qqline</span><span class="op">(</span><span class="va">mito_expr</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/remove-low-quality-spots-rendered-unnamed-chunk-9-1.png" alt="mito" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In this case, there may be a reason to filter out spots with greater
than 35% mitochondrial counts.</p>
</section><section id="filter-by-umi-count-and-number-of-detected-genes"><h2 class="section-heading">Filter by UMI Count and Number of Detected Genes<a class="anchor" aria-label="anchor" href="#filter-by-umi-count-and-number-of-detected-genes"></a>
</h2>
<hr class="half-width">
<p>In the previous lesson, we plotted number of UMIs and genes detected
spatially across the tissue. Let’s plot these values again, but this
time as a histogram.</p>

<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">layout</span><span class="op">(</span><span class="fu">matrix</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">hist</span><span class="op">(</span><span class="fu">FetchData</span><span class="op">(</span><span class="va">filter_st</span>, <span class="st">"nCount_Spatial"</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>,</span>
<span>     main <span class="op">=</span> <span class="st">'UMIs per Spot'</span>, xlab <span class="op">=</span> <span class="st">'Counts'</span>, las <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu">hist</span><span class="op">(</span><span class="fu">FetchData</span><span class="op">(</span><span class="va">filter_st</span>, <span class="st">"nFeature_Spatial"</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>, </span>
<span>     main <span class="op">=</span> <span class="st">'Genes per Spot'</span>, xlab <span class="op">=</span> <span class="st">'Genes'</span>, las <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/remove-low-quality-spots-rendered-unnamed-chunk-10-1.png" alt="mito" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Again, most of the spots fall within a reasonable distribution. The
right tail of the distribution is not very thick. We might filter spots
with over 14,000 UMIs or 3,000 genes. We will use these thresholds to
add a “keep” column to the Seurat object metadata.</p>
<p>First, we will create variables for each threshold. While we could
type the numbers directly into the logical comparison statements,
creating variables makes it clear what each number represents.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mito_thr</span>     <span class="op">&lt;-</span> <span class="fl">32</span></span>
<span><span class="va">counts_thr</span>   <span class="op">&lt;-</span> <span class="fl">14000</span></span>
<span><span class="va">features_thr</span> <span class="op">&lt;-</span> <span class="fl">5000</span></span></code></pre>
</div>
<p>Next, we will create a “keep” variable which will be TRUE for spots
that we want to keep.</p>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu">FetchData</span><span class="op">(</span><span class="va">filter_st</span>, <span class="st">"percent.mt"</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>       <span class="op">&lt;</span> <span class="va">mito_thr</span></span>
<span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu">FetchData</span><span class="op">(</span><span class="va">filter_st</span>, <span class="st">"nCount_Spatial"</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>   <span class="op">&lt;</span> <span class="va">counts_thr</span>   <span class="op">&amp;</span> <span class="va">keep</span></span>
<span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu">FetchData</span><span class="op">(</span><span class="va">filter_st</span>, <span class="st">"nFeature_Spatial"</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;</span> <span class="va">features_thr</span> <span class="op">&amp;</span> <span class="va">keep</span></span>
<span></span>
<span><span class="va">filter_st</span><span class="op">$</span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="va">keep</span></span></code></pre>
</div>
<p>Now let’s plot the spots on the tissue and color them based on
whether we will keep them.</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">SpatialDimPlot</span><span class="op">(</span><span class="va">filter_st</span>, group.by <span class="op">=</span> <span class="st">"keep"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/remove-low-quality-spots-rendered-unnamed-chunk-13-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>When you examine the spots that have been flagged, it is important to
look for patterns. If a contiguous section of tissue contains spots that
will be removed, it is worth looking at the histology slide to see if
there are structures that correlate with the removed spots. If it is a
section of necrotic tissue, then, depending on your experimental
question, you may want to remove those spots. But you should always look
for patterns in the removed spots and convince yourself that they are
not biasing your results.</p>
<div id="challenge-1-change-the-spot-filtering-thresholds." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-change-the-spot-filtering-thresholds." class="callout-inner">
<h3 class="callout-title">Challenge 1: Change the spot filtering
thresholds.<a class="anchor" aria-label="anchor" href="#challenge-1-change-the-spot-filtering-thresholds."></a>
</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>Change the threshold for the number of UMIs per spot to keep spots
with <strong>more</strong> than 2000 counts. Note that we are filtering
on the lower side of the distribution.</li>
<li>Add new variable called “keep_counts” to the Seurat object.</li>
<li>Plot the spot overlaid on the tissue section, colored by whether you
are keeping them.</li>
</ol>
<p>Is there a pattern to the removed spots that seems to correlate with
the tissue structure?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">Solution 1</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">new_counts_thr</span> <span class="op">&lt;-</span> <span class="fl">2000</span></span>
<span><span class="va">keep_counts</span>    <span class="op">&lt;-</span> <span class="fu">FetchData</span><span class="op">(</span><span class="va">filter_st</span>, <span class="st">"nCount_Spatial"</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&gt;</span> <span class="va">new_counts_thr</span></span>
<span><span class="va">filter_st</span><span class="op">$</span><span class="va">keep_counts</span> <span class="op">&lt;-</span> <span class="va">keep_counts</span></span>
<span><span class="fu">SpatialDimPlot</span><span class="op">(</span><span class="va">filter_st</span>, group.by <span class="op">=</span> <span class="st">"keep_counts"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/remove-low-quality-spots-rendered-unnamed-chunk-14-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Note that the spots that we have flagged seem to correspond to
stripes in the tissue section. These may be regions of the brain which
have lower levels of gene expression, so we may want to revise or remove
this threshold. Overall, this exercise shows that it is important to use
judgement when filtering spots.</p>
</div>
</div>
</div>
</div>
<p>Note that we will only remove a few spots in this filtering step.</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">table</span><span class="op">(</span><span class="fu">FetchData</span><span class="op">(</span><span class="va">filter_st</span>, <span class="st">"keep"</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>
FALSE  TRUE 
    6  3633 </code></pre>
</div>
<p>We can remove the spots directly using the following syntax. In this
case, the “columns” of the Seurat object correspond to the spots.</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filter_st</span> <span class="op">&lt;-</span> <span class="va">filter_st</span><span class="op">[</span>,<span class="va">keep</span><span class="op">]</span></span></code></pre>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>Spot filtering should be light.</li>
<li>Inspect the spots that you are filtering to confirm that you are not
discarding important tissue structures.</li>
</ul>
</div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</section></section><section id="aio-apply-normalization-methods"><p>Content from <a href="apply-normalization-methods.html">Normalization in Spatial Transcriptomics</a></p>
<hr>
<p>Last updated on 2024-06-13 |
        
        <a href="https://github.com/smcclatchy/spatial-transcriptomics/edit/main/episodes/apply-normalization-methods.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do we determine the necessity of normalization in spatial
transcriptomics?</li>
<li>What insights do additional modalities like H&amp;E staining provide
in assessing normalization needs?</li>
<li>How do specific normalization techniques like SCTransform and log
scaling work and when should they be applied?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Assess the need for normalization using both spatial transcriptomics
data and ancillary modalities like H&amp;E staining.</li>
<li>Understand the specific applications and mechanisms of normalization
techniques such as SCTransform and log scaling.</li>
<li>Implement adaptive normalization strategies that accurately reflect
both absolute and relative cellular information.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section id="understanding-normalization-in-spatial-transcriptomics"><h2 class="section-heading">Understanding Normalization in Spatial Transcriptomics<a class="anchor" aria-label="anchor" href="#understanding-normalization-in-spatial-transcriptomics"></a>
</h2>
<hr class="half-width">
<p>Normalization in spatial transcriptomics must be carefully tailored
to each dataset, balancing the technical corrections with the
preservation of biologically meaningful signals. There are two artifacts
of the data that we need to adjust for:</p>
<ol style="list-style-type: decimal">
<li>the difference in total counts across spots, and</li>
<li>the difference in variance across genes.</li>
</ol>
<p>For the first point, each spot may have a different number of total
counts, which makes it difficult to compare gene expression levels
between spots. On the other hand, different spots may contain different
types of cells, which may express differing numbers of transcripts. So
there is a balance between normalizing all spots to have the same total
counts and leaving some variation in total counts which may be due to
the biology of the tissue.</p>
<p>For the second point, in order to compare gene expression values
between different genes, the within-gene variance should be similar
between genes. This is because many statistical tests require that the
within-group variance be the same. As you’ll see below, there is a
relationship between the mean and variance of genes that will allow us
to correct for this difference.</p>
<div class="section level3">
<h3 id="assessing-normalization-needs">Assessing Normalization Needs<a class="anchor" aria-label="anchor" href="#assessing-normalization-needs"></a>
</h3>
<div class="section level4">
<h4 id="using-he-staining-to-guide-normalization">Using H&amp;E Staining to Guide Normalization<a class="anchor" aria-label="anchor" href="#using-he-staining-to-guide-normalization"></a>
</h4>
<p>Hematoxylin and Eosin (H&amp;E) staining is critical for preliminary
assessments of tissue sections. It highlights structural and
pathological features, guiding the interpretation of transcriptomic
data. For example, high RNA counts in a necrotic region, typically
characterized by reduced cellular material, might suggest technical
artifacts, indicating a need for normalization.</p>
</div>
<div class="section level4">
<h4 id="total-counts-per-spot">Total Counts per Spot<a class="anchor" aria-label="anchor" href="#total-counts-per-spot"></a>
</h4>
<p>The spots are arranged in column in the data matrix. We will look at
the distribution of total counts per spot by summing the counts in each
column and making a histogram.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu">LayerData</span><span class="op">(</span><span class="va">filter_st</span>, layer <span class="op">=</span> <span class="st">'counts'</span><span class="op">)</span></span>
<span><span class="fu">hist</span><span class="op">(</span><span class="fu">colSums2</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fl">100</span>, </span>
<span>     main <span class="op">=</span> <span class="st">"Histogram of Counts per Spot"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/apply-normalization-methods-rendered-unnamed-chunk-2-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>As you can see, the total counts per spot ranges cross three orders
of magnitude. Some of this may be due to the biology of the tissue,
i.e. some cells may express more transcripts. But some of this may be
due to technical issues.</p>
</div>
<div class="section level4">
<h4 id="mean-variance-plotting">Mean-Variance Plotting<a class="anchor" aria-label="anchor" href="#mean-variance-plotting"></a>
</h4>
<p>Mean-variance plots are an essential tool for assessing gene
expression variability relative to the mean expression levels across
different spots. By plotting the variance of gene expression against the
mean expression level, researchers can identify genes with variance that
deviates significantly from what would be expected under normal
biological conditions. This can be particularly useful for spotting
genes that are overly influenced by technical artifacts or biological
outliers, suggesting the corresponding normalization choices.</p>
<p>In the raw counts, each spot will have different total number of
counts. This is termed the “library size”. Since each spot has a
different number of counts, it will be difficult to compare gene
expression values between them in a meaningful way because the
denominator (total spot counts) is different in each spot.</p>
</div>
</div>
<div class="section level3">
<h3 id="normalization-techniques-and-their-implications">Normalization Techniques and Their Implications<a class="anchor" aria-label="anchor" href="#normalization-techniques-and-their-implications"></a>
</h3>
<div class="section level4">
<h4 id="sctransform">SCTransform<a class="anchor" aria-label="anchor" href="#sctransform"></a>
</h4>
<p><a href="https://satijalab.org/seurat/articles/sctransform_vignette.html" class="external-link">SCTransform</a>
is a normalization method for single-cell and spatial transcriptomics
that uses a regularized negative binomial regression to stabilize
variance across expression levels <a href="https://link.springer.com/article/10.1186/s13059-021-02584-9" class="external-link">Choudhary
et al.</a>. It selects highly variable genes and corrects for technical
noise by modeling gene expression counts with Pearson residuals. This
approach effectively adjusts for confounding factors such as sequencing
depth, facilitating more accurate downstream analyses like
clustering.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filter_st</span> <span class="op">&lt;-</span> <span class="fu">SCTransform</span><span class="op">(</span><span class="va">filter_st</span>, </span>
<span>                         assay <span class="op">=</span> <span class="st">"Spatial"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Running SCTransform on assay: Spatial</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Running SCTransform on layer: counts</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>vst.flavor='v2' set. Using model with fixed slope and excluding poisson genes.</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning: replacing previous import 'S4Arrays::makeNindexFromArrayViewport' by
'DelayedArray::makeNindexFromArrayViewport' when loading 'SummarizedExperiment'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Variance stabilizing transformation of count matrix of size 17996 by 3633</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Model formula is y ~ log_umi</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Get Negative Binomial regression parameters per gene</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Using 2000 genes, 3633 cells</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Found 105 outliers - those will be ignored in fitting/regularization step</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Second step: Get residuals using fitted parameters for 17996 genes</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Computing corrected count matrix for 17996 genes</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Calculating gene attributes</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Wall clock passed: Time difference of 21.15982 secs</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Determine variable features</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Centering data matrix</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Set default assay to SCT</code></pre>
</div>
<p>The SCTransform method added a new Assay called “SCT.”</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">Assays</span><span class="op">(</span><span class="va">filter_st</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "Spatial" "SCT"    </code></pre>
</div>
<p>It made this new Assay the default. Be aware that Seurat functions
often operate on the DefaultAssay.</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">DefaultAssay</span><span class="op">(</span><span class="va">filter_st</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "SCT"</code></pre>
</div>
<p>Within this new “SCT” Assay, SCTransform has created three
Layers.</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">Layers</span><span class="op">(</span><span class="va">filter_st</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "counts"     "data"       "scale.data"</code></pre>
</div>
<p>As you can see by reading its documentation, these new Layers are
“counts” (counts corrected for differences in sequencing depth between
cells), “data” (<code>log1p</code> transformation or (log(1+x)) of the
corrected counts), and “scale.data” (scaled Pearson residuals, i.e., the
difference between an observed count and its expected value under the
model used by SCTransform, divided by the standard deviation in that
count under the model).</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">?</span><span class="va">SCTransform</span></span></code></pre>
</div>
<p>Notice, in particular, that the “counts” Layers in the “Spatial” and
“SCT” Assays are different. As mentioned above, the latter have been
corrected for differences in sequencing depth between cells. As such,
the distribution in total counts per cell is much more uniform in the
latter case.</p>
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">layout</span><span class="op">(</span><span class="fu">matrix</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">raw_counts_spatial</span> <span class="op">&lt;-</span> <span class="fu">LayerData</span><span class="op">(</span><span class="va">filter_st</span>, layer <span class="op">=</span> <span class="st">"counts"</span>, assay <span class="op">=</span> <span class="st">"Spatial"</span><span class="op">)</span></span>
<span><span class="fu">hist</span><span class="op">(</span><span class="fu">colSums2</span><span class="op">(</span><span class="va">raw_counts_spatial</span><span class="op">)</span>, main <span class="op">=</span> <span class="st">"Raw counts (Spatial)"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">corrected_counts_sct</span> <span class="op">&lt;-</span> <span class="fu">LayerData</span><span class="op">(</span><span class="va">filter_st</span>, layer <span class="op">=</span> <span class="st">"data"</span>, assay <span class="op">=</span> <span class="st">"SCT"</span><span class="op">)</span></span>
<span><span class="fu">hist</span><span class="op">(</span><span class="fu">colSums2</span><span class="op">(</span><span class="va">corrected_counts_sct</span><span class="op">)</span>, main <span class="op">=</span> <span class="st">"Corrected counts (SCT)"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/apply-normalization-methods-rendered-unnamed-chunk-8-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Let’s plot the mean versus the variance of the genes using the <a href="https://satijalab.org/seurat/reference/variablefeatureplot" class="external-link">VariableFeaturePlot</a>
function.</p>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">VariableFeaturePlot</span><span class="op">(</span><span class="va">filter_st</span>, log <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Variable Features - SCT"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/apply-normalization-methods-rendered-unnamed-chunk-9-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The geometric mean (mean of the log counts) is shown on the X-axis
and the residual variance is on the Y-axis. Each point shows one gene.
By default, Seurat selects a set of 3,000 variable genes which are
colored in red. The variance is largely stable across a range of mean
expression values.</p>
</div>
<div class="section level4">
<h4 id="lognormalize">LogNormalize<a class="anchor" aria-label="anchor" href="#lognormalize"></a>
</h4>
<p>LogNormalize is a specific normalization method that scales gene
expression data to account for differences in spot-specific total RNA
counts. This process involves dividing the raw gene expression counts in
each spot by the total counts in that cell, multiplying by a scale
factor, and then applying a natural logarithm transformation using log1p
(log(x+1)). This method helps in reducing the skewness caused by highly
expressed genes and stabilizes the variance across the dataset, making
it more suitable for downstream analytical comparisons.</p>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lognorm_st</span> <span class="op">&lt;-</span> <span class="fu">NormalizeData</span><span class="op">(</span><span class="va">filter_st</span>, </span>
<span>                           assay                <span class="op">=</span> <span class="st">"Spatial"</span>, </span>
<span>                           normalization.method <span class="op">=</span> <span class="st">"LogNormalize"</span>, </span>
<span>                           scale.factor         <span class="op">=</span> <span class="fl">1e6</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Normalizing layer: counts</code></pre>
</div>
<p>In this normalization, feature counts for each spot are divided by
the total counts for that spot and multiplied by the scale.factor. This
is then natural-log transformed using log1p (log(x + 1)). The
“scale.factor” argument has a default of 10,000. Here, we selected
1,000,000 because it made the mean- variance relationship somewhat
flatter.</p>
<p>The Log Normalization adds a “data” object to the Seurat object. Note
that, unlike SCTransform, NormalizeData does not set the DefaultAssay,
so we need to do explicitly.</p>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">DefaultAssay</span><span class="op">(</span><span class="va">lognorm_st</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"Spatial"</span></span>
<span><span class="fu">Layers</span><span class="op">(</span><span class="va">lognorm_st</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "counts" "data"  </code></pre>
</div>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lognorm_st</span> <span class="op">&lt;-</span> <span class="fu">FindVariableFeatures</span><span class="op">(</span><span class="va">lognorm_st</span>,  </span>
<span>                                  assay            <span class="op">=</span> <span class="st">'Spatial'</span>, </span>
<span>                                  selection.method <span class="op">=</span> <span class="st">"mean.var.plot"</span>,</span>
<span>                                  nfeatures        <span class="op">=</span> <span class="fl">3000</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Finding variable features for layer data</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb34">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">VariableFeaturePlot</span><span class="op">(</span><span class="va">lognorm_st</span>, log <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Variable Features - Log-norm"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
</div>
<figure><img src="fig/apply-normalization-methods-rendered-unnamed-chunk-12-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>It is not entirely clear what is plotted on each axis. As best we can
tell, the mean is on the X-axis and the standardized variance is on the
Y-axis.</p>
<p>Our goal is to maintain consistent variance across average gene
expression levels. Ideally, when plotting variance against mean
expression, we aim for a straight, flat line. This comparison helps us
determine which normalization method best suits our data.</p>
<div id="challenge-1-compare-mean-variance-plots" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-compare-mean-variance-plots" class="callout-inner">
<h3 class="callout-title">Challenge 1: Compare Mean-Variance Plots<a class="anchor" aria-label="anchor" href="#challenge-1-compare-mean-variance-plots"></a>
</h3>
<div class="callout-content">
<p>Above, we created mean-variance plots for the SCT and Log
normalizations. Which method does a better job of stabilizing the
variance across genes? Turn to the person next to you and put the
Log-normalized plot on one of your screens and the SCT transform plot on
the other person’s screen. Discuss the mean-variance relationship in
each plot and decide which one you think stabilizes the variance across
genes better.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">Solution 1</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>In the Log-normalization plot, the variation in the variance is wider
for genes with lower expression and lower for genes with higher
expression. Also, the range of variances is quite wide. In the SCT plot,
the variance of the low- expressed genes is smaller and it rises as mean
expression increases. The height of the variances becomes stable for
genes with mean expression over about 100. For this reason, we prefer
the SCT transform.</p>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="comparing-normalizations">Comparing Normalizations<a class="anchor" aria-label="anchor" href="#comparing-normalizations"></a>
</h3>

<p>First, let’s look at the total counts per spot after normalization by
each method. The normalized counts are stored in the “data” slot of the
Seurat object.</p>
<div class="codewrapper sourceCode" id="cb36">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">layout</span><span class="op">(</span><span class="fu">matrix</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">counts_log</span> <span class="op">&lt;-</span> <span class="fu">LayerData</span><span class="op">(</span><span class="va">lognorm_st</span>, layer <span class="op">=</span> <span class="st">"data"</span><span class="op">)</span></span>
<span><span class="fu">hist</span><span class="op">(</span><span class="fu">colSums2</span><span class="op">(</span><span class="va">counts_log</span><span class="op">)</span>, main <span class="op">=</span> <span class="st">"Log-norm"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">counts_sct</span> <span class="op">&lt;-</span> <span class="fu">LayerData</span><span class="op">(</span><span class="va">filter_st</span>, layer <span class="op">=</span> <span class="st">"data"</span><span class="op">)</span></span>
<span><span class="fu">hist</span><span class="op">(</span><span class="fu">colSums2</span><span class="op">(</span><span class="va">counts_sct</span><span class="op">)</span>, main <span class="op">=</span> <span class="st">"SCT"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/apply-normalization-methods-rendered-unnamed-chunk-13-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Notice that the log-normalization has a range of total counts per
spot that ranges across several orders of magnitude. The SCT transform
has a more uniform distribution of total counts and spans a factor of
three, from ~1000 to ~2700.</p>
<p>Next, we will compare the mean-variance plots between the two
methods.</p>
<div class="codewrapper sourceCode" id="cb37">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top15</span> <span class="op">&lt;-</span> <span class="fu">head</span><span class="op">(</span><span class="fu">VariableFeatures</span><span class="op">(</span><span class="va">lognorm_st</span><span class="op">)</span>, <span class="fl">15</span><span class="op">)</span></span>
<span><span class="va">plot1</span> <span class="op">&lt;-</span> <span class="fu">VariableFeaturePlot</span><span class="op">(</span><span class="va">lognorm_st</span>, log <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span> </span>
<span>           <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Variable Features - Log normalization"</span><span class="op">)</span></span>
<span><span class="va">plot1</span> <span class="op">&lt;-</span> <span class="fu">LabelPoints</span><span class="op">(</span>plot <span class="op">=</span> <span class="va">plot1</span>, points <span class="op">=</span> <span class="va">top15</span>, repel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>When using repel, set xnudge and ynudge to 0 for optimal results</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb39">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top15SCT</span> <span class="op">&lt;-</span> <span class="fu">head</span><span class="op">(</span><span class="fu">VariableFeatures</span><span class="op">(</span><span class="va">filter_st</span><span class="op">)</span>, <span class="fl">15</span><span class="op">)</span></span>
<span><span class="va">plot2</span>    <span class="op">&lt;-</span> <span class="fu">VariableFeaturePlot</span><span class="op">(</span><span class="va">filter_st</span><span class="op">)</span> <span class="op">+</span> </span>
<span>              <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Variable Features - SCT"</span><span class="op">)</span></span>
<span><span class="va">plot2</span>    <span class="op">&lt;-</span> <span class="fu">LabelPoints</span><span class="op">(</span>plot <span class="op">=</span> <span class="va">plot2</span>, points <span class="op">=</span> <span class="va">top15SCT</span>, repel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>When using repel, set xnudge and ynudge to 0 for optimal results</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb41">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot1</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
</div>
<figure><img src="fig/apply-normalization-methods-rendered-unnamed-chunk-14-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb43">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot2</span></span></code></pre>
</div>
<figure><img src="fig/apply-normalization-methods-rendered-unnamed-chunk-14-2.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure>
</div>
<div class="section level3">
<h3 id="no-one-size-fits-all-approach">No One-Size-Fits-All Approach<a class="anchor" aria-label="anchor" href="#no-one-size-fits-all-approach"></a>
</h3>
<p>Raw read counts provide essential insights into the absolute cell
type densities within a sample, which are crucial for mapping cellular
distribution. In contrast, normalized data adjusts for technical
variations like sequencing depth and RNA capture efficiency, thus
revealing the relative proportions of cell types and identifying
specific tissue structures, such as epithelium or fibrosis. Saiselet et
al. demonstrated that while normalized data effectively identify
distinct morphologies, raw counts are vital for detecting areas with
unusual cell-type concentrations, such as high epithelial regions
expressing vimentin (VIM) (Saiselet, M., et al., Journal of Molecular
Cell Biology, 2020). Hence, choosing the right normalization method
depends on the specific characteristics of each dataset and the
biological questions at hand. Researchers must understand the impact of
each normalization strategy on both the biological and technical aspects
of their data.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>Normalization is essential but must be selectively applied based on
the unique characteristics of each dataset and the specific biological
questions at hand.</li>
<li>Techniques like SCTransform and log scaling offer ways to balance
technical correction with biological integrity.</li>
<li>Examining both raw and normalized data can provide comprehensive
insights into the absolute and relative characteristics of cellular
components in spatial transcriptomics.</li>
</ul>
</div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</div>
</section></section><section id="aio-feature-selection-dimensionality-reduction-clustering"><p>Content from <a href="feature-selection-dimensionality-reduction-clustering.html">Feature Selection, Dimensionality Reduction, and Spot Clustering</a></p>
<hr>
<p>Last updated on 2024-06-13 |
        
        <a href="https://github.com/smcclatchy/spatial-transcriptomics/edit/main/episodes/feature-selection-dimensionality-reduction-clustering.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>Why is feature selection important in spatial transcriptomics?</li>
<li>What are the implications of using different proportions of highly
variable genes (HVGs) in data analysis?</li>
<li>Why is feature selection in spatial transcriptomics not typically
necessary with normalization techniques like SCTransform?</li>
<li>How do PCA and UMAP differ in their approach to dimensionality
reduction in spatial transcriptomics?</li>
<li>What advantages do linear methods like PCA offer before applying
nonlinear methods like UMAP?</li>
<li>How do these dimensionality reduction techniques impact downstream
analysis such as clustering and visualization?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Identify appropriate feature selection methods for different
normalization techniques in spatial transcriptomics.</li>
<li>Evaluate the effects of varying the proportion of highly variable
genes on the resolution of clustering and PCA outcomes.</li>
<li>Understand the rationale behind the dependency of feature selection
on specific normalization methods like NormalizeData.</li>
<li>Differentiate between linear and nonlinear dimensionality reduction
methods and their applications in spatial transcriptomics.</li>
<li>Implement PCA to preprocess data before applying UMAP to enhance
interpretability and structure recognition.</li>
<li>Assess the effectiveness of each method in revealing spatial and
molecular patterns within the data.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section id="understanding-the-morphology-of-your-tissue."><h2 class="section-heading">Understanding the Morphology of your Tissue.<a class="anchor" aria-label="anchor" href="#understanding-the-morphology-of-your-tissue."></a>
</h2>
<hr class="half-width">
<p>In any analysis, it is important that you understand the structure
and cell types in the tissue which you are analyzing. While it would be
ideal to have an unbiased analysis that uses normalizations and
clustering methods to automatically assign cell types and define tissue
structure, in practice we adjust these parameters based on the tissue
structure that we expect to find.</p>
</section><section id="understanding-feature-selection-in-spatial-transcriptomics"><h2 class="section-heading">Understanding Feature Selection in Spatial Transcriptomics<a class="anchor" aria-label="anchor" href="#understanding-feature-selection-in-spatial-transcriptomics"></a>
</h2>
<hr class="half-width">
<p>Feature selection in spatial transcriptomics is essential for
reducing the dimensionality of high-dimensional datasets, enhancing
model performance, and improving interpretability. This process is
crucial because it helps in minimizing computational demands, reducing
noise, and speeding up downstream analyses like clustering and PCA. By
focusing on a subset of genes that show significant variability or are
biologically relevant, researchers can achieve more robust and
generalizable models, draw clearer conclusions, and facilitate
hypothesis testing.</p>
<div class="section level3">
<h3 id="choosing-feature-selection-methods">Choosing Feature Selection Methods<a class="anchor" aria-label="anchor" href="#choosing-feature-selection-methods"></a>
</h3>
<div class="section level4">
<h4 id="importance-of-high-variable-gene-selection">Importance of High Variable Gene Selection<a class="anchor" aria-label="anchor" href="#importance-of-high-variable-gene-selection"></a>
</h4>
<p>Feature selection methods such as variance stabilizing transformation
(VST) and mean-variance plotting are crucial for refining the dataset to
include genes that exhibit meaningful variability across different
spatial regions. These methods help focus on genes that are most
informative for downstream analyses like clustering and dimensionality
reduction.</p>
</div>
</div>
<div class="section level3">
<h3 id="feature-selection-with-normalizedata">Feature Selection with NormalizeData<a class="anchor" aria-label="anchor" href="#feature-selection-with-normalizedata"></a>
</h3>
<p>When using normalization methods like NormalizeData, which focuses on
scaling gene expression data without variance stabilization, applying
feature selection becomes essential. This method requires the selection
of highly variable genes to enhance the analysis, particularly in
clustering and principal component analysis (PCA). Typically, you will
select in the range of 2,000 to 5,000 highly variable genes.</p>
</div>
<div class="section level3">
<h3 id="feature-selection-with-sctransform">Feature Selection with SCTransform<a class="anchor" aria-label="anchor" href="#feature-selection-with-sctransform"></a>
</h3>
<p>SCTransform, a normalization method, adjusts gene expression data to
stabilize the variance, and it also provides default feature selection.
This method ensures that the genes retained are already adjusted for
technical variability, highlighting those with biological
significance.</p>
<p>The SCTransform selects 3,000 variable features by default. We will
use those variable features to calculate principal components (PCs) of
the gene expression data. As always, we must first scale (or
standardize) the normalized counts data.</p>
<div id="challenge-1-number-of-variable-genes" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-number-of-variable-genes" class="callout-inner">
<h3 class="callout-title">Challenge 1: Number of Variable Genes<a class="anchor" aria-label="anchor" href="#challenge-1-number-of-variable-genes"></a>
</h3>
<div class="callout-content">
<p>How do you think that your results would be affected by selecting too
few highly variable genes? What about too many highly variable
genes?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">Show me the solution</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>Too few variable genes may underestimate the variance between spots
and tissue sections and would reduce our ability to discern tissue
structure. Too many variable genes would increase computational time and
might add noise to the analysis.</p>
</div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="dimensionality-reduction-using-principal-components">Dimensionality Reduction using Principal Components<a class="anchor" aria-label="anchor" href="#dimensionality-reduction-using-principal-components"></a>
</h3>
<p>Dimensionality reduction is a crucial step in managing
high-dimensional spatial transcriptomics data, enhancing analytical
clarity, and reducing computational load. Linear methods like PCA and
nonlinear methods like UMAP each play distinct roles in processing and
interpreting complex datasets.</p>
<p>Principal Component Analysis (PCA) is a linear technique that reduces
dimensionality by transforming data into a set of uncorrelated variables
called “principal components” (PCs). This method efficiently captures
the main variance in the data, which is vital for preliminary data
exploration and noise reduction.</p>
<p>In order to cluster the spots by similarity, we use principal
components (PCs) to reduce the number of dimensions in the data. Using a
smaller number of PCs allows us to capture the variability in the data
set while using a smaller number of dimensions. The PCs are then used to
cluster the spots by similarity.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filter_st</span> <span class="op">&lt;-</span> <span class="fu">ScaleData</span><span class="op">(</span><span class="va">filter_st</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>               <span class="fu">RunPCA</span><span class="op">(</span>npcs <span class="op">=</span> <span class="fl">75</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Centering and scaling data matrix</code></pre>
</div>
<p>In the command above, we specified asked Seurat to calculate 75
principal components. There are thousands of genes and we know that we
don’t need to calculate all PCs. By default, the <a href="https://satijalab.org/seurat/reference/runpca" class="external-link">RunPCA</a> method
calculates 50 PCs. Let’s make an elbow plot of the number of PCs versus
the standard deviation explained by each PC. Traditionally, there is a
bend in the curve which indicates that adding more PCs doesn’t account
for more of the variance.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ElbowPlot</span><span class="op">(</span><span class="va">filter_st</span>, ndims <span class="op">=</span> <span class="fl">75</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/feature-selection-dimensionality-reduction-clustering-rendered-unnamed-chunk-3-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>From the plot above, we would normally select somewhere between 10
and 20 PCs because there seems to be little benefit in adding more PCs.
Adding more PCs does not seem to add more explanatory variance.</p>
<p>But in spatial transcriptomics, the elbow plot often does not tell
the whole story. We want to select a number of PCs such that we are able
to discern the structure and cell type composition of our tissue. So it
is better to try a range of number of PCs.</p>
<p>For now, let’s use all 75 PCs.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n_pcs</span> <span class="op">=</span> <span class="fl">75</span></span></code></pre>
</div>
<p>In the next step, we will cluster the spots based on expression
similarity using the principal components that we just generated. We
will use Seurat’s <a href="https://satijalab.org/seurat/reference/findneighbors" class="external-link">FindNeighbors</a>
and <a href="https://satijalab.org/seurat/reference/findclusters" class="external-link">FindClusters</a>
functions. FindNeighbors find the K nearest neighbors of each spot in
the data set. The default values is to use the 20 nearest neighbors in
the data set and we will use this value in this lesson. However, as with
the number of features used to create the PCs, this is another parameter
that is worth varying before proceeding with your analysis.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filter_st</span> <span class="op">&lt;-</span> <span class="fu">FindNeighbors</span><span class="op">(</span><span class="va">filter_st</span>, </span>
<span>                           reduction <span class="op">=</span> <span class="st">"pca"</span>, </span>
<span>                           dims      <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="va">n_pcs</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>               <span class="fu">FindClusters</span><span class="op">(</span>resolution <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Computing nearest neighbor graph</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Computing SNN</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3633
Number of edges: 187716

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7178
Number of communities: 9
Elapsed time: 0 seconds</code></pre>
</div>
<p>The clustering has added a new column to the spot metadata in the
Seurat object called “seurat_clusters”. Let’s look the metadata to see
this.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">filter_st</span><span class="op">[[</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                      orig.ident nCount_Spatial nFeature_Spatial in_tissue
AAACAAGTATCTCCCA-1 SeuratProject           8458             3586         1
AAACAATCTACTAGCA-1 SeuratProject           1667             1150         1
AAACACCAATAACTGC-1 SeuratProject           3769             1960         1
AAACAGAGCGACTCCT-1 SeuratProject           5433             2424         1
AAACAGCTTTCAGAAG-1 SeuratProject           4278             2264         1
AAACAGGGTCTATATT-1 SeuratProject           4004             2178         1
                   array_row array_col pxl_row_in_fullres pxl_col_in_fullres
AAACAAGTATCTCCCA-1        50       102               8468               9791
AAACAATCTACTAGCA-1         3        43               2807               5769
AAACACCAATAACTGC-1        59        19               9505               4068
AAACAGAGCGACTCCT-1        14        94               4151               9271
AAACAGCTTTCAGAAG-1        43         9               7583               3393
AAACAGGGTCTATATT-1        47        13               8064               3665
                   percent.mt keep keep_counts nCount_SCT nFeature_SCT
AAACAAGTATCTCCCA-1   16.63514 TRUE        TRUE       5159         3183
AAACAATCTACTAGCA-1   12.23755 TRUE       FALSE       3400         1264
AAACACCAATAACTGC-1   11.40886 TRUE        TRUE       3828         1941
AAACAGAGCGACTCCT-1   24.22234 TRUE        TRUE       4682         2393
AAACAGCTTTCAGAAG-1   15.21739 TRUE        TRUE       4228         2239
AAACAGGGTCTATATT-1   15.50949 TRUE        TRUE       3991         2148
                   SCT_snn_res.1 seurat_clusters
AAACAAGTATCTCCCA-1             1               1
AAACAATCTACTAGCA-1             4               4
AAACACCAATAACTGC-1             6               6
AAACAGAGCGACTCCT-1             5               5
AAACAGCTTTCAGAAG-1             0               0
AAACAGGGTCTATATT-1             3               3</code></pre>
</div>
<p>Before we proceed, it is worth thinking about the structure of the
tissue. Below, we show the tissue layer structure deduced by experts in
brain morphology and described in <a href="https://www.nature.com/articles/s41593-020-00787-0" alt.text="Maynard et al" class="external-link">Maynard et al</a>. The authors provide these
annotations. Let’s add them to our Seurat object and plot them. We will
use a simple wrapper, SpatialDimPlotColorSafe, around the Seurat
function SpatialDimPlot. This is defined in code/spatial_utils.R and
uses a color-blind safe palette.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spot_metadata</span> <span class="op">&lt;-</span> <span class="fu">read.table</span><span class="op">(</span><span class="st">"./data/spot-meta.tsv"</span>, sep<span class="op">=</span><span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="co"># Subset to our sample</span></span>
<span><span class="va">spot_metadata</span> <span class="op">&lt;-</span> <span class="fu">subset</span><span class="op">(</span><span class="va">spot_metadata</span>, <span class="va">sample_name</span> <span class="op">==</span> <span class="fl">151673</span><span class="op">)</span></span>
<span><span class="fu">rownames</span><span class="op">(</span><span class="va">spot_metadata</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">spot_metadata</span><span class="op">$</span><span class="va">barcode</span></span>
<span><span class="fu">stopifnot</span><span class="op">(</span><span class="fu">all</span><span class="op">(</span><span class="fu">Cells</span><span class="op">(</span><span class="va">filter_st</span><span class="op">)</span> <span class="op">%in%</span> <span class="fu">rownames</span><span class="op">(</span><span class="va">spot_metadata</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">spot_metadata</span> <span class="op">&lt;-</span> <span class="va">spot_metadata</span><span class="op">[</span><span class="fu">Cells</span><span class="op">(</span><span class="va">filter_st</span><span class="op">)</span>,<span class="op">]</span></span>
<span></span>
<span><span class="va">filter_st</span> <span class="op">&lt;-</span> <span class="fu">AddMetaData</span><span class="op">(</span>object <span class="op">=</span> <span class="va">filter_st</span>, metadata <span class="op">=</span> <span class="va">spot_metadata</span><span class="op">[</span>, <span class="fu">c</span><span class="op">(</span><span class="st">"layer_guess"</span><span class="op">)</span>, drop<span class="op">=</span><span class="cn">FALSE</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">SpatialDimPlotColorSafe</span><span class="op">(</span><span class="va">filter_st</span><span class="op">[</span>, <span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">filter_st</span><span class="op">[[</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">layer_guess</span><span class="op">)</span><span class="op">]</span>, <span class="st">"layer_guess"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">labs</span><span class="op">(</span>fill<span class="op">=</span><span class="st">"Layer"</span><span class="op">)</span> </span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning: Not validating Centroids objects
Not validating Centroids objects</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning: Not validating FOV objects
Not validating FOV objects
Not validating FOV objects
Not validating FOV objects
Not validating FOV objects
Not validating FOV objects</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning: Not validating Seurat objects</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Scale for fill is already present.
Adding another scale for fill, which will replace the existing scale.</code></pre>
</div>
<figure><img src="fig/feature-selection-dimensionality-reduction-clustering-rendered-unnamed-chunk-7-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The authors describe six layers arranged from the upper right to the
lower left, and a white matter (WM) later. At this stage of the
analysis, we have nine clusters, but they do not show the clear
separation of the ground truch in the source publication.</p>
<p>We will now plot the spots in the tissue, colored by the clusters
which we have identified in “seurat_clusters” to evaluate the quality of
the cluster identities by looking for the clarity of the stripes forming
each layer.</p>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">num_clusters</span> <span class="op">&lt;-</span> <span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">filter_st</span><span class="op">[[</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">seurat_clusters</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">color_pal</span>    <span class="op">&lt;-</span> <span class="fu">setNames</span><span class="op">(</span><span class="fu">carto_pal</span><span class="op">(</span><span class="va">num_clusters</span>, <span class="st">"Safe"</span><span class="op">)</span>, <span class="fl">0</span><span class="op">:</span><span class="op">(</span><span class="va">num_clusters</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">SpatialDimPlot</span><span class="op">(</span><span class="va">filter_st</span>, </span>
<span>               group.by <span class="op">=</span> <span class="st">"seurat_clusters"</span>, </span>
<span>               cols     <span class="op">=</span> <span class="va">color_pal</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/feature-selection-dimensionality-reduction-clustering-rendered-unnamed-chunk-8-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>How many layers do we have compared to the publication? What do you
think about the quality of the layers in this plot? Are there clear
layers in the tissue?</p>
<p>Another way for us to look at the clusters it to plot them in a
Uniform Manifold Approximation and Projection (UMAP). UMAP is a
non-linear dimension reduction technique that can be used to visualize
data. This is often used in single-cell RNASeq to identify different
cell types. Here, each spot is potentially composed of more than one
cell type, so the clustering may not be as clear.</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filter_st</span> <span class="op">&lt;-</span> <span class="fu">RunUMAP</span><span class="op">(</span><span class="va">filter_st</span>, </span>
<span>                     reduction <span class="op">=</span> <span class="st">'pca'</span>, </span>
<span>                     dims      <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="va">n_pcs</span>, </span>
<span>                     verbose   <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">filter_st</span>, </span>
<span>         label      <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>         cols       <span class="op">=</span> <span class="va">color_pal</span>,</span>
<span>         pt.size    <span class="op">=</span> <span class="fl">2</span>,</span>
<span>         label.size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/feature-selection-dimensionality-reduction-clustering-rendered-unnamed-chunk-9-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>We have made some decisions above which might affect the quality of
out spot clusters, including the number of nearest neighbors, the number
of variable features, the number of PCs, and the cluster resolution. As
we have mentioned, it is critical to have some understanding of the
structure of the tissue that you are analyzing. In the absence of
“ground truth”, you will need to try several different parameters and
look at how the clusters group in your tissue. Below, we have included
code which plots the tissue with different numbers of nearest neighbors,
principal components, and cluster resolutions. We will not run this code
because it takes a long time to run, but we show the output below this
code block.</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Set several cluster resolution values.</span></span>
<span><span class="va">resol</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co"># Set several numbers of principal components.</span></span>
<span><span class="va">npcs</span>  <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="fl">25</span>, <span class="fl">50</span>, <span class="fl">75</span><span class="op">)</span></span>
<span></span>
<span><span class="va">umap_plots</span> <span class="op">&lt;-</span> <span class="fu">vector</span><span class="op">(</span><span class="st">'list'</span>, <span class="fu">length</span><span class="op">(</span><span class="va">resol</span><span class="op">)</span> <span class="op">*</span> <span class="fu">length</span><span class="op">(</span><span class="va">npcs</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">plots</span>      <span class="op">&lt;-</span> <span class="fu">vector</span><span class="op">(</span><span class="st">'list'</span>, <span class="fu">length</span><span class="op">(</span><span class="va">resol</span><span class="op">)</span> <span class="op">*</span> <span class="fu">length</span><span class="op">(</span><span class="va">npcs</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu">seq_along</span><span class="op">(</span><span class="va">resol</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw">for</span><span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="fu">seq_along</span><span class="op">(</span><span class="va">npcs</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span></span>
<span>    <span class="va">index</span>     <span class="op">&lt;-</span> <span class="op">(</span><span class="va">i</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span> <span class="op">*</span> <span class="fu">length</span><span class="op">(</span><span class="va">npcs</span><span class="op">)</span> <span class="op">+</span> <span class="va">j</span></span>
<span>    <span class="fu">print</span><span class="op">(</span><span class="fu">paste</span><span class="op">(</span><span class="st">'Index ='</span>, <span class="va">index</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span>    <span class="va">filter_st</span> <span class="op">&lt;-</span> <span class="fu">FindNeighbors</span><span class="op">(</span><span class="va">filter_st</span>, </span>
<span>                               reduction <span class="op">=</span> <span class="st">"pca"</span>, </span>
<span>                               dims      <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="va">npcs</span><span class="op">[</span><span class="va">j</span><span class="op">]</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                   <span class="fu">FindClusters</span><span class="op">(</span>resolution <span class="op">=</span> <span class="va">resol</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                   <span class="fu">RunUMAP</span><span class="op">(</span>reduction <span class="op">=</span> <span class="st">'pca'</span>, </span>
<span>                           dims      <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="va">npcs</span><span class="op">[</span><span class="va">j</span><span class="op">]</span>,</span>
<span>                           verbose   <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span>    <span class="va">umap_plots</span><span class="op">[[</span><span class="va">index</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">filter_st</span>, label <span class="op">=</span> <span class="cn">TRUE</span>, cols <span class="op">=</span> <span class="va">color_pal</span>, </span>
<span>                                    pt.size <span class="op">=</span> <span class="fl">2</span>, label.size <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="op">+</span></span>
<span>                             <span class="fu">ggtitle</span><span class="op">(</span><span class="fu">paste</span><span class="op">(</span><span class="st">"res ="</span>, <span class="va">resol</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, <span class="st">": pc ="</span>, <span class="va">npcs</span><span class="op">[</span><span class="va">j</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>                             <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span></span>
<span>    <span class="va">plots</span><span class="op">[[</span><span class="va">index</span><span class="op">]</span><span class="op">]</span>      <span class="op">&lt;-</span> <span class="fu">SpatialDimPlot</span><span class="op">(</span><span class="va">filter_st</span>, group.by <span class="op">=</span> <span class="st">"seurat_clusters"</span>, cols <span class="op">=</span> <span class="va">color_pal</span><span class="op">)</span> <span class="op">+</span></span>
<span>                             <span class="fu">ggtitle</span><span class="op">(</span><span class="fu">paste</span><span class="op">(</span><span class="st">"res ="</span>, <span class="va">resol</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, <span class="st">": pc ="</span>, <span class="va">npcs</span><span class="op">[</span><span class="va">j</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>                             <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span></span>
<span>  <span class="op">}</span> <span class="co"># for(j)</span></span>
<span></span>
<span><span class="op">}</span> <span class="co"># for(i)</span></span>
<span></span>
<span><span class="fu">png</span><span class="op">(</span><span class="fu">file.path</span><span class="op">(</span><span class="st">'episodes'</span>, <span class="st">'fig'</span>, <span class="st">'tissue_cluster_resol.png'</span><span class="op">)</span>,</span>
<span>    width <span class="op">=</span> <span class="fl">1000</span>, height <span class="op">=</span> <span class="fl">1000</span>, res <span class="op">=</span> <span class="fl">128</span><span class="op">)</span></span>
<span><span class="fu">print</span><span class="op">(</span><span class="fu">gridExtra</span><span class="fu">::</span><span class="fu">grid.arrange</span><span class="op">(</span>grobs <span class="op">=</span> <span class="va">plots</span>, nrow <span class="op">=</span> <span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">dev.off</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">png</span><span class="op">(</span><span class="fu">file.path</span><span class="op">(</span><span class="st">'episodes'</span>, <span class="st">'fig'</span>, <span class="st">'umap_cluster_resol.png'</span><span class="op">)</span>,</span>
<span>    width <span class="op">=</span> <span class="fl">1000</span>, height <span class="op">=</span> <span class="fl">1000</span>, res <span class="op">=</span> <span class="fl">128</span><span class="op">)</span></span>
<span><span class="fu">print</span><span class="op">(</span><span class="fu">gridExtra</span><span class="fu">::</span><span class="fu">grid.arrange</span><span class="op">(</span>grobs <span class="op">=</span> <span class="va">umap_plots</span>, nrow <span class="op">=</span> <span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">dev.off</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>The plot below shows the clusters as colors overlayed on the tissue.
Each row shows a different cluster resolutions and each column snows the
number of PCs.</p>
<figure><img src="fig/tissue_cluster_resol.png" alt="Tissue Clustering showing the tissue layers at different resolutions and number of PCs" class="figure mx-auto d-block"><div class="figcaption">Tissue Clustering with Different Numbers of PCs
and Clustering Resolutions</div>
</figure><p>The plot below shows the UMAP clustering for the same set of
parameters in the same order. The cluster colors are the same in the
plots above and below. Increasing the cluster resolution increases the
number of clusters.</p>
<figure><img src="fig/umap_cluster_resol.png" alt="UMAP Clustering with Different Numbers of PCs and Clustering Resolutions" class="figure mx-auto d-block"><div class="figcaption">UMAP Clustering with Different Numbers of PCs
and Clustering Resolutions</div>
</figure><div id="challenge-1-select-cluster-resolution-and-number-of-pcs." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-select-cluster-resolution-and-number-of-pcs." class="callout-inner">
<h3 class="callout-title">Challenge 1: Select cluster resolution and
number of PCs.<a class="anchor" aria-label="anchor" href="#challenge-1-select-cluster-resolution-and-number-of-pcs."></a>
</h3>
<div class="callout-content">
<p>Look at the two plots above which show the tissue and UMAP clusters
at different cluster resolutions and number of PCs. Think about which
settings seem to produce clustering that matches the expectations from
the publication. Turn to the person next to you and discuss your
opinions about which settings to use.</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2">Solution 1</h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>All of the plots in the tissue clustering show layers which broadly
conform to the publication’s “ground truth.” In the top row at a
resolution of 0.5, we see about four layers with varying levels of
clarity, depending on the number of PCs. As we move down in the tissue
clustering plot, the layers become clearer at a resolution of 1, but
then become less clear at a resolution of 2 in the bottom row. When we
look at the corresponding row in the UMAP plots, we see that the white
matter is clearly separated in all of the plots. The number of clusters
increases with increasing cluster resolution, with the middle row
(cluster resolution = 1), having close to seven layers, like the
publication ground truth. The middle plot in the middle row has nine
clusters, which is also close to the publication.</p>
</div>
</div>
</div>
</div>
<p>We selected a cluster resolution of 0.8 and 50 PCs for the following
work.</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filter_st</span> <span class="op">&lt;-</span> <span class="fu">FindNeighbors</span><span class="op">(</span><span class="va">filter_st</span>, </span>
<span>                           reduction <span class="op">=</span> <span class="st">"pca"</span>, </span>
<span>                           dims      <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>               <span class="fu">FindClusters</span><span class="op">(</span>resolution <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3633
Number of edges: 171945

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7408
Number of communities: 9
Elapsed time: 0 seconds</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">SpatialDimPlot</span><span class="op">(</span><span class="va">filter_st</span>, group.by <span class="op">=</span> <span class="st">"seurat_clusters"</span>, cols <span class="op">=</span> <span class="va">color_pal</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span>label <span class="op">=</span> <span class="st">"Tissue Clusters: 50 PCs, resol = 0.8"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/feature-selection-dimensionality-reduction-clustering-rendered-unnamed-chunk-12-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>Feature selection is a crucial step in spatial transcriptomics
analysis, particularly for non-variance-stabilizing normalization
methods like NormalizeData.</li>
<li>Techniques such as VST and mean-variance plotting enable researchers
to focus on genes that provide the most biological insight.</li>
<li>Different proportions of highly variable genes and feature selection
methods can significantly influence the analytical outcomes, emphasizing
the need for tailored approaches based on the specific characteristics
of each dataset.</li>
<li>Linear dimensionality reduction methods like PCA are crucial for
initial data simplification and noise reduction.</li>
<li>Nonlinear methods like UMAP are valuable for detailed exploration of
data structures post-linear preprocessing.</li>
<li>The sequential application of PCA and UMAP can provide a
comprehensive view of the spatial transcriptomics data, leveraging the
strengths of both linear and nonlinear approaches.</li>
</ul>
</div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</div>
</section></section><section id="aio-deconvolve-cell-types-in-a-spot"><p>Content from <a href="deconvolve-cell-types-in-a-spot.html">Deconvolution in Spatial Transcriptomics</a></p>
<hr>
<p>Last updated on 2024-06-13 |
        
        <a href="https://github.com/smcclatchy/spatial-transcriptomics/edit/main/episodes/deconvolve-cell-types-in-a-spot.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How does deconvolution enhance the analysis of spatial
transcriptomics data?</li>
<li>What are the steps to integrate single-cell RNA-seq data for
deconvolution in spatial transcriptomics?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Perform deconvolution to quantify different cell types in spatial
transcriptomics spots using single-cell RNA-seq data.</li>
<li>Understand the process of integrating single-cell RNA-seq data with
spatial transcriptomics data.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section id="deconvolution-in-spatial-transcriptomics"><h2 class="section-heading">Deconvolution in Spatial Transcriptomics<a class="anchor" aria-label="anchor" href="#deconvolution-in-spatial-transcriptomics"></a>
</h2>
<hr class="half-width">
<p>Each spatial spot in an ST experiment generally contains multiple
cells. For example, spots in the Visium assay are 55 microns in
diameter, whereas a typical T cells is ~10 microns. As such, the
expression read out from the spot mixes together the expression of the
individual cells encompassed by it. Deconvolution is the approach for
unmixing this combined expression signal. Most often, deconvolution
methods predict the fraction of each spot’s expression derived from each
particular cell type. Supervised methods deconvolve spot expression
using cell type expression profiles (e.g., from scRNA-seq) or marker
genes. Unsupervised approaches instead infer the expression of the cell
types first.</p>
<p><img src="https://ars.els-cdn.com/content/image/1-s2.0-S200103702200558X-ga1_lrg.jpg" alt="alt text for accessibility purposes" class="figure"><a href="https://www.sciencedirect.com/science/article/pii/S200103702200558X#f0015" class="external-link">Zhang
et al, Comput Struct Biotechnol J 21, 176–184 (2023)</a>
<a href="https://creativecommons.org/licenses/by-nc-nd/4.0/" rel="license" class="external-link">CC
BY-NC-ND 4.0</a> ## Deconvolution with RCTD</p>
<p>RCTD (Robust Cell Type Decomposition) We will apply RCTD (Robust Cell
Type Decomposition) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8606190" class="external-link">Cable, D.
M., et al., Nature Biotechnology, 2021</a>. The algorithm uses
single-cell RNA sequencing (scRNA-seq) data as a reference to deconvolve
the spatial transcriptomics data, estimating the proportions of
different cell types in each spatial spot. The RCTD algorithm models the
observed gene expression in each spatial spot as a mixture of the gene
expression profiles of different cell types and estimates the proportion
of each cell type in each spot using a non-negative least squares (NNLS)
approach. RCTD can operate in two modes: in “doublet” mode it fits at
most two cell types per spot, in “full” mode it fits potentially all
cell types in the reference per spot, and in “multi” mode it again fits
more than two cell types per spot by extending the “doublet” approach.
Here, we will use “full” mode.</p>
</section><section id="loading-single-cell-rna-seq-data"><h2 class="section-heading">Loading Single Cell RNA-Seq Data<a class="anchor" aria-label="anchor" href="#loading-single-cell-rna-seq-data"></a>
</h2>
<hr class="half-width">
<p>First, we load the single-cell RNA-seq data that will serve as a
reference for deconvolution. This data is essential for mapping the gene
expression profiles from the ST data to specific cell types.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load single-cell RNA-seq data</span></span>
<span><span class="va">sc.counts</span> <span class="op">&lt;-</span> <span class="fu">fread</span><span class="op">(</span><span class="st">"data/scRNA-seq/sc_counts.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">sc.counts</span> <span class="op">&lt;-</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="va">sc.counts</span><span class="op">)</span></span>
<span><span class="fu">rownames</span><span class="op">(</span><span class="va">sc.counts</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sc.counts</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">sc.counts</span>           <span class="op">&lt;-</span> <span class="fu">as.matrix</span><span class="op">(</span><span class="va">sc.counts</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load cell type annotations</span></span>
<span><span class="va">sc.metadata</span>   <span class="op">&lt;-</span> <span class="fu">read.delim</span><span class="op">(</span><span class="st">"data/scRNA-seq/sc_cell_types.tsv"</span><span class="op">)</span></span>
<span><span class="va">sc.cell.types</span> <span class="op">&lt;-</span> <span class="fu">setNames</span><span class="op">(</span><span class="fu">factor</span><span class="op">(</span><span class="va">sc.metadata</span><span class="op">$</span><span class="va">Value</span><span class="op">)</span>, <span class="va">sc.metadata</span><span class="op">$</span><span class="va">Name</span><span class="op">)</span></span>
<span><span class="va">shared.cells</span>  <span class="op">&lt;-</span> <span class="fu">intersect</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">sc.counts</span><span class="op">)</span>, <span class="fu">names</span><span class="op">(</span><span class="va">sc.cell.types</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sc.cell.types</span> <span class="op">&lt;-</span> <span class="va">sc.cell.types</span><span class="op">[</span><span class="va">shared.cells</span><span class="op">]</span></span>
<span><span class="va">sc.counts</span>     <span class="op">&lt;-</span> <span class="va">sc.counts</span><span class="op">[</span>, <span class="va">shared.cells</span><span class="op">]</span></span></code></pre>
</div>
</section><section id="deconvolution-with-rctd"><h2 class="section-heading">Deconvolution with RCTD<a class="anchor" aria-label="anchor" href="#deconvolution-with-rctd"></a>
</h2>
<hr class="half-width">
<p>First, we will create the reference object encapsulating the
scRNA-seq data. We will use the <a href="https://rdrr.io/github/dmcable/RCTD/man/Reference.html" class="external-link">Reference</a>
function from the <a href="https://github.com/dmcable/spacexr" class="external-link">spacexr</a> package, which
will organize and store the single cell RNASeq data for the next
steps.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc_reference</span> <span class="op">&lt;-</span> <span class="fu">Reference</span><span class="op">(</span><span class="va">sc.counts</span>, <span class="va">sc.cell.types</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s write a wrapper function that performs RCTD deconvolution. This
will facilitate running RCTD on other samples within this dataset.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">run.rctd</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">reference</span>, <span class="va">st.obj</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="co"># Get raw ST counts</span></span>
<span>  <span class="va">st.counts</span> <span class="op">&lt;-</span> <span class="fu">GetAssayData</span><span class="op">(</span><span class="va">st.obj</span>, assay <span class="op">=</span> <span class="st">"Spatial"</span>, layer <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Get the spot coordinates</span></span>
<span>  <span class="va">st.coords</span>           <span class="op">&lt;-</span> <span class="va">st.obj</span><span class="op">[[</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>, <span class="fu">c</span><span class="op">(</span><span class="st">"array_col"</span>, <span class="st">"array_row"</span><span class="op">)</span><span class="op">]</span></span>
<span>  <span class="fu">colnames</span><span class="op">(</span><span class="va">st.coords</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">"x"</span>,<span class="st">"y"</span><span class="op">)</span></span>
<span>    </span>
<span>  <span class="co"># Create the RCTD 'puck', representing the ST data</span></span>
<span>  <span class="va">puck</span>   <span class="op">&lt;-</span> <span class="fu">SpatialRNA</span><span class="op">(</span><span class="va">st.coords</span>, <span class="va">st.counts</span><span class="op">)</span></span>
<span></span>
<span>  <span class="va">myRCTD</span> <span class="op">&lt;-</span> <span class="fu">create.RCTD</span><span class="op">(</span><span class="va">puck</span>, <span class="va">reference</span>, max_cores <span class="op">=</span> <span class="fl">1</span>, keep_reference <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span>  <span class="co"># Run deconvolution -- note that we are using 'full' mode to devolve a spot into </span></span>
<span>  <span class="co"># (potentially) all available cell types.</span></span>
<span>  <span class="va">myRCTD</span> <span class="op">&lt;-</span> <span class="fu">suppressWarnings</span><span class="op">(</span><span class="fu">run.RCTD</span><span class="op">(</span><span class="va">myRCTD</span>, doublet_mode <span class="op">=</span> <span class="st">'full'</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">myRCTD</span></span>
<span><span class="op">}</span></span></code></pre>
</div>
</section><section id="running-deconvolution-on-brain-samples"><h2 class="section-heading">Running Deconvolution on Brain Samples<a class="anchor" aria-label="anchor" href="#running-deconvolution-on-brain-samples"></a>
</h2>
<hr class="half-width">
<p>We apply the RCTD wrapper to our spatial transcriptomics data to
deconvolute the spots and quantify the cell types. This may take ~10
minutes. If you prefer, you can load the precomputed results
directly.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Change this variable to TRUE to load precomputed results, or FALSE to compute</span></span>
<span><span class="co"># the results here.</span></span>
<span><span class="va">load.precomputed.results</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span></span>
<span><span class="va">rds.file</span> <span class="op">&lt;-</span> <span class="fu">paste0</span><span class="op">(</span><span class="st">"data/rctd-sample-1.rds"</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="va">load.precomputed.results</span> <span class="op">||</span> <span class="op">!</span><span class="fu">file.exists</span><span class="op">(</span><span class="va">rds.file</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span></span>
<span>  <span class="va">result_1</span> <span class="op">&lt;-</span> <span class="fu">run.rctd</span><span class="op">(</span><span class="va">sc_reference</span>, <span class="va">filter_st</span><span class="op">)</span></span>
<span>  <span class="co"># The RCTD file is large. To save space, we will remove the reference counts.</span></span>
<span>  <span class="va">result_1</span> <span class="op">&lt;-</span> <span class="fu">remove.RCTD.reference.counts</span><span class="op">(</span><span class="va">result_1</span><span class="op">)</span></span>
<span>  <span class="fu">saveRDS</span><span class="op">(</span><span class="va">result_1</span>, <span class="va">rds.file</span><span class="op">)</span></span>
<span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span></span>
<span>  <span class="va">result_1</span> <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span><span class="va">rds.file</span><span class="op">)</span></span>
<span></span>
<span><span class="op">}</span></span></code></pre>
</div>
</section><section id="interpreting-deconvolution-results"><h2 class="section-heading">Interpreting Deconvolution Results<a class="anchor" aria-label="anchor" href="#interpreting-deconvolution-results"></a>
</h2>
<hr class="half-width">
<p>The deconvolution process outputs the proportion of different cell
types in each spatial spot. Let’s write a utility function to extract
these proportions from the RCTD output. This function is also defined in
code/spatial_utils.R.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">format.rctd.output_</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">rctd</span>, <span class="va">normalize</span> <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span>
<span></span>
<span>  <span class="va">barcodes</span> <span class="op">&lt;-</span> <span class="fu">colnames</span><span class="op">(</span><span class="va">rctd</span><span class="op">@</span><span class="va">spatialRNA</span><span class="op">@</span><span class="va">counts</span><span class="op">)</span></span>
<span>  <span class="va">weights</span>  <span class="op">&lt;-</span> <span class="va">rctd</span><span class="op">@</span><span class="va">results</span><span class="op">$</span><span class="va">weights</span></span>
<span></span>
<span>  <span class="kw">if</span><span class="op">(</span><span class="va">normalize</span><span class="op">)</span> <span class="op">{</span></span>
<span></span>
<span>    <span class="va">weights</span> <span class="op">&lt;-</span> <span class="fu">normalize_weights</span><span class="op">(</span><span class="va">weights</span><span class="op">)</span></span>
<span></span>
<span>  <span class="op">}</span></span>
<span></span>
<span>  <span class="va">df</span>   <span class="op">&lt;-</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="va">weights</span><span class="op">)</span></span>
<span>  <span class="va">df</span><span class="op">$</span><span class="va">x</span> <span class="op">&lt;-</span> <span class="va">rctd</span><span class="op">@</span><span class="va">spatialRNA</span><span class="op">@</span><span class="va">coords</span><span class="op">$</span><span class="va">x</span></span>
<span>  <span class="va">df</span><span class="op">$</span><span class="va">y</span> <span class="op">&lt;-</span> <span class="va">rctd</span><span class="op">@</span><span class="va">spatialRNA</span><span class="op">@</span><span class="va">coords</span><span class="op">$</span><span class="va">y</span></span>
<span>  <span class="va">df</span></span>
<span><span class="op">}</span></span></code></pre>
</div>
<p>And now let’s see the predicted proportions in our sample:</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">props</span> <span class="op">&lt;-</span> <span class="fu">format.rctd.output_</span><span class="op">(</span><span class="va">result_1</span>, normalize <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">props</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                      AST-FB         L2-3           L4         L5-6
AAACAAGTATCTCCCA-1 0.2906892 1.989539e-01 3.568361e-01 9.495123e-04
AAACAATCTACTAGCA-1 0.3665241 4.652610e-04 4.652610e-04 4.601510e-01
AAACACCAATAACTGC-1 0.1069532 5.473749e-05 5.473749e-05 5.473749e-05
AAACAGAGCGACTCCT-1 0.3845201 4.758347e-01 3.256827e-04 3.256827e-04
AAACAGCTTTCAGAAG-1 0.2411203 3.694851e-01 1.995963e-01 6.646586e-04
AAACAGGGTCTATATT-1 0.2096408 3.256827e-04 3.624793e-04 5.117059e-01
                   Oligodendrocytes   x  y
AAACAAGTATCTCCCA-1       0.04291383 102 50
AAACAATCTACTAGCA-1       0.12784650  43  3
AAACACCAATAACTGC-1       1.25940541  19 59
AAACAGAGCGACTCCT-1       0.06181266  94 14
AAACAGCTTTCAGAAG-1       0.23374357   9 43
AAACAGGGTCTATATT-1       0.46602478  13 47</code></pre>
</div>
<p>Notice that the proportions don’t sum exactly to one.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="fu">rowSums</span><span class="op">(</span><span class="fu">select</span><span class="op">(</span><span class="va">props</span>, <span class="op">-</span><span class="fu">c</span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>AAACAAGTATCTCCCA-1 AAACAATCTACTAGCA-1 AAACACCAATAACTGC-1 AAACAGAGCGACTCCT-1 
         0.8903426          0.9554521          1.3665228          0.9228188 
AAACAGCTTTCAGAAG-1 AAACAGGGTCTATATT-1 
         1.0446099          1.1880596 </code></pre>
</div>
<p>Let’s classify the spot according to the layer type with highest
proportion</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">props</span><span class="op">$</span><span class="va">classification</span> <span class="op">&lt;-</span> <span class="fu">apply</span><span class="op">(</span><span class="fu">select</span><span class="op">(</span><span class="va">props</span>, <span class="op">-</span><span class="fu">c</span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span><span class="op">)</span><span class="op">)</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">row</span><span class="op">)</span> <span class="fu">names</span><span class="op">(</span><span class="va">row</span><span class="op">)</span><span class="op">[</span><span class="fu">which.max</span><span class="op">(</span><span class="va">row</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s add the deconvolution results to our Seurat object.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filter_st</span> <span class="op">&lt;-</span> <span class="fu">AddMetaData</span><span class="op">(</span>object <span class="op">=</span> <span class="va">filter_st</span>, metadata <span class="op">=</span>  <span class="fu">select</span><span class="op">(</span><span class="va">props</span>, <span class="op">-</span><span class="fu">c</span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>We can now visualize the predicted layer classifications and compare
them alongside the ground truth annotations that we saw previously.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">SpatialDimPlotColorSafe</span><span class="op">(</span><span class="va">filter_st</span><span class="op">[</span>, <span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">filter_st</span><span class="op">[[</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">classification</span><span class="op">)</span><span class="op">]</span>, <span class="st">"classification"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/deconvolve-cell-types-in-a-spot-rendered-unnamed-chunk-11-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">SpatialDimPlotColorSafe</span><span class="op">(</span><span class="va">filter_st</span><span class="op">[</span>, <span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">filter_st</span><span class="op">[[</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">layer_guess</span><span class="op">)</span><span class="op">]</span>, <span class="st">"layer_guess"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/deconvolve-cell-types-in-a-spot-rendered-unnamed-chunk-12-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>To be more quantitative, we can compute a confusion matrix comparing
the predicted and observed layers.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span>            <span class="op">&lt;-</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="fu">table</span><span class="op">(</span><span class="va">filter_st</span><span class="op">[[</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">layer_guess</span>, <span class="va">filter_st</span><span class="op">[[</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">colnames</span><span class="op">(</span><span class="va">df</span><span class="op">)</span>  <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Annotation"</span>, <span class="st">"Prediction"</span>, <span class="st">"Freq"</span><span class="op">)</span></span>
<span><span class="va">df</span><span class="op">$</span><span class="va">Annotation</span> <span class="op">&lt;-</span> <span class="fu">factor</span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">Annotation</span><span class="op">)</span></span>
<span><span class="va">df</span><span class="op">$</span><span class="va">Prediction</span> <span class="op">&lt;-</span> <span class="fu">factor</span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">Prediction</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">df</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Annotation</span>, y <span class="op">=</span> <span class="va">Prediction</span>, fill <span class="op">=</span> <span class="va">Freq</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/deconvolve-cell-types-in-a-spot-rendered-unnamed-chunk-13-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Note that there is a fairly strong correlation between the predicted
and observed layers, particularly for the pairs Oligodendrocytes and WM
(White Matter), L4 and Layer 4, and L2-3 and Layer 3.</p>
</section><section id="summary"><h2 class="section-heading">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<hr class="half-width">
<p>Incorporating deconvolution into the spatial transcriptomics workflow
enhances the ability to quantify the different cell types in each
spatial spot, providing a richer and more detailed understanding of the
tissue architecture. By following these steps, researchers can ensure
that their spatial transcriptomics data is accurately deconvoluted,
leading to more robust and insightful analyses.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>Deconvolution enhances spatial transcriptomics by quantifying the
different cell types within spatial spots.</li>
<li>Integrating single-cell RNA-seq data with spatial transcriptomics
data is essential for accurate deconvolution.</li>
<li>The RCTD method is effective for quantifying the proportion of
different cell types in spatial transcriptomics data.</li>
</ul>
</div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</section></section><section id="aio-differential-expression-testing"><p>Content from <a href="differential-expression-testing.html">Differential Expression Testing</a></p>
<hr>
<p>Last updated on 2024-06-13 |
        
        <a href="https://github.com/smcclatchy/spatial-transcriptomics/edit/main/episodes/differential-expression-testing.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is the purpose of differential expression testing in
bioinformatics?</li>
<li>Can Moran’s I algorithm independently identify region-specific
differential expressions that align with results obtained from expert
annotations?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Identify differentially expressed genes across different layers
using expert annotations.</li>
<li>Utilize Moran’s I algorithm to find spatially variable genes.</li>
<li>Explore the correlation between genes identified through expert
annotations and those detected by Moran’s I algorithm.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section id="introduction-to-differential-expression-testing"><h2 class="section-heading">Introduction to Differential Expression Testing<a class="anchor" aria-label="anchor" href="#introduction-to-differential-expression-testing"></a>
</h2>
<hr class="half-width">
<p>Differential expression testing is crucial in bioinformatics for
identifying genes that show significant differences in expression across
different samples or groups. This method helps find genes that are
upregulated or downregulated in specific contexts, providing insights
into biological functions and disease mechanisms.</p>
</section><section id="differential-expression-analysis"><h2 class="section-heading">Differential Expression Analysis<a class="anchor" aria-label="anchor" href="#differential-expression-analysis"></a>
</h2>
<hr class="half-width">
<div class="section level3">
<h3 id="differential-expression-using-experts-annotation">Differential Expression Using Expert’s Annotation<a class="anchor" aria-label="anchor" href="#differential-expression-using-experts-annotation"></a>
</h3>
<p>We will begin by performing differential expression across the
annotated layers. As a reminder, those look like:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">SpatialDimPlotColorSafe</span><span class="op">(</span><span class="va">filter_st</span><span class="op">[</span>, <span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">filter_st</span><span class="op">[[</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">layer_guess</span><span class="op">)</span><span class="op">]</span>, <span class="st">"layer_guess"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">labs</span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"Layer"</span><span class="op">)</span> </span></code></pre>
</div>
<figure><img src="fig/differential-expression-testing-rendered-layers-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>We identify genes that are upregulated in each annotated brain region
using the <code>FindAllMarkers</code> function from the Seurat toolkit,
utilized with the Presto library, by comparing their expression levels
against all other regions. Presto significantly enhances the
computational speed of this analysis, delivering quicker results. For
more details on the function and its parameters, see the <a href="https://satijalab.org/seurat/reference/findallmarkers" class="external-link">Seurat
FindAllMarkers documentation</a>.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">Idents</span><span class="op">(</span><span class="va">filter_st</span><span class="op">)</span>  <span class="op">&lt;-</span> <span class="st">"layer_guess"</span></span>
<span><span class="va">de_genes</span>           <span class="op">&lt;-</span> <span class="fu">FindAllMarkers</span><span class="op">(</span><span class="va">filter_st</span>, </span>
<span>                                     assay    <span class="op">=</span> <span class="st">"SCT"</span>,</span>
<span>                                     verbose  <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                     only.pos <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                     min.pct  <span class="op">=</span> <span class="fl">0.25</span>, </span>
<span>                                     logfc.threshold <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span></span></code></pre>
</div>
</div>
</section><section id="morans-i-statistic"><h2 class="section-heading">Moran’s I Statistic<a class="anchor" aria-label="anchor" href="#morans-i-statistic"></a>
</h2>
<hr class="half-width">
<p>Moran’s I is a measure used to assess spatial autocorrelation in
data, indicating whether similar values are clustered, dispersed, or
random. In bioinformatics, it’s applied to detect genes whose expression
patterns exhibit clear spatial structure, aiding in understanding
spatially localized biological processes.</p>
<p><img src="https://en.wikipedia.org/wiki/File:Moran%27s_I_example.png" alt="Moran’s I statistic visualized across four different spatial patterns." class="figure"><strong>Top Left:</strong> Checkerboard pattern results in negative
Moran’s I, indicating anti-correlation. <strong>Top Right:</strong>
Linear gradient shows a high positive Moran’s I, reflecting a strong
spatial gradient. <strong>Bottom Left:</strong> Random pattern leads to
a Moran’s I near zero, suggesting no significant spatial
autocorrelation. <strong>Bottom Right:</strong> ‘Ink blot’ pattern
demonstrates positive autocorrelation, indicative of a clustered or
spreading pattern. Relationships are calculated using direct, equally
weighted neighbors, normalized for each cell. Image by
<a href="https://commons.wikimedia.org/wiki/File:Moran%27s_I_example.png" class="external-link">WikiNukalito</a>,
<a href="https://creativecommons.org/licenses/by-sa/4.0" class="external-link">CC BY-SA
4.0</a>, via Wikimedia Commons.</p>
<div class="section level3">
<h3 id="spatial-differential-expression-using-morans-i">Spatial Differential Expression Using Moran’s I<a class="anchor" aria-label="anchor" href="#spatial-differential-expression-using-morans-i"></a>
</h3>
<p>We identify the genes whose expression patterns exhibit clear spatial
structure using Moran’s I algorithm.</p>
<blockquote>
<p>DMG: Can one of you elaborate on what Moran’s I is? And possibly add
a reference?</p>
</blockquote>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">svg</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu">FindSpatiallyVariableFeatures</span><span class="op">(</span><span class="va">filter_st</span>, </span>
<span>                                assay            <span class="op">=</span> <span class="st">"SCT"</span>, </span>
<span>                                features         <span class="op">=</span> <span class="fu">VariableFeatures</span><span class="op">(</span><span class="va">filter_st</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">1000</span><span class="op">]</span>, </span>
<span>                                selection.method <span class="op">=</span> <span class="st">"moransi"</span><span class="op">)</span></span></code></pre>
</div>
</div>
</section><section id="correlation-of-differentially-expressed-genes-in-each-brain-region-and-genes-with-highset-morans-i-value."><h2 class="section-heading">Correlation of Differentially Expressed Genes in each Brain Region
and genes with highset Moran’s I value.<a class="anchor" aria-label="anchor" href="#correlation-of-differentially-expressed-genes-in-each-brain-region-and-genes-with-highset-morans-i-value."></a>
</h2>
<hr class="half-width">
<div class="section level3">
<h3 id="heatmap-of-differential-expression">Heatmap of Differential Expression<a class="anchor" aria-label="anchor" href="#heatmap-of-differential-expression"></a>
</h3>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get and sort 'MoransI_observed' values</span></span>
<span><span class="va">morans_i_genes</span> <span class="op">&lt;-</span> <span class="va">svg</span><span class="op">@</span><span class="va">assays</span><span class="op">[[</span><span class="st">"SCT"</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">meta.features</span> <span class="op">%&gt;%</span></span>
<span>                    <span class="fu">rownames_to_column</span><span class="op">(</span><span class="st">"gene"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                    <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">MoransI_observed</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                    <span class="fu">slice_head</span><span class="op">(</span>n <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Merge the Moran's I values with the DE genes</span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu">merge</span><span class="op">(</span><span class="va">morans_i_genes</span>, <span class="va">de_genes</span>, all.x <span class="op">=</span> <span class="cn">TRUE</span>, by <span class="op">=</span> <span class="st">"gene"</span><span class="op">)</span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu">subset</span><span class="op">(</span><span class="va">df</span>, <span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a matrix whose rows are the spatially variable genes (indicated by Moran's I),</span></span>
<span><span class="co"># whose columns are the clusters, and whose entries are the adjusted DE pvalue for the</span></span>
<span><span class="co"># corresponding gene and cluster.</span></span>
<span><span class="va">p_val_adj_matrix</span> <span class="op">&lt;-</span> <span class="va">df</span> <span class="op">%&gt;%</span></span>
<span>                       <span class="fu">select</span><span class="op">(</span><span class="va">gene</span>, <span class="va">cluster</span>,<span class="va">p_val_adj</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                       <span class="fu">pivot_wider</span><span class="op">(</span>names_from <span class="op">=</span> <span class="va">cluster</span>, values_from <span class="op">=</span> <span class="va">p_val_adj</span>, values_fill <span class="op">=</span> <span class="fl">1.0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                       <span class="fu">column_to_rownames</span><span class="op">(</span><span class="st">"gene"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                       <span class="fu">as.matrix</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a heatmap of the DE pvalues of spatially variable genes</span></span>
<span><span class="fu">Heatmap</span><span class="op">(</span><span class="va">p_val_adj_matrix2</span>,</span>
<span>        column_title      <span class="op">=</span> <span class="st">"Heatmap of DE p-values of spatially DE genes"</span>,</span>
<span>        name              <span class="op">=</span> <span class="st">"DE p-values"</span>, <span class="co"># Title for the heatmap legend</span></span>
<span>        row_title         <span class="op">=</span> <span class="st">"Spatially variable genes"</span>,</span>
<span>        cluster_rows      <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>        cluster_columns   <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        show_row_names    <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>        show_column_names <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>        show_row_dend     <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>        show_column_dend  <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in eval(expr, envir, enclos): object 'p_val_adj_matrix2' not found</code></pre>
</div>
<p>The heatmap visualization reveals a key finding of our analysis:
genes displaying the highest Moran’s I values show distinct expression
patterns that align with specific brain regions identified through
expert annotations. This observation underscores the spatial correlation
of gene expression, highlighting its potential relevance in
understanding regional brain functions and pathologies.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>Differential expression testing pinpoints genes with significant
expression variations across samples, helping to decode biological and
disease mechanisms.</li>
<li>Moran’s I statistic is applied to reveal spatial autocorrelation in
gene expression, critical for examining spatially dependent biological
activities.</li>
<li>Moran’s I algorithm effectively identifies genes expressed in
anatomically distinct regions, as validated from the correlation
analysis with the DE genes from the annotated regions.</li>
</ul>
</div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</div>
</section></section>
</div>
    </main>
</div>
<!-- END  : inst/pkgdown/templates/content-extra.html -->

      </div>
<!--/div.row-->
      		<footer class="row footer mx-md-3"><hr>
<div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>
        
        <a href="https://github.com/smcclatchy/spatial-transcriptomics/edit/main/README.md" class="external-link">Edit on GitHub</a>
        
	
        | <a href="https://github.com/smcclatchy/spatial-transcriptomics/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/smcclatchy/spatial-transcriptomics/" class="external-link">Source</a></p>
				<p><a href="https://github.com/smcclatchy/spatial-transcriptomics/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:susan.mcclatchy@jax.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">
        
        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>
        
        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/epiverse-trace/sandpaper/tree/patch-renv-github-bug" class="external-link">sandpaper (0.16.4)</a>, <a href="https://github.com/carpentries/pegboard/tree/a32a7836d4455f407c3cafe8ab95edc636e5e919" class="external-link">pegboard (0.7.5)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.2" class="external-link">varnish (1.0.2)</a></p>
			</div>
		</footer>
</div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://smcclatchy.github.io/spatial-transcriptomics/aio.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, The Carpentries, transcriptome, Visium, cancer",
  "name": "All in One View",
  "creativeWorkStatus": "active",
  "url": "https://smcclatchy.github.io/spatial-transcriptomics/aio.html",
  "identifier": "https://smcclatchy.github.io/spatial-transcriptomics/aio.html",
  "dateCreated": "2024-03-01",
  "dateModified": "2024-06-13",
  "datePublished": "2024-06-13"
}

  </script><script>
		feather.replace();
	</script><!-- Matomo
    2022-11-07: we have gotten a notification that we have an overage for our
    tracking and I'm pretty sure this has to do with Workbench usage.
    Considering that I am not _currently_ using this tracking because I do not
    yet know how to access the data, I am turning this off for now.
  <script>
    var _paq = window._paq = window._paq || [];
    /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
    _paq.push(["setDocumentTitle", document.domain + "/" + document.title]);
    _paq.push(["setDomains", ["*.preview.carpentries.org","*.datacarpentry.github.io","*.datacarpentry.org","*.librarycarpentry.github.io","*.librarycarpentry.org","*.swcarpentry.github.io", "*.carpentries.github.io"]]);
    _paq.push(["setDoNotTrack", true]);
    _paq.push(["disableCookies"]);
    _paq.push(['trackPageView']);
    _paq.push(['enableLinkTracking']);
    (function() {
          var u="https://carpentries.matomo.cloud/";
          _paq.push(['setTrackerUrl', u+'matomo.php']);
          _paq.push(['setSiteId', '1']);
          var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
          g.async=true; g.src='https://cdn.matomo.cloud/carpentries.matomo.cloud/matomo.js'; s.parentNode.insertBefore(g,s);
        })();
  </script>
  End Matomo Code -->
</body>
</html><!-- END:   inst/pkgdown/templates/layout.html-->

