<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Spatial Transcriptomics: Data Preprocessing</title><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="stylesheet" type="text/css" href="assets/styles.css"><script src="assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png"><link rel="manifest" href="site.webmanifest"><link rel="mask-icon" href="safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" content="#ffffff"></head><body>
    <header id="top" class="navbar navbar-expand-md navbar-light bg-white top-nav incubator"><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-6">
      <div class="large-logo">
        <img alt="Carpentries Incubator" src="assets/images/incubator-logo.svg"><abbr class="badge badge-light" title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught." style="background-color: #FF4955; border-radius: 5px">
          <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link" style="color: #000">
            <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
            Pre-Alpha
          </a>
          <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
        </abbr>
        
      </div>
    </div>
    <div class="selector-container">
      
      
      <div class="dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Learner View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='instructor/data-preprocessing.html';">Instructor View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl navbar-light bg-white bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Carpentries Incubator" src="assets/images/incubator-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Spatial Transcriptomics
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Spatial Transcriptomics
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="reference.html#glossary">Glossary</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="profiles.html">Learner Profiles</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a class="btn btn-primary" href="aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Spatial Transcriptomics
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 33%" class="percentage">
    33%
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: 33%" aria-valuenow="33" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Learner View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="instructor/data-preprocessing.html">Instructor View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->
      
            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Setup</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="introduction.html">1. Spatially Resolved Transcriptomics in Life Sciences Research</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="data-and-study-design.html">2. Data and Study Design</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        3. Data Preprocessing
        </span>
      </button>
    </div><!--/div.accordion-header-->
        
    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#introduction">Introduction</a></li>
<li><a href="#set-up-environment">Set up Environment</a></li>
<li><a href="#load-raw-and-filtered-spatial-expression-data">Load Raw and Filtered Spatial Expression Data</a></li>
<li><a href="#add-spot-metadata">Add Spot Metadata</a></li>
<li><a href="#plot-transcript-and-feature-counts">Plot Transcript and Feature Counts</a></li>
<li><a href="#conclusion">Conclusion</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="remove-low-quality-spots.html">4. Remove Low-quality Spots</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="apply-normalization-methods.html">5. Normalization in Spatial Transcriptomics</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="apply-feature-selection-methods.html">6. Apply Feature Selection Methods</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="apply-dimensionality-reduction-methods.html">7. Apply Dimensionality Reduction Methods</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="apply-clustering-algorithms.html">8. Apply Clustering Algorithms</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="deconvolve-cell-types-in-a-spot.html">9. Deconvolve Cell Types in a Spot</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush11">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading11">
        <a href="differential-expression-testing.html">10. Differential Expression Testing</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="reference.html#glossary">Glossary</a>
                      </li>
                      <li>
                        <a href="profiles.html">Learner Profiles</a>
                      </li>
                      <li><a href="reference.html">Reference</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width resources"><a href="aio.html">See all in one page</a>
            

            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">
            
            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="data-and-study-design.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="remove-low-quality-spots.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="data-and-study-design.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Data and Study
        </a>
        <a class="chapter-link float-end" href="remove-low-quality-spots.html" rel="next">
          Next: Remove Low-quality... 
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Data Preprocessing</h1>
        <p>Last updated on 2024-04-19 |
        
        <a href="https://github.com/smcclatchy/spatial-transcriptomics/edit/main/episodes/data-preprocessing.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
        
        
        
        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>What data files should I expect from the sequencing core?</li>
<li>Which data preprocessing steps are required to prepare the raw data
files for further analysis?</li>
<li>What software will we use for data preprocessing?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Explain how to use markdown with the new lesson template</li>
<li>Demonstrate how to include pieces of code, figures, and nested
challenge blocks</li>
</ul></div>
</div>
</div>
</div>
</div>
<section id="introduction"><h2 class="section-heading">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width"><p>The sequencing core will deliver a set of files and directories. The
complete set of files is listed in the table below.</p>
<table class="table"><colgroup><col width="48%"><col width="52%"></colgroup><thead><tr class="header"><th>File Name</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td>web_summary.html</td>
<td>Run summary metrics and plots in HTML format</td>
</tr><tr class="even"><td>cloupe.cloupe</td>
<td>Loupe Browser visualization and analysis file</td>
</tr><tr class="odd"><td>spatial/</td>
<td>Folder containing outputs that capture the spatiality of the
data.</td>
</tr><tr class="even"><td>spatial/aligned_fiducials.jpg</td>
<td>Aligned fiducials QC image</td>
</tr><tr class="odd"><td>spatial/aligned_tissue_image.jpg</td>
<td>Aligned CytAssist and Microscope QC image. Present only for
CytAssist workflow</td>
</tr><tr class="even"><td>spatial/barcode_fluorescence_intensity.csv</td>
<td>CSV file containing the mean and standard deviation of fluorescence
intensity for each spot and each channel. Present for the fluorescence
image input specified by –darkimage</td>
</tr><tr class="odd"><td>spatial/cytassist_image.tiff</td>
<td>Input CytAssist image in original resolution that can be used to
re-run the pipeline. Present only for CytAssist workflow</td>
</tr><tr class="even"><td>spatial/detected_tissue_image.jpg</td>
<td>Detected tissue QC image.</td>
</tr><tr class="odd"><td>spatial/scalefactors_json.json</td>
<td>Scale conversion factors for spot diameter and coordinates at
various image resolutions</td>
</tr><tr class="even"><td>spatial/spatial_enrichment.csv</td>
<td>Feature spatial autocorrelation analysis using Moran’s I in CSV
format</td>
</tr><tr class="odd"><td>spatial/tissue_hires_image.png</td>
<td>Downsampled full resolution image. The image dimensions depend on
the input image and slide version</td>
</tr><tr class="even"><td>spatial/tissue_lowres_image.png</td>
<td>Full resolution image downsampled to 600 pixels on the longest
dimension</td>
</tr><tr class="odd"><td>spatial/tissue_positions.csv</td>
<td>CSV containing spot barcode; if the spot was called under (1) or out
(0) of tissue, the array position, image pixel position x, and image
pixel position y for the full resolution image</td>
</tr><tr class="even"><td>analysis/</td>
<td>Folder containing secondary analysis data including graph-based
clustering and K-means clustering (K = 2-10); differential gene
expression between clusters; PCA, t-SNE, and UMAP dimensionality
reduction.</td>
</tr><tr class="odd"><td>metrics_summary.csv</td>
<td>Run summary metrics in CSV format</td>
</tr><tr class="even"><td>probe_set.csv</td>
<td>Copy of the input probe set reference CSV file. Present for Visium
FFPE and CytAssist workflow</td>
</tr><tr class="odd"><td>possorted_genome_bam.bam</td>
<td>Indexed BAM file containing position-sorted reads aligned to the
genome and transcriptome, annotated with barcode information</td>
</tr><tr class="even"><td>possorted_genome_bam.bam.bai</td>
<td>Index for possorted_genome_bam.bam. In cases where the reference
transcriptome is generated from a genome with very long chromosomes
(&gt;512 Mbp), Space Ranger v2.0+ generates a
possorted_genome_bam.bam.csi index file instead.</td>
</tr><tr class="odd"><td>filtered_feature_bc_matrix/</td>
<td>Contains only tissue-associated barcodes in MEX format. Each element
of the matrix is the number of UMIs associated with a feature (row) and
a barcode (column). This file can be input into third-party packages and
allows users to wrangle the barcode-feature matrix (e.g. to filter
outlier spots, run dimensionality reduction, normalize gene
expression).</td>
</tr><tr class="even"><td><strong>filtered_feature_bc_matrix.h5</strong></td>
<td><strong>Same information as filtered_feature_bc_matrix/ but in HDF5
format.</strong></td>
</tr><tr class="odd"><td>raw_feature_bc_matrices/</td>
<td>Contains all detected barcodes in MEX format. Each element of the
matrix is the number of UMIs associated with a feature (row) and a
barcode (column).</td>
</tr><tr class="even"><td><strong>raw_feature_bc_matrix.h5</strong></td>
<td><strong>Same information as raw_feature_bc_matrices/ in HDF5
format.</strong></td>
</tr><tr class="odd"><td>﻿</td>
<td>raw_probe_bc_matrix.h5</td>
</tr><tr class="even"><td>molecule_info.h5</td>
<td>Contains per-molecule information for all molecules that contain a
valid barcode, valid UMI, and were assigned with high confidence to a
gene or protein barcode. This file is required for additional analysis
spaceranger pipelines including aggr, targeted-compare and
targeted-depth.</td>
</tr></tbody></table><p>Fortunately, you will not need to look at all of these files. We
provide a brief description for you in case you are curious or need to
look at one of the files for technical reasons.</p>
<p>The two files that you will use are “raw_feature_bc_matrix.h5” and
“filtered_feature_bc_matrix.h5”. These files have an “h5” suffix, which
means that they are HDF5 files. <a href="https://www.hdfgroup.org/solutions/hdf5/" class="external-link">HDF5</a> is a compressed
file format for storing complex high-dimensional data. “HDF5 stands
for”Hierarchical Data Formats, version 5”. There is an R package
designed to read and write HDF5 files called <a href="https://bioconductor.org/packages/release/bioc/html/rhdf5.html" class="external-link">rhdf5</a>.
This was one of the packages which you installed during the lesson
setup.</p>
<p>Briefly, HDF5 organizes data into directories within the compressed
file. There are three “files” within the HDF5 file:</p>
<table class="table"><colgroup><col width="53%"><col width="46%"></colgroup><thead><tr class="header"><th>File Name</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td>features.csv</td>
<td>Contains the features (i.e. genes in this case) for each row in the
data matrix.</td>
</tr><tr class="even"><td>barcodes.csv</td>
<td>Contains the probe barcodes for each spot on the tissue block.</td>
</tr><tr class="odd"><td>matrix.mtx</td>
<td>Contains the counts for each gene in each spot. Features
(e.g. genes) are in rows and barcodes (e.g. spots) are in columns.</td>
</tr></tbody></table></section><section id="set-up-environment"><h2 class="section-heading">Set up Environment<a class="anchor" aria-label="anchor" href="#set-up-environment"></a>
</h2>
<hr class="half-width"><p>Go to the “File” menu and select “Open Project…”. Open the
“spatialRNA” project which you created in the workshop Setup.</p>
<p>First, we will load in some utility functions to make our lives a bit
easier. The <a href="https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/source" class="external-link">source</a>
function reads an R file and runs the code in it. In this case, this
will load several useful functions.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">source</span><span class="op">(</span><span class="st">"https://raw.githubusercontent.com/smcclatchy/spatial-transcriptomics/main/code/spatial_utils.R"</span><span class="op">)</span></span></code></pre>
</div>
<p>We will then load the libraries that we need for this lesson.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span><span class="kw">library</span><span class="op">(</span><span class="va">hdf5r</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span><span class="kw">library</span><span class="op">(</span><span class="va">Seurat</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Note that the <a href="https://here.r-lib.org/" class="external-link">here library</a>
helps you to find your files by taking care of the absolute path.</p>
</section><section id="load-raw-and-filtered-spatial-expression-data"><h2 class="section-heading">Load Raw and Filtered Spatial Expression Data<a class="anchor" aria-label="anchor" href="#load-raw-and-filtered-spatial-expression-data"></a>
</h2>
<hr class="half-width"><p>We will use the <a href="https://www.rdocumentation.org/packages/Seurat/versions/5.0.1/topics/Load10X_Spatial" class="external-link">Load10X_Spatial</a>
function from the <a href="https://satijalab.org/seurat/" class="external-link">Seurat</a>
package to read in the spatial transcription data. This is the data
which you downloaded in the setup section.</p>
<p>First, we will read in the raw data for sample 151508.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">raw_st</span> <span class="op">&lt;-</span> <span class="fu">Load10X_Spatial</span><span class="op">(</span>data.dir <span class="op">=</span> <span class="st">"./data/151508"</span>, </span>
<span>                          filename <span class="op">=</span> <span class="st">"151508_raw_feature_bc_matrix.h5"</span>,</span>
<span>                          filter.matrix <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre>
</div>
<p>If you did not see any error messages, then the data loaded in and
you should see an “raw_st” object in your “Environment” tab on the
right.</p>

<p>Let’s look at the “raw_st” object.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">raw_st</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>An object of class Seurat 
33538 features across 4992 samples within 1 assay 
Active assay: Spatial (33538 features, 0 variable features)
 1 layer present: counts
 1 image present: slice1</code></pre>
</div>
<p>The output says that we have 33,538 “features” and 4,992 “samples”
with one assay. “Feature” is a generic term for anything that we
measured. In this case, we measured gene expression, so each feature is
a gene. Each “sample” is one spot on the spatial slide. So this tissue
sample has 33,538 genes assayed across 4,992 spots.</p>
<p>An experiment may have more than one assay. For example, you may run
both RNA sequencing and chromatin accessibility in the same set of
samples. In this case, we have one assay, which contains the RNA-seq
counts for each spot.</p>
<p>There is also a single image called ‘slice1’ attached to the Seurat
object.</p>
<p>Next, we will load in the filtered data. Use the code above and look
in a file browser to identify the “filtered” file for sample 151508.</p>
<div id="challenge-1-read-in-filtered-hdf5-file-for-sample-151508." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-read-in-filtered-hdf5-file-for-sample-151508." class="callout-inner">
<h3 class="callout-title">Challenge 1: Read in filtered HDF5 file for
sample 151508.<a class="anchor" aria-label="anchor" href="#challenge-1-read-in-filtered-hdf5-file-for-sample-151508."></a>
</h3>
<div class="callout-content">
<p>Open a file browser and navigate to “Desktop/spatialRNA/data/151508”.
Can you find an HDF5 file (with an “.h5” suffix) that haw the word
“filtered” in it? If so, read that file in and assign it to a variable
called “filter_st”.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">Solution 1</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filter_st</span> <span class="op">&lt;-</span> <span class="fu">Load10X_Spatial</span><span class="op">(</span>data.dir <span class="op">=</span> <span class="st">"./data/151508"</span>, </span>
<span>                             filename <span class="op">=</span> <span class="st">"151508_filtered_feature_bc_matrix.h5"</span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>
</div>
</div>
<p>Once you have the filtered data loaded in, look at the object.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filter_st</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>An object of class Seurat 
33538 features across 4384 samples within 1 assay 
Active assay: Spatial (33538 features, 0 variable features)
 1 layer present: counts
 1 image present: slice1</code></pre>
</div>
<p>The raw and filtered data both have 33,538 genes. But the filtered
data has fewer spots. The raw data had 4,992 spots and the filtered data
has 4,384 spots.</p>
<p>Look at the H &amp; E slide below and notice the grey spots around
the border. These are used by the spatial transcriptomics software to
“register” the image.</p>
<figure><img src="fig/tissue_hires_image.png" alt="H &amp; E slide of sample 151508" class="figure mx-auto d-block"><div class="figcaption">H &amp; E slide of sample 151508</div>
</figure><div id="challenge-2-how-does-the-computer-know-how-to-orient-the-image." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-how-does-the-computer-know-how-to-orient-the-image." class="callout-inner">
<h3 class="callout-title">Challenge 2: How does the computer know how to
orient the image?.<a class="anchor" aria-label="anchor" href="#challenge-2-how-does-the-computer-know-how-to-orient-the-image."></a>
</h3>
<div class="callout-content">
<p>Look carefully at the spots in the H &amp; E image above. Are the
spots symmetric? Is there anything different about the spots that might
help a computer to assign up/down and left/right to the image?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2">Solution 2</h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>Look at the spots in each corner. In the upper-left, you will see the
following patterns:</p>
<figure><img src="fig/tissue_diagram_with_registration_spots.png" alt="Diagram showing tissue and registration spots in corners" class="figure mx-auto d-block"><div class="figcaption">Slide Diagram</div>
</figure><p>The patterns in each corner allow the spatial transcriptomics
software to orient the slide.</p>
</div>
</div>
</div>
</div>
</section><section id="add-spot-metadata"><h2 class="section-heading">Add Spot Metadata<a class="anchor" aria-label="anchor" href="#add-spot-metadata"></a>
</h2>
<hr class="half-width"><p>Next, we will read a file containing information about whether each
spot is in the background or the tissue. This file does not contain
column names, although the next version of <a href="https://www.10xgenomics.com/support/software/space-ranger/latest/analysis/outputs/spatial-outputs" class="external-link">SpaceRanger
2.0</a>, which is used to process the data at the sequencing core,
should add column names to this file.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tissue_position</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="st">"./data/151508/spatial/tissue_positions_list.csv"</span>,</span>
<span>                            col_names <span class="op">=</span> <span class="cn">FALSE</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>                     <span class="fu">column_to_rownames</span><span class="op">(</span><span class="st">'X1'</span><span class="op">)</span></span>
<span><span class="fu">colnames</span><span class="op">(</span><span class="va">tissue_position</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">"in_tissue"</span>, </span>
<span>                               <span class="st">"array_row"</span>, </span>
<span>                               <span class="st">"array_col"</span>, </span>
<span>                               <span class="st">"pxl_row_in_fullres"</span>, </span>
<span>                               <span class="st">"pxl_col_in_fullres"</span><span class="op">)</span></span></code></pre>
</div>
<p>It is important to note that the order of the spots differs between
the Seurat object and the tissue position file. We need to reorder the
tissue positions to match the Seurat object. We can extract the spot
barcodes using the <a href="https://satijalab.org/seurat/reference/cells" class="external-link">Cells()</a>
function. This is named for the earlier versions of Seurat, which
processed single cell transcript data. In this case, we are getting
<strong>spot</strong> IDs, even though the function is called
“Cells”.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tissue_position</span> <span class="op">&lt;-</span> <span class="va">tissue_position</span><span class="op">[</span><span class="fu">Cells</span><span class="op">(</span><span class="va">raw_st</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="fu">stopifnot</span><span class="op">(</span><span class="fu">rownames</span><span class="op">(</span><span class="va">tissue_position</span><span class="op">)</span> <span class="op">==</span> <span class="fu">Cells</span><span class="op">(</span><span class="va">raw_st</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Now that we have aligned the barcodes between the Seurat object and
the tissue positions, we can add the tissue positions to the Seurat
object’s metadata.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">raw_st</span>    <span class="op">&lt;-</span> <span class="fu">AddMetaData</span><span class="op">(</span>object <span class="op">=</span> <span class="va">raw_st</span>,    metadata <span class="op">=</span> <span class="va">tissue_position</span><span class="op">)</span></span>
<span><span class="va">filter_st</span> <span class="op">&lt;-</span> <span class="fu">AddMetaData</span><span class="op">(</span>object <span class="op">=</span> <span class="va">filter_st</span>, metadata <span class="op">=</span> <span class="va">tissue_position</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we will plot the spot annotation, indicating spots that are in
the tissue in blue and background spots in red.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">SpatialPlot</span><span class="op">(</span><span class="va">raw_st</span>, group.by <span class="op">=</span> <span class="st">"in_tissue"</span><span class="op">)</span></span></code></pre>
</div>
<figure style="text-align: center"><img src="fig/data-preprocessing-rendered-unnamed-chunk-10-1.png" alt="Histology slide with tissue and background spots labelled" class="figure mx-auto d-block"><figcaption>
Spots identified in Tissue and Background
</figcaption></figure><p>We expect most of the transcript counts to be in the tissue spots.
The Seurat object metadata contains a count of the number of transcripts
in each spot called “nCount_Spatial”. Let’s plot the counts in the
tissue and background spots.</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">raw_st</span><span class="op">@</span><span class="va">meta.data</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="fu">as.logical</span><span class="op">(</span><span class="va">in_tissue</span><span class="op">)</span>, <span class="va">nCount_Spatial</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">'Transcript Counts in Tissue and Background'</span>,</span>
<span>         x     <span class="op">=</span> <span class="st">'In Tissue?'</span>,</span>
<span>         y     <span class="op">=</span> <span class="st">'Counts'</span><span class="op">)</span></span></code></pre>
</div>
<figure style="text-align: center"><img src="fig/data-preprocessing-rendered-unnamed-chunk-11-1.png" alt="Boxplot showing lower trancript counts in background area of slide" class="figure mx-auto d-block"><figcaption>
Transcript Counts in Tissue and Background
</figcaption></figure><p>As expected, we see most of the counts in the tissue spots.</p>
<p>We can also plot the number of genes detected in each spot. Seurat
calls genes “features”, so we will plot the “nFeature_Spatial” value.
This is stored in the metadata of the Seurat object.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">raw_st</span><span class="op">@</span><span class="va">meta.data</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="fu">as.logical</span><span class="op">(</span><span class="va">in_tissue</span><span class="op">)</span>, <span class="va">nFeature_Spatial</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">'Number of Genes in Tissue and Background'</span>,</span>
<span>         x     <span class="op">=</span> <span class="st">'In Tissue?'</span>,</span>
<span>         y     <span class="op">=</span> <span class="st">'Number of Genes'</span><span class="op">)</span></span></code></pre>
</div>
<figure style="text-align: center"><img src="fig/data-preprocessing-rendered-unnamed-chunk-12-1.png" alt="Boxplot showing lower numbers of genes in background area of slide" class="figure mx-auto d-block"><figcaption>
Number of Genes in Tissue and Background
</figcaption></figure><div id="challenge-3-why-might-there-be-transcript-counts-outside-of-the-tissue-boundaries" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3-why-might-there-be-transcript-counts-outside-of-the-tissue-boundaries" class="callout-inner">
<h3 class="callout-title">Challenge 3: Why might there be transcript
counts outside of the tissue boundaries?<a class="anchor" aria-label="anchor" href="#challenge-3-why-might-there-be-transcript-counts-outside-of-the-tissue-boundaries"></a>
</h3>
<div class="callout-content">
<p>We expect transcript counts in the spots which overlap with the
tissue section. What reasons can you think of that might lead transcript
counts to occur in the background spots?</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3">Solution 3</h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">
<ol style="list-style-type: decimal"><li>When the tissue section is lysed, some transcripts may leak out of
the cells and into the background region of the slide. &gt; DMG: Ask the
core.</li>
</ol></div>
</div>
</div>
</div>
<p>Up to this point, we have been working with the raw, unfiltered data
to show you how the spots are filtered. However, in most workflows, you
will work directly with the filtered file. From this point forward, we
will work with the filtered data object.</p>
<p>Let’s plot the spots in the tissue in the filtered object to verify
that it is only using spots in the tissue.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot1</span> <span class="op">&lt;-</span> <span class="fu">SpatialDimPlot</span><span class="op">(</span><span class="va">filter_st</span>, alpha <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>           <span class="fu">NoLegend</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">plot2</span> <span class="op">&lt;-</span> <span class="fu">SpatialDimPlot</span><span class="op">(</span><span class="va">filter_st</span><span class="op">)</span> <span class="op">+</span> </span>
<span>           <span class="fu">NoLegend</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">plot1</span> <span class="op">|</span> <span class="va">plot2</span></span></code></pre>
</div>
<figure><img src="fig/data-preprocessing-rendered-unnamed-chunk-13-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure></section><section id="plot-transcript-and-feature-counts"><h2 class="section-heading">Plot Transcript and Feature Counts<a class="anchor" aria-label="anchor" href="#plot-transcript-and-feature-counts"></a>
</h2>
<hr class="half-width"><p>Next, we want to look at the distribution of transcript counts and
numbers of genes in each spot across the tissue. This can be helpful in
identifying technical issues with the sample processing. We will use
Seurat’s <a href="https://satijalab.org/seurat/reference/spatialplot" class="external-link">SpatialFeaturePlot</a>
function to look at these values. We can color the spots based on the
spot metadata stored in the Seurat object. You can find these column
names by looking at the “meta.data” slot of the Seurat object.</p>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">filter_st</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                      orig.ident nCount_Spatial nFeature_Spatial in_tissue
AAACAAGTATCTCCCA-1 SeuratProject           3627             1847         1
AAACAATCTACTAGCA-1 SeuratProject            956              635         1
AAACACCAATAACTGC-1 SeuratProject            990              716         1
AAACAGAGCGACTCCT-1 SeuratProject           1377              862         1
AAACAGCTTTCAGAAG-1 SeuratProject           3215             1720         1
AAACAGGGTCTATATT-1 SeuratProject           3035             1617         1
                   array_row array_col pxl_row_in_fullres pxl_col_in_fullres
AAACAAGTATCTCCCA-1        50       102               8586               9171
AAACAATCTACTAGCA-1         3        43               2944               5127
AAACACCAATAACTGC-1        59        19               9646               3454
AAACAGAGCGACTCCT-1        14        94               4273               8634
AAACAGCTTTCAGAAG-1        43         9               7728               2772
AAACAGGGTCTATATT-1        47        13               8208               3046</code></pre>
</div>
<p>To plot the transcript counts, we will use the “nCount_Spatial”
column in the spot metadata.</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot1</span> <span class="op">&lt;-</span> <span class="fu">SpatialDimPlot</span><span class="op">(</span><span class="va">filter_st</span>, alpha <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>           <span class="fu">NoLegend</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">plot2</span> <span class="op">&lt;-</span> <span class="fu">SpatialFeaturePlot</span><span class="op">(</span><span class="va">filter_st</span>, features <span class="op">=</span> <span class="st">"nCount_Spatial"</span><span class="op">)</span></span>
<span><span class="va">plot1</span> <span class="op">|</span> <span class="va">plot2</span></span></code></pre>
</div>
<figure style="text-align: center"><img src="fig/data-preprocessing-rendered-unnamed-chunk-15-1.png" alt="Figure showing transcript counts in each spot with varying intensity across the tissue" class="figure mx-auto d-block"><figcaption>
Transcript Counts in each Spot
</figcaption></figure><p>In this case, we see a band of higher counts running from upper-left
to lower-right. There are also bands of lower counts above and below
this band. The band in the upper-right corner may be due to the fissure
in the tissue. It is less clear why the expression is low in the
lower-left corner.</p>
<p>We can also look at the number of genes detected in each spot using
“nFeature_Spatial”.</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot1</span> <span class="op">&lt;-</span> <span class="fu">SpatialDimPlot</span><span class="op">(</span><span class="va">filter_st</span>, alpha <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>           <span class="fu">NoLegend</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">plot2</span> <span class="op">&lt;-</span> <span class="fu">SpatialFeaturePlot</span><span class="op">(</span><span class="va">filter_st</span>, features <span class="op">=</span> <span class="st">"nFeature_Spatial"</span><span class="op">)</span></span>
<span><span class="va">plot1</span> <span class="op">|</span> <span class="va">plot2</span></span></code></pre>
</div>
<figure style="text-align: center"><img src="fig/data-preprocessing-rendered-unnamed-chunk-16-1.png" alt="Figure showing number of genes detected in each spot with varying intensity across the tissue" class="figure mx-auto d-block"><figcaption>
Number of Genes in each Spot
</figcaption></figure><p>It is difficult to lay out a broad set of rules that will work for
all types of tissues and samples. Some tissues may have homogeneous
transcript counts across the section, while others may show variation in
transcript counts due to tissue structure. For example, in cancer tissue
sections, stromal cells tend to have lower counts than tumor cells and
this should be evident in a transcript count plot. In the brain sample
below, we might expect some variation in transcript counts in different
regions of the brain.</p>
</section><section id="conclusion"><h2 class="section-heading">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<hr class="half-width"><div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul><li>The sequencing core will provide you with an unfiltered and a
filtered data file.</li>
<li>
</li>
</ul></div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="data-and-study-design.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="remove-low-quality-spots.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="data-and-study-design.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Data and Study
        </a>
        <a class="chapter-link float-end" href="remove-low-quality-spots.html" rel="next">
          Next: Remove Low-quality... 
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>
        
        <a href="https://github.com/smcclatchy/spatial-transcriptomics/edit/main/episodes/data-preprocessing.Rmd" class="external-link">Edit on GitHub</a>
        
	
        | <a href="https://github.com/smcclatchy/spatial-transcriptomics/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/smcclatchy/spatial-transcriptomics/" class="external-link">Source</a></p>
				<p><a href="https://github.com/smcclatchy/spatial-transcriptomics/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:susan.mcclatchy@jax.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">
        
        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>
        
        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.4" class="external-link">sandpaper (0.16.4)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.5" class="external-link">pegboard (0.7.5)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.2" class="external-link">varnish (1.0.2)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://smcclatchy.github.io/spatial-transcriptomics/data-preprocessing.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, The Carpentries, transcriptome, Visium, cancer",
  "name": "Data Preprocessing",
  "creativeWorkStatus": "active",
  "url": "https://smcclatchy.github.io/spatial-transcriptomics/data-preprocessing.html",
  "identifier": "https://smcclatchy.github.io/spatial-transcriptomics/data-preprocessing.html",
  "dateCreated": "2024-03-01",
  "dateModified": "2024-04-19",
  "datePublished": "2024-04-19"
}

  </script><script>
		feather.replace();
	</script><!-- Matomo
    2022-11-07: we have gotten a notification that we have an overage for our
    tracking and I'm pretty sure this has to do with Workbench usage.
    Considering that I am not _currently_ using this tracking because I do not
    yet know how to access the data, I am turning this off for now.
  <script>
    var _paq = window._paq = window._paq || [];
    /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
    _paq.push(["setDocumentTitle", document.domain + "/" + document.title]);
    _paq.push(["setDomains", ["*.preview.carpentries.org","*.datacarpentry.github.io","*.datacarpentry.org","*.librarycarpentry.github.io","*.librarycarpentry.org","*.swcarpentry.github.io", "*.carpentries.github.io"]]);
    _paq.push(["setDoNotTrack", true]);
    _paq.push(["disableCookies"]);
    _paq.push(['trackPageView']);
    _paq.push(['enableLinkTracking']);
    (function() {
          var u="https://carpentries.matomo.cloud/";
          _paq.push(['setTrackerUrl', u+'matomo.php']);
          _paq.push(['setSiteId', '1']);
          var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
          g.async=true; g.src='https://cdn.matomo.cloud/carpentries.matomo.cloud/matomo.js'; s.parentNode.insertBefore(g,s);
        })();
  </script>
  End Matomo Code --></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

