<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Spatial Transcriptomics: Differential Expression Testing</title><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="stylesheet" type="text/css" href="assets/styles.css"><script src="assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png"><link rel="manifest" href="site.webmanifest"><link rel="mask-icon" href="safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" content="#ffffff"></head><body>
    <header id="top" class="navbar navbar-expand-md navbar-light bg-white top-nav incubator"><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-6">
      <div class="large-logo">
        <img alt="Carpentries Incubator" src="assets/images/incubator-logo.svg"><abbr class="badge badge-light" title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught." style="background-color: #FF4955; border-radius: 5px">
          <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link" style="color: #000">
            <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
            Pre-Alpha
          </a>
          <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
        </abbr>
        
      </div>
    </div>
    <div class="selector-container">
      
      
      <div class="dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Learner View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='instructor/differential-expression-testing.html';">Instructor View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl navbar-light bg-white bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Carpentries Incubator" src="assets/images/incubator-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Spatial Transcriptomics
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Spatial Transcriptomics
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="reference.html#glossary">Glossary</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="profiles.html">Learner Profiles</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a class="btn btn-primary" href="aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Spatial Transcriptomics
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 66%" class="percentage">
    66%
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: 66%" aria-valuenow="66" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Learner View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="instructor/differential-expression-testing.html">Instructor View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->
      
            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Setup</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="introduction.html">1. Spatially Resolved Transcriptomics in Life Sciences Research</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="data-and-study-design.html">2. Data and Study Design</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="data-preprocessing.html">3. Data Preprocessing</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="remove-low-quality-spots.html">4. Remove Low-quality Spots</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="apply-normalization-methods.html">5. Normalization in Spatial Transcriptomics</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="feature-selection-dimensionality-reduction-clustering.html">6. Feature Selection, Dimensionality Reduction, and Spot Clustering</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="deconvolve-cell-types-in-a-spot.html">7. Deconvolution in Spatial Transcriptomics</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        8. Differential Expression Testing
        </span>
      </button>
    </div><!--/div.accordion-header-->
        
    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#spot-level-differential-expression-using-expert-annotation">Spot-level Differential Expression Using Expert Annotation</a></li>
<li><a href="#morans-i-statistic">Moran’s I Statistic</a></li>
<li><a href="#correlation-of-region-specific-differentially-expressed-genes-and-spatially-variable-genes">Correlation of Region-specific Differentially Expressed Genes and
Spatially Variable Genes</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="final-exercise.html">9. Putting it all Together</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="reference.html#glossary">Glossary</a>
                      </li>
                      <li>
                        <a href="profiles.html">Learner Profiles</a>
                      </li>
                      <li><a href="reference.html">Reference</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width resources"><a href="aio.html">See all in one page</a>
            

            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">
            
            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="deconvolve-cell-types-in-a-spot.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="final-exercise.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="deconvolve-cell-types-in-a-spot.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Deconvolution in
        </a>
        <a class="chapter-link float-end" href="final-exercise.html" rel="next">
          Next: Putting it all... 
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Differential Expression Testing</h1>
        <p>Last updated on 2024-06-14 |
        
        <a href="https://github.com/smcclatchy/spatial-transcriptomics/edit/main/episodes/differential-expression-testing.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
        
        
        
        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>How can we access region-specific gene expression using differential
expression?</li>
<li>How can we access spatially varying gene expression using spatial
statistics?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Identify differentially expressed genes across different layers
defined by expert annotations.</li>
<li>Utilize the Moran’s I statistic to find spatially variable
genes.</li>
<li>Explore the relationship between layer-specific genes identified by
differential expression and spatially varying genes identified by
Moran’s I.</li>
</ul></div>
</div>
</div>
</div>
</div>
<section id="spot-level-differential-expression-using-expert-annotation"><h2 class="section-heading">Spot-level Differential Expression Using Expert Annotation<a class="anchor" aria-label="anchor" href="#spot-level-differential-expression-using-expert-annotation"></a>
</h2>
<hr class="half-width"><p>We will begin by performing differential expression across the
annotated layers. We will use the source publication’s spot annotation,
which is stored in the “layer_guess” column of the metadata. As a
reminder, those look like:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">SpatialDimPlotColorSafe</span><span class="op">(</span><span class="va">filter_st</span><span class="op">[</span>, <span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">filter_st</span><span class="op">[[</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">layer_guess</span><span class="op">)</span><span class="op">]</span>, <span class="st">"layer_guess"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">labs</span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"Layer"</span><span class="op">)</span> </span></code></pre>
</div>
<figure><img src="fig/differential-expression-testing-rendered-layers-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>We identify genes that are upregulated in each annotated brain region
using the <a href="https://satijalab.org/seurat/reference/findallmarkers" class="external-link">FindAllMarkers</a>
function in Seurat. This performs a “one versus rest” comparison of a
gene’s expression in one region relative to that gene’s expression in
all other regions. The default test used here, the Wilcoxon Rank Sum
test, will use an efficient implementation within the <a href="https://github.com/immunogenomics/presto" class="external-link">presto</a> library, if
installed. The speedup over the default implementation is substantial,
and we highly recommend installing presto and using Seurat v5, which
leverages it.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">Idents</span><span class="op">(</span><span class="va">filter_st</span><span class="op">)</span>  <span class="op">&lt;-</span> <span class="st">"layer_guess"</span></span>
<span><span class="va">de_genes</span>           <span class="op">&lt;-</span> <span class="fu">FindAllMarkers</span><span class="op">(</span><span class="va">filter_st</span>, </span>
<span>                                     assay    <span class="op">=</span> <span class="st">"SCT"</span>,</span>
<span>                                     verbose  <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                     only.pos <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                     min.pct  <span class="op">=</span> <span class="fl">0.25</span>, </span>
<span>                                     logfc.threshold <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span></span></code></pre>
</div>
<p>The resulting table indicates DE p-values and adjusted p-values for
each gene, along with the percentage of spots in which the gene was
detected (pct.1) in the corresponding cluster and the percentage of
spots in which it was detected in all <em>other</em> clusters
(pct.2):</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">de_genes</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                p_val avg_log2FC pct.1 pct.2     p_val_adj cluster    gene
MT-CO1  7.105768e-163  0.3732786 1.000 1.000 1.278754e-158  Layer3  MT-CO1
ENC1    1.654660e-142  0.8693181 0.998 0.929 2.977725e-138  Layer3    ENC1
MT-ATP6 6.825005e-127  0.3333658 1.000 1.000 1.228228e-122  Layer3 MT-ATP6
MT-CYB  8.895560e-118  0.3290618 1.000 1.000 1.600845e-113  Layer3  MT-CYB
MT-CO2  3.349139e-116  0.2869993 1.000 1.000 6.027110e-112  Layer3  MT-CO2
MT-CO3  5.357942e-105  0.2766345 1.000 1.000 9.642152e-101  Layer3  MT-CO3</code></pre>
</div>
<p>It is not uncommon to have pathologist annotations of regions. The
Visium assay is performed on a tissue that is also stained for
hematoxylin and eosin (H&amp;E) – a routine practice in pathology for
diagnosing cancer, for example. A pathologist could manually annotate
this H&amp;E image using a histology viewer, such as QuPath.</p>
<p>In cases where we do not have expert annotations, we could perform
the analysis above across clusters. Indeed, this is often the first step
to interpreting a cluster – comparing its marker genes to those of
regions expected within the tissue. It is important to remember,
however, that these markers are for clusters of spots – aggregations of
cells – not individual cells, as would be the case in scRNA-seq. Those
aggregations may be of cells with similar types, in which case analysis
may be similar to that of scRNA-seq, or of different cell types, in
which case the interpretation would be quite different than with
scRNA-seq.</p>
</section><section id="morans-i-statistic"><h2 class="section-heading">Moran’s I Statistic<a class="anchor" aria-label="anchor" href="#morans-i-statistic"></a>
</h2>
<hr class="half-width"><p>We next consider an alternative to the above approach that relies on
defined regions – either through expert annotation or via clustering.
Instead, we will apply the Moran’s I statistic. Moran’s I is a measure
used to assess spatial autocorrelation in data, indicating whether
similar values of a feature (e.g., expression levels of a gene) are
clustered, dispersed, or random (<a href="https://ij-healthgeographics.biomedcentral.com/articles/10.1186/1476-072X-9-33" class="external-link">Jackson,
et al. 2010</a>). These correspond to Moran’s I values that are
positive, negative, or near zero, respectively.</p>
<p>Here, we can apply it to detect genes whose expression patterns
exhibit spatial structure, which may reflect region-specific, biological
function. That is, we anticipate that spatially variable genes will
exhibit region-specific expression. Let’s check that hypothesis by first
computing spatially variable genes and then assessing whether they are
differentially expressed across regions.</p>
<figure><img src="https://upload.wikimedia.org/wikipedia/commons/f/f0/Moran%27s_I_example.png" alt="Moran's I statistic quantifies spatial correlation." class="figure mx-auto d-block"><div class="figcaption">Moran’s I statistic quantifies spatial
correlation. <strong>Top Left:</strong> Checkerboard pattern results in
negative Moran’s I, indicating anti-correlation. <strong>Top
Right:</strong> Linear gradient shows a high positive Moran’s I,
reflecting a strong spatial gradient. <strong>Bottom Left:</strong>
Random pattern leads to a Moran’s I near zero, suggesting no significant
spatial autocorrelation. <strong>Bottom Right:</strong> ‘Ink blot’
pattern demonstrates positive autocorrelation, indicative of a clustered
or spreading pattern. Relationships are calculated using direct, equally
weighted neighbors, normalized for each cell.</div>
</figure><p>Image by
<a href="https://commons.wikimedia.org/wiki/File:Moran%27s_I_example.png" class="external-link">WikiNukalito</a>,
<a href="https://creativecommons.org/licenses/by-sa/4.0" class="external-link">CC BY-SA
4.0</a>, via Wikimedia Commons.</p>
<div class="section level3">
<h3 id="spatial-differential-expression-using-morans-i">Spatial Differential Expression Using Moran’s I<a class="anchor" aria-label="anchor" href="#spatial-differential-expression-using-morans-i"></a></h3>
<p>We identify the genes whose expression patterns exhibit clear spatial
structure using Moran’s I algorithm, as implemented in <a href="https://satijalab.org/seurat/reference/findspatiallyvariablefeatures" class="external-link">FindSpatiallyVariableFeatures</a>.
We have selected the top 1,000 genes, which should be sufficient to
identify brain regions. The following will take several minutes to
run.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">svg</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu">FindSpatiallyVariableFeatures</span><span class="op">(</span><span class="va">filter_st</span>, </span>
<span>                                assay            <span class="op">=</span> <span class="st">"SCT"</span>, </span>
<span>                                features         <span class="op">=</span> <span class="fu">VariableFeatures</span><span class="op">(</span><span class="va">filter_st</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">1000</span><span class="op">]</span>, </span>
<span>                                selection.method <span class="op">=</span> <span class="st">"moransi"</span><span class="op">)</span></span></code></pre>
</div>
<p>FindSpatiallyVariableFeatures returns a Seurat object, populated with
Moran’s I-derived results. Normally, we would use the <a href="https://satijalab.github.io/seurat-object/reference/VariableFeatures.html" class="external-link">SpatiallyVariableFeatures</a>
function to query those results. But, there is a bug in that function as
described <a href="https://github.com/satijalab/seurat/issues/7422" class="external-link">here</a>. So,
instead we will manually extract the top 100 ranked spatially variable
genes, along with their Moran’s I values and associated p-values:</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get and sort 'MoransI_observed' values</span></span>
<span><span class="va">morans_i_genes</span> <span class="op">&lt;-</span> <span class="va">svg</span><span class="op">@</span><span class="va">assays</span><span class="op">[[</span><span class="st">"SCT"</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">meta.features</span> <span class="op">%&gt;%</span></span>
<span>                    <span class="fu">rownames_to_column</span><span class="op">(</span><span class="st">"gene"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                    <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">MoransI_observed</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                    <span class="fu">slice_head</span><span class="op">(</span>n <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">morans_i_genes</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>    gene MoransI_observed MoransI_p.value moransi.spatially.variable
1    MBP        0.6824006    0.0009756098                       TRUE
2   PLP1        0.6075391    0.0009756098                       TRUE
3 MT-CO1        0.5627272    0.0009756098                       TRUE
4   MOBP        0.5447334    0.0009756098                       TRUE
5   GFAP        0.5407996    0.0009756098                       TRUE
6    CNP        0.5249332    0.0009756098                       TRUE
  moransi.spatially.variable.rank
1                               1
2                               2
3                               3
4                               4
5                               5
6                               6</code></pre>
</div>
</div>
</section><section id="correlation-of-region-specific-differentially-expressed-genes-and-spatially-variable-genes"><h2 class="section-heading">Correlation of Region-specific Differentially Expressed Genes and
Spatially Variable Genes<a class="anchor" aria-label="anchor" href="#correlation-of-region-specific-differentially-expressed-genes-and-spatially-variable-genes"></a>
</h2>
<hr class="half-width"><div class="section level3">
<h3 id="heatmap-of-differential-expression">Heatmap of Differential Expression<a class="anchor" aria-label="anchor" href="#heatmap-of-differential-expression"></a></h3>
<p>As a sanity check of both the region-specific differential expression
and the Moran’s I approaches, let’s check out hypothesis that spatially
variable genes are likely to show regional differential expression. To
do this, we will plot a heatmap of the <em>differential expression</em>
p-values for the top 100 spatially variable genes, organized by brain
region.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Merge the Moran's I values with the DE genes</span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu">merge</span><span class="op">(</span><span class="va">morans_i_genes</span>, <span class="va">de_genes</span>, all.x <span class="op">=</span> <span class="cn">TRUE</span>, by <span class="op">=</span> <span class="st">"gene"</span><span class="op">)</span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu">subset</span><span class="op">(</span><span class="va">df</span>, <span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># We will plot the -log2 pvalues. Compute this and adjust for taking log of 0.</span></span>
<span><span class="va">df</span><span class="op">$</span><span class="va">log_p_val_adj</span> <span class="op">&lt;-</span> <span class="op">-</span><span class="fu">log2</span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">p_val_adj</span><span class="op">)</span></span>
<span><span class="va">df</span><span class="op">$</span><span class="va">log_p_val_adj</span><span class="op">[</span><span class="fu">is.infinite</span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">log_p_val_adj</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu">max</span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">log_p_val_adj</span><span class="op">[</span><span class="op">!</span><span class="fu">is.infinite</span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">log_p_val_adj</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a matrix whose rows are the spatially variable genes (indicated by Moran's I),</span></span>
<span><span class="co"># whose columns are the clusters, and whose entries are the adjusted DE pvalue for the</span></span>
<span><span class="co"># corresponding gene and cluster.</span></span>
<span><span class="va">p_val_adj_matrix</span> <span class="op">&lt;-</span> <span class="va">df</span> <span class="op">%&gt;%</span></span>
<span>                       <span class="fu">select</span><span class="op">(</span><span class="va">gene</span>, <span class="va">cluster</span>,<span class="va">log_p_val_adj</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                       <span class="fu">pivot_wider</span><span class="op">(</span>names_from <span class="op">=</span> <span class="va">cluster</span>, values_from <span class="op">=</span> <span class="va">log_p_val_adj</span>, values_fill <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                       <span class="fu">column_to_rownames</span><span class="op">(</span><span class="st">"gene"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                       <span class="fu">as.matrix</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Order the regions according to their spatial organization from inner to outer layers</span></span>
<span><span class="va">p_val_adj_matrix</span> <span class="op">&lt;-</span> </span>
<span>  <span class="va">p_val_adj_matrix</span><span class="op">[</span>, <span class="fu">c</span><span class="op">(</span><span class="st">"WM"</span>, <span class="st">"Layer1"</span>, <span class="st">"Layer2"</span>, <span class="st">"Layer3"</span>, <span class="st">"Layer4"</span>, <span class="st">"Layer5"</span>, <span class="st">"Layer6"</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Create a heatmap of the DE p-values of spatially variable genes</span></span>
<span><span class="fu">Heatmap</span><span class="op">(</span><span class="va">p_val_adj_matrix</span>,</span>
<span>        column_title      <span class="op">=</span> <span class="st">"Heatmap of DE p-values of spatially variable genes"</span>,</span>
<span>        name              <span class="op">=</span> <span class="st">"DE -log2(p-values)"</span>, <span class="co"># Title for the heatmap legend</span></span>
<span>        row_title         <span class="op">=</span> <span class="st">"Spatially variable genes"</span>,</span>
<span>        cluster_rows      <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>        cluster_columns   <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        show_row_names    <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>        show_column_names <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>        show_row_dend     <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>        show_column_dend  <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/differential-expression-testing-rendered-heatmap-de-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The heatmap visualization reveals a key finding of our analysis:
genes displaying the highest Moran’s I values show distinct expression
patterns that align with specific brain regions identified through
expert annotations. This observation underscores the spatial correlation
of gene expression, highlighting its potential relevance in
understanding regional brain functions and pathologies.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul><li>Differential expression testing pinpoints genes with significant
expression variations across regions or clusters.</li>
<li>Moran’s I statistic reveals spatial autocorrelation in gene
expression, critical for examining spatially dependent biological
activities.</li>
<li>Moran’s I algorithm effectively identifies genes expressed in
anatomically distinct regions, as validated from the correlation
analysis with the DE genes from the annotated regions.</li>
</ul></div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</div>
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="deconvolve-cell-types-in-a-spot.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="final-exercise.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="deconvolve-cell-types-in-a-spot.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Deconvolution in
        </a>
        <a class="chapter-link float-end" href="final-exercise.html" rel="next">
          Next: Putting it all... 
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>
        
        <a href="https://github.com/smcclatchy/spatial-transcriptomics/edit/main/episodes/differential-expression-testing.Rmd" class="external-link">Edit on GitHub</a>
        
	
        | <a href="https://github.com/smcclatchy/spatial-transcriptomics/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/smcclatchy/spatial-transcriptomics/" class="external-link">Source</a></p>
				<p><a href="https://github.com/smcclatchy/spatial-transcriptomics/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:susan.mcclatchy@jax.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">
        
        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>
        
        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/epiverse-trace/sandpaper/tree/patch-renv-github-bug" class="external-link">sandpaper (0.16.4)</a>, <a href="https://github.com/carpentries/pegboard/tree/a32a7836d4455f407c3cafe8ab95edc636e5e919" class="external-link">pegboard (0.7.5)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.2" class="external-link">varnish (1.0.2)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://smcclatchy.github.io/spatial-transcriptomics/differential-expression-testing.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, The Carpentries, transcriptome, Visium, cancer",
  "name": "Differential Expression Testing",
  "creativeWorkStatus": "active",
  "url": "https://smcclatchy.github.io/spatial-transcriptomics/differential-expression-testing.html",
  "identifier": "https://smcclatchy.github.io/spatial-transcriptomics/differential-expression-testing.html",
  "dateCreated": "2024-03-01",
  "dateModified": "2024-06-14",
  "datePublished": "2024-06-14"
}

  </script><script>
		feather.replace();
	</script><!-- Matomo
    2022-11-07: we have gotten a notification that we have an overage for our
    tracking and I'm pretty sure this has to do with Workbench usage.
    Considering that I am not _currently_ using this tracking because I do not
    yet know how to access the data, I am turning this off for now.
  <script>
    var _paq = window._paq = window._paq || [];
    /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
    _paq.push(["setDocumentTitle", document.domain + "/" + document.title]);
    _paq.push(["setDomains", ["*.preview.carpentries.org","*.datacarpentry.github.io","*.datacarpentry.org","*.librarycarpentry.github.io","*.librarycarpentry.org","*.swcarpentry.github.io", "*.carpentries.github.io"]]);
    _paq.push(["setDoNotTrack", true]);
    _paq.push(["disableCookies"]);
    _paq.push(['trackPageView']);
    _paq.push(['enableLinkTracking']);
    (function() {
          var u="https://carpentries.matomo.cloud/";
          _paq.push(['setTrackerUrl', u+'matomo.php']);
          _paq.push(['setSiteId', '1']);
          var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
          g.async=true; g.src='https://cdn.matomo.cloud/carpentries.matomo.cloud/matomo.js'; s.parentNode.insertBefore(g,s);
        })();
  </script>
  End Matomo Code --></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

