<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Spatial Transcriptomics: Spatially Resolved Transcriptomics in Life Sciences Research</title><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="stylesheet" type="text/css" href="assets/styles.css"><script src="assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png"><link rel="manifest" href="site.webmanifest"><link rel="mask-icon" href="safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" content="#ffffff"></head><body>
    <header id="top" class="navbar navbar-expand-md navbar-light bg-white top-nav incubator"><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-6">
      <div class="large-logo">
        <img alt="Carpentries Incubator" src="assets/images/incubator-logo.svg"><abbr class="badge badge-light" title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught." style="background-color: #FF4955; border-radius: 5px">
          <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link" style="color: #000">
            <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
            Pre-Alpha
          </a>
          <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
        </abbr>
        
      </div>
    </div>
    <div class="selector-container">
      
      
      <div class="dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Learner View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='instructor/introduction.html';">Instructor View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl navbar-light bg-white bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Carpentries Incubator" src="assets/images/incubator-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Spatial Transcriptomics
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="search button" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Spatial Transcriptomics
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="reference.html#glossary">Glossary</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="profiles.html">Learner Profiles</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul></li>
      </ul></div>
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled><input class="form-control me-2 searchbox" type="search" placeholder="Search" aria-label="Search"><button class="btn btn-outline-success tablet-search-button" type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="search button"></i>
        </button>
      </fieldset></form>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Spatial Transcriptomics
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 0%" class="percentage">
    0%
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: 0%" aria-valuenow="0" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Learner View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="instructor/introduction.html">Instructor View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->
      
            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Setup</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        1. Spatially Resolved Transcriptomics in Life Sciences Research
        </span>
      </button>
    </div><!--/div.accordion-header-->
        
    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#spatial-transcriptomics-in-biomedical-research">Spatial transcriptomics in biomedical research</a></li>
<li><a href="#spatial-transcriptomics-technologies">Spatial transcriptomics technologies</a></li>
<li><a href="#x-genomics-visium-technology">10X Genomics Visium technology</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="data-and-study-design.html">2. Data and Study Design</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="data-preprocessing.html">3. Data Preprocessing</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="remove-low-quality-spots.html">4. Remove Low-quality Spots</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="apply-normalization-methods.html">5. Apply Normalization Methods</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="apply-feature-selection-methods.html">6. Apply Feature Selection Methods</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="apply-dimensionality-reduction-methods.html">7. Apply Dimensionality Reduction Methods</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="apply-clustering-algorithms.html">8. Apply Clustering Algorithms</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="deconvolve-cell-types-in-a-spot.html">9. Deconvolve Cell Types in a Spot</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush11">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading11">
        <a href="differential-expression-testing.html">10. Differential Expression Testing</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="reference.html#glossary">Glossary</a>
                      </li>
                      <li>
                        <a href="profiles.html">Learner Profiles</a>
                      </li>
                      <li><a href="reference.html">Reference</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width resources"><a href="aio.html">See all in one page</a>
            

            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">
            
            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="index.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="data-and-study-design.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="index.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Home
        </a>
        <a class="chapter-link float-end" href="data-and-study-design.html" rel="next">
          Next: Data and Study... 
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Spatially Resolved Transcriptomics in Life Sciences Research</h1>
        <p>Last updated on 2023-12-22 |
        
        <a href="https://github.com/smcclatchy/spatial-transcriptomics/edit/main/episodes/introduction.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
        
        
        
        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>What is spatial transcriptomics?</li>
<li>What research questions or problems can spatial transcriptomics
address?</li>
<li>How do the technologies work?</li>
<li>Which technology will we learn about in this lesson?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Describe why and how spatial transcriptomics can be used in
research.</li>
<li>Describe how spatial transcriptomics technology works.</li>
<li>Describe how spatial transcriptomics addresses the limitations of
single-cell or bulk RNA sequencing technologies.</li>
</ul></div>
</div>
</div>
</div>
</div>
<section id="spatial-transcriptomics-in-biomedical-research"><h2 class="section-heading">Spatial transcriptomics in biomedical research<a class="anchor" aria-label="anchor" href="#spatial-transcriptomics-in-biomedical-research"></a>
</h2>
<hr class="half-width"><p>Investigating the organization of cells and tissues is fundamental to
life sciences research. Tissues located in different regions of an organ
can possess diverse functions and cell types. These cells in turn are
influenced by varying tissue microenvironments, receiving and processing
distinct information from that microenvironment. Co-located cells can
communicate directly with one another through chemical and physical
signals, responding to these signals with changes in their own state.
The spatial organization and signals of cells in a tissue determine cell
and tissue function.</p>
<figure><img src="https://upload.wikimedia.org/wikipedia/commons/c/c3/Skeletal_muscle_-_cross_section%2C_nerve_bundle.jpg" alt="A cross-section of human skeletal muscle showing muscle cells and a nerve nearby. Stained with hematoxylin and eosin." class="figure mx-auto d-block"><figcaption>A cross-section of skeletal muscle tissue showing
muscle cells and a small nerve.</figcaption></figure><p><a href="https://commons.wikimedia.org/wiki/File:Skeletal_muscle_-_cross_section,_nerve_bundle.jpg" class="external-link">Department
of Histology, Jagiellonian University Medical College</a>
<a href="https://creativecommons.org/licenses/by-sa/3.0/deed.en" rel="license" class="external-link">CC
BY-SA 3.0 DEED</a></p>
<p>Describing spatial organization and cell signals, specifically gene
expression signals, is the focus of spatially resolved transcriptomics.
Spatial patterns of gene expression determine how genes are regulated
within a tissue system and how those tissues and their component cells
function. Spatial transcriptomic (ST) methods map cell position in a
tissue, clarifying the physical relationships between individual cells
and cellular structures. ST simultaneously measures gene expression,
delivering valuable information about cell phenotype, state, and cell
and tissue organization and function. The combination of cellular
expression and position sheds light on signals a cell sends or receives
through cell-to-cell interactions or from soluble signaling molecules
nearby. Spatial information localizes cell signaling while delivering
comprehensive gene expression profiling within tissues.</p>
<figure><img src="https://upload.wikimedia.org/wikipedia/commons/0/04/Notchccr.svg" alt="alt text for accessibility purposes" class="figure mx-auto d-block"><figcaption>Signaling between adjacent cells. The Notch protein
functions as a receptor for ligands that activate or inhibit such
receptors. Receptor-ligand interactions ground cell signaling and
communication, often requiring close proximity between cells.</figcaption></figure><p><a href="https://commons.wikimedia.org/wiki/File:Notchccr.svg" class="external-link">Fred
the Oyster</a> Public domain, via Wikimedia Commons
<a href="https://creativecommons.org/licenses/by-sa/4.0/" rel="license" class="external-link">CC
BY-SA 4.0 DEED</a></p>
<p>Spatial transcriptomics addresses a key obstacle in single-cell and
bulk RNA sequencing studies: the loss of spatial information when
tissues are dissociated. Spatial organization and structure determine
function in most tissues and organs, so capturing both spatial and
expression information is critical for understanding tissue function in
neuroscience, cancer biology, developmental biology, and most other
fields. Immunology is the only field that doesn’t depend on spatial
information to determine structure and function because immune cells
often circulate in liquid media like blood and lymph.</p>
</section><section id="spatial-transcriptomics-technologies"><h2 class="section-heading">Spatial transcriptomics technologies<a class="anchor" aria-label="anchor" href="#spatial-transcriptomics-technologies"></a>
</h2>
<hr class="half-width"><p>Spatial transcriptomics technologies broadly fall within two groups:
imaging-based and sequencing-based methods. Both imaging- and
sequence-based datasets are available through the <a href="https://biccn.org/" class="external-link">The BRAIN Initiative - Cell Census
Network</a>. Sequencing-based datasets are featured in the <a href="https://data.humancellatlas.org/" class="external-link">Human Cell Atlas</a>. These
technologies vary in ability to profile entire transcriptomes, deliver
single-cell resolution, and detect genes efficiently.</p>
<div class="section level3">
<h3 id="imaging-based-technologies">Imaging-based technologies<a class="anchor" aria-label="anchor" href="#imaging-based-technologies"></a></h3>
<p>Imaging-based technologies read transcriptomes in situ using
microscopy and feature single-cell or even single-molecule resolution.
They identify mRNA species through hybridization with fluorescent
probes. These probes are gene-specific in fluorescence in situ
hybridization (FISH). Current FISH methods employ multiple hybridization
rounds, with risk of error for each transcript growing exponentially
with each round. FISH methods are limited in the size of tissue that
they can profile and most accept fresh-frozen (FF) tissue only. They can
also be time-consuming and expensive due to microscopic imaging they
require. Since they target specific genes, they can only detect genes
that are in the probe set employed. They have high spatial resolution
though, even delivering single-molecule resolution in single-molecule
FISH (smFISH).</p>
<p>In situ sequencing amplifies and sequences mRNAs directly within a
block or section of fresh-frozen (FF) or formalin-fixed paraffin
embedded (FFPE) tissue. Probes profile one or two bases at a time using
different fluorophores, eventually revealing the identity of the mRNA
through imaging. In situ sequencing can accommodate larger tissue
sections than can FISH, though FISH methods are more efficient at
detecting mRNA of genes in the probe set. Like FISH, in situ sequencing
requires considerable imaging time on a microscope but delivers high
spatial resolution.</p>
</div>
<div class="section level3">
<h3 id="sequencing-based-technologies">Sequencing-based technologies<a class="anchor" aria-label="anchor" href="#sequencing-based-technologies"></a></h3>
<p>Sequencing-based methods capture, sequence, and count mRNA in situ
using next-generation sequencing while retaining positional information.
This is distinct from in situ sequencing because next-generation
sequencing is employed, not imaging. The main advantage of
sequenced-based methods is that they are unbiased in capturing RNA
because they don’t rely on predefined probesets targeting specific
genes. Sequencing-based methods retain spatial information through
laser-capture microdissection (LCM), microfluidics, or through ligation
of mRNAs to arrays of barcoded probes that record position. Since
sequencing-based technologies use probes that record position instead of
gene-specific probes, they can deliver transcriptome-wide profiling.</p>
<p>LCM-based methods employ lasers to cut a tissue slice or fuse tissue
to a membrane followed by sequencing of individual cells. LCM techniques
process tissue sections and individual cells for transcriptomic
profiling by isolating regions of interest. They are useful for
profiling transcriptomes as a first pass and for identifying RNA
isoforms, but their blunt approach to capturing spatial expression data
limits spatial resolution and requires many samples for sequencing.
Since they focus on regions of interest, it is often not possible to
obtain a picture of spatial expression across a whole tissue. LCM is an
older technology that has long been used with FFPE tissues. Modern
LCM-based approaches include Nanostring’s GeoMx DSP and STRP-seq.</p>
<p>Microfluidics places a chip with multiple barcode-containing channels
onto a tissue section followed by a second chip with channels
perpendicular to the first. The barcodes are then ligated to each other
to create an array of unique barcodes on the tissue. This “deterministic
barcoding” is employed in DBiT-seq. DBiT-seq can be used with FFPE
tissues. This approach is helpful to avoid diffusion of mRNA away from
capture areas, though a disadvantage is that cells often sit astride
multiple capture areas.</p>
<p>Other array-based methods capture mRNA with spatially-barcoded probes
and sequence them. They can profile larger tissue sections than can FISH
or in situ sequencing and they don’t rely on microscopic imaging, which
can be quite time consuming. Spatial resolution is lower, however.</p>
<table class="table"><colgroup><col width="27%"><col width="18%"><col width="18%"><col width="18%"><col width="18%"></colgroup><thead><tr class="header"><th>Technology</th>
<th align="center">Gene detection efficiency</th>
<th align="center">Transcriptome-wide profiling</th>
<th align="center">Spatial resolution</th>
<th align="center">Tissue area</th>
</tr></thead><tbody><tr class="odd"><td>FISH</td>
<td align="center">+</td>
<td align="center">-</td>
<td align="center">+</td>
<td align="center">-</td>
</tr><tr class="even"><td>In situ sequencing</td>
<td align="center">-</td>
<td align="center">-</td>
<td align="center">+</td>
<td align="center">+</td>
</tr><tr class="odd"><td>LCM-based</td>
<td align="center"></td>
<td align="center">+</td>
<td align="center">-</td>
<td align="center">-</td>
</tr><tr class="even"><td>Microfluidics</td>
<td align="center">-</td>
<td align="center">+</td>
<td align="center">-</td>
<td align="center">+</td>
</tr><tr class="odd"><td>Array-based</td>
<td align="center">-</td>
<td align="center">+</td>
<td align="center">-</td>
<td align="center">+</td>
</tr></tbody></table><p>Table 1. Relative strengths and weaknesses of spatial transcriptomics
technologies by general category.</p>
<div id="callout1" class="callout callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout<a class="anchor" aria-label="anchor" href="#callout1"></a>
</h3>
<div class="callout-content">
<p>The diversity in spatial transcriptomics technologies is enormous and
rapidly developing. This introduction to the technologies is intended to
help you navigate a complex technological landscape so that you can
learn more about existing technologies. It is not intended to be
comprehensive. If you would like to learn more about spatial
transcriptomics technologies, please see the
<a href="./reference.html"><strong>list of references located
here</strong></a>.</p>
</div>
</div>
</div>
<div id="discussion-which-technology-is-right-for-your-research" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="discussion-which-technology-is-right-for-your-research" class="callout-inner">
<h3 class="callout-title">Discussion: Which technology is right for your
research?<a class="anchor" aria-label="anchor" href="#discussion-which-technology-is-right-for-your-research"></a>
</h3>
<div class="callout-content">
<ol style="list-style-type: decimal"><li><p>Would an imaging-based or a sequencing-based solution be
preferable for your research? Why?</p></li>
<li><p>From the descriptions above, which technology do you think would
best suit your research? Even if your institution does not offer service
using a specific technology, describe which best suits your research and
why you think it’s best suited. Would you use fresh-frozen (FF) or
formalin-fixed tissues embedded in paraffin (FFPE)? Why is the
technology best suited to your research?</p></li>
</ol></div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">Show me the solution</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>varying - refer to strengths and weaknesses of each category, best
uses of different technologies and how they have been used in research
to date</p>
</div>
</div>
</div>
</div>
<p>In this lesson we will use data from positionally barcoded
arrays.</p>
<p><img src="https://upload.wikimedia.org/wikipedia/commons/1/14/Spatial_transcriptomics_ii.png" alt="A graphic showing printed spots on a glass slide that are identified by a barcode and that contain primers to capture messenger RNA from the tissue laid on top of them" class="figure"><a href="https://commons.wikimedia.org/wiki/User:Jasquatch" class="external-link">James
Chell</a>,
<a href="https://commons.wikimedia.org/wiki/File:Spatial_transcriptomics_ii.png" class="external-link">Spatial
transcriptomics ii</a>,
<a href="https://creativecommons.org/licenses/by-sa/4.0/legalcode" rel="license" class="external-link">CC
BY-SA 4.0</a></p>
</div>
</section><section id="x-genomics-visium-technology"><h2 class="section-heading">10X Genomics Visium technology<a class="anchor" aria-label="anchor" href="#x-genomics-visium-technology"></a>
</h2>
<hr class="half-width"><p>In this lesson we will use data from an array-based method called
Visium that is offered by 10X Genomics. Visium is an upgrade and
commercialization of the spatial transcriptomics method described in
2016 in <a href="https://doi.org/10.1126/science.aaf2403" class="external-link">Science,
353(6294)</a> and illustrated in general in the figure above. In brief,
thin tissue sections are placed atop spots printed with spatial
barcodes. When the tissue is permeabilized, mRNA is released from the
cells and hybridized to the spatial barcodes. Hybridized mRNA is reverse
transcribed to cDNA and then sequenced. Spatial transcriptomics combines
two key modes: histological imaging and gene expression profiling.
Histological imaging captures tissue morphology with standard staining
protocols while expression profiling is captured by sequencing spatially
barcoded cDNA.</p>
<p>Sequencing-based datasets have grown faster than have imaging-based
datasets, with Visium dominating published datasets. Unlike most other
sequencing-based technologies, Visium accommodates both FF or FFPE
tissue. Each spot provides average gene expression measurements from
between one to ten cells, approaching single-cell resolution. Average
gene expression measurements are combined with histological images that
couple molecular detail and tissue morphology and structure.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul><li>Spatial transcriptomics provides the location of individual cells
relative to neighboring cells and cell structures.</li>
<li>A cell’s location is useful data for describing its phenotype,
state, and cell and tissue function.</li>
<li>Spatial transcriptomics addresses a key obstacle in single-cell
studies: the loss of spatial information when tissues are
dissociated.</li>
<li>The main goal of spatial transcriptomics studies is to integrate
expression with spatial information.</li>
</ul></div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="index.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="data-and-study-design.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="index.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Home
        </a>
        <a class="chapter-link float-end" href="data-and-study-design.html" rel="next">
          Next: Data and Study... 
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>
        
        <a href="https://github.com/smcclatchy/spatial-transcriptomics/edit/main/episodes/introduction.Rmd" class="external-link">Edit on GitHub</a>
        
	
        | <a href="https://github.com/smcclatchy/spatial-transcriptomics/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/smcclatchy/spatial-transcriptomics/" class="external-link">Source</a></p>
				<p><a href="https://github.com/smcclatchy/spatial-transcriptomics/blob/main/CITATION" class="external-link">Cite</a> | <a href="mailto:susan.mcclatchy@jax.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">
        
        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>
        
        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.2" class="external-link">sandpaper (0.16.2)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.3" class="external-link">pegboard (0.7.3)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.1" class="external-link">varnish (1.0.1)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://smcclatchy.github.io/spatial-transcriptomics/introduction.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, The Carpentries, transcriptome, Visium, cancer",
  "name": "Spatially Resolved Transcriptomics in Life Sciences Research",
  "creativeWorkStatus": "active",
  "url": "https://smcclatchy.github.io/spatial-transcriptomics/introduction.html",
  "identifier": "https://smcclatchy.github.io/spatial-transcriptomics/introduction.html",
  "dateCreated": "2024-02-02",
  "dateModified": "2023-12-22",
  "datePublished": "2024-02-06"
}

  </script><script>
		feather.replace();
	</script><!-- Matomo
    2022-11-07: we have gotten a notification that we have an overage for our
    tracking and I'm pretty sure this has to do with Workbench usage.
    Considering that I am not _currently_ using this tracking because I do not
    yet know how to access the data, I am turning this off for now.
  <script>
    var _paq = window._paq = window._paq || [];
    /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
    _paq.push(["setDocumentTitle", document.domain + "/" + document.title]);
    _paq.push(["setDomains", ["*.preview.carpentries.org","*.datacarpentry.github.io","*.datacarpentry.org","*.librarycarpentry.github.io","*.librarycarpentry.org","*.swcarpentry.github.io", "*.carpentries.github.io"]]);
    _paq.push(["setDoNotTrack", true]);
    _paq.push(["disableCookies"]);
    _paq.push(['trackPageView']);
    _paq.push(['enableLinkTracking']);
    (function() {
          var u="https://carpentries.matomo.cloud/";
          _paq.push(['setTrackerUrl', u+'matomo.php']);
          _paq.push(['setSiteId', '1']);
          var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
          g.async=true; g.src='https://cdn.matomo.cloud/carpentries.matomo.cloud/matomo.js'; s.parentNode.insertBefore(g,s);
        })();
  </script>
  End Matomo Code --></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

