<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Spatial Transcriptomics: Deconvolution in Spatial Transcriptomics</title><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="stylesheet" type="text/css" href="../assets/styles.css"><script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="manifest" href="../site.webmanifest"><link rel="mask-icon" href="../safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" content="#ffffff"></head><body>
    <header id="top" class="navbar navbar-expand-md navbar-light bg-white top-nav incubator"><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-6">
      <div class="large-logo">
        <img alt="Carpentries Incubator" src="../assets/images/incubator-logo.svg"><abbr class="badge badge-light" title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught." style="background-color: #FF4955; border-radius: 5px">
          <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link" style="color: #000">
            <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
            Pre-Alpha
          </a>
          <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
        </abbr>
        
      </div>
    </div>
    <div class="selector-container">
      
      
      <div class="dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='../deconvolve-cell-types-in-a-spot.html';">Learner View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl navbar-light bg-white bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Carpentries Incubator" src="../assets/images/incubator-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Spatial Transcriptomics
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Spatial Transcriptomics
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><hr><li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a class="btn btn-primary" href="../aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Spatial Transcriptomics
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 69%" class="percentage">
    69%
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: 69%" aria-valuenow="69" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../deconvolve-cell-types-in-a-spot.html">Learner View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->
      
            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="introduction.html">1. Spatially Resolved Transcriptomics in Life Sciences Research</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="data-and-study-design.html">2. Data and Study Design</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="data-preprocessing.html">3. Data Preprocessing</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="remove-low-quality-spots.html">4. Remove Low-quality Spots</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="apply-normalization-methods.html">5. Normalization in Spatial Transcriptomics</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="feature-selection-dimensionality-reduction-clustering.html">6. Feature Selection, Dimensionality Reduction, and Spot Clustering</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        7. Deconvolution in Spatial Transcriptomics
        </span>
      </button>
    </div><!--/div.accordion-header-->
        
    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#deconvolution-in-spatial-transcriptomics">Deconvolution in Spatial Transcriptomics</a></li>
<li><a href="#why-deconvolution-is-needed">Why Deconvolution is Needed</a></li>
<li><a href="#deconvolution-with-rctd">Deconvolution with RCTD</a></li>
<li><a href="#loading-single-cell-rna-seq-data">Loading Single Cell RNA-Seq Data</a></li>
<li><a href="#deconvolution-with-rctd-1">Deconvolution with RCTD</a></li>
<li><a href="#running-deconvolution-on-brain-samples">Running Deconvolution on Brain Samples</a></li>
<li><a href="#interpreting-deconvolution-results">Interpreting Deconvolution Results</a></li>
<li><a href="#summary">Summary</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="differential-expression-testing.html">8. Differential Expression Testing</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr><li><a class="dropdown-item" href="reference.html">Reference</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width resources"><a href="../instructor/aio.html">See all in one page</a>
            

            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">
            
            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/feature-selection-dimensionality-reduction-clustering.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/differential-expression-testing.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/feature-selection-dimensionality-reduction-clustering.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Feature Selection,
        </a>
        <a class="chapter-link float-end" href="../instructor/differential-expression-testing.html" rel="next">
          Next: Differential... 
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Deconvolution in Spatial Transcriptomics</h1>
        <p>Last updated on 2024-06-06 |
        
        <a href="https://github.com/smcclatchy/spatial-transcriptomics/edit/main/episodes/deconvolve-cell-types-in-a-spot.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
        
        
        
        <p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 70 minutes</p>
        
        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>How does deconvolution enhance the analysis of spatial
transcriptomics data?</li>
<li>What are the steps to integrate single-cell RNA-seq data for
deconvolution in spatial transcriptomics?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Perform deconvolution to quantify different cell types in spatial
transcriptomics spots using single-cell RNA-seq data.</li>
<li>Understand the process of integrating single-cell RNA-seq data with
spatial transcriptomics data.</li>
</ul></div>
</div>
</div>
</div>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in readRDS(file_info$files[i]): unknown input format</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'filter_st' not found</code></pre>
</div>
<section id="deconvolution-in-spatial-transcriptomics"><h2 class="section-heading">Deconvolution in Spatial Transcriptomics<a class="anchor" aria-label="anchor" href="#deconvolution-in-spatial-transcriptomics"></a>
</h2>
<hr class="half-width"><p>Spatial transcriptomics (ST) provides valuable insights into the
spatial distribution of gene expression within tissue sections. However,
each spatial spot in an ST experiment generally contains multiple cells,
leading to mixed gene expression signals. Deconvolution is needed to
resolve these mixed signals into individual cell type contributions,
allowing for a more accurate interpretation of the spatial gene
expression data.</p>
</section><section id="why-deconvolution-is-needed"><h2 class="section-heading">Why Deconvolution is Needed<a class="anchor" aria-label="anchor" href="#why-deconvolution-is-needed"></a>
</h2>
<hr class="half-width"><p>Mixed cell populations in each spot of an ST dataset can lead to
complex, mixed gene expression profiles. Deconvolution helps quantify
the different cell types within each spot, enabling more precise
biological insights and downstream analyses such as cell type-specific
expression patterns and interactions.</p>
</section><section id="deconvolution-with-rctd"><h2 class="section-heading">Deconvolution with RCTD<a class="anchor" aria-label="anchor" href="#deconvolution-with-rctd"></a>
</h2>
<hr class="half-width"><p>RCTD (Robust Cell Type Decomposition) (Cable, D. M., et al., Nature
Biotechnology, 2021) is a popular deconvolution algorithm designed to
handle the complexities of mixed cell populations in spatial
transcriptomics data. The algorithm uses single-cell RNA sequencing
(scRNA-seq) data as a reference to deconvolute the spatial
transcriptomics data, estimating the proportions of different cell types
in each spatial spot. The RCTD algorithm models the observed gene
expression in each spatial spot as a mixture of the gene expression
profiles of different cell types and estimates the proportion of each
cell type in each spot using a non-negative least squares (NNLS)
approach. RCTD can operate in two modes: in “doublet” mode it fits at
most two cell types per spot, in “full” mode it fits potentially all
cell types in the reference per spot, and in “multi” mode it again fits
more than two cell types per spot by extending the “doublet” approach.
Here, we will use “full” mode.</p>
</section><section id="loading-single-cell-rna-seq-data"><h2 class="section-heading">Loading Single Cell RNA-Seq Data<a class="anchor" aria-label="anchor" href="#loading-single-cell-rna-seq-data"></a>
</h2>
<hr class="half-width"><p>First, we load the single-cell RNA-seq data that will serve as a
reference for deconvolution. This data is essential for mapping the gene
expression profiles from the ST data to specific cell types.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load single-cell RNA-seq data</span></span>
<span><span class="va">sc.counts</span> <span class="op">&lt;-</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="fu">fread</span><span class="op">(</span><span class="st">"data/scRNA-seq/sc_counts.tsv.gz"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: To read gz and bz2 files directly, fread() requires 'R.utils' package which cannot be found. Please install 'R.utils' using 'install.packages('R.utils')'.</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rownames</span><span class="op">(</span><span class="va">sc.counts</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sc.counts</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in eval(expr, envir, enclos): object 'sc.counts' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.counts</span> <span class="op">&lt;-</span> <span class="va">sc.counts</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in eval(expr, envir, enclos): object 'sc.counts' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load cell type annotations</span></span>
<span><span class="va">sc.metadata</span> <span class="op">&lt;-</span> <span class="fu">read.table</span><span class="op">(</span><span class="st">"data/scRNA-seq/sc_cell_types.tsv"</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, quote <span class="op">=</span> <span class="st">""</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">sc.cell.types</span> <span class="op">&lt;-</span> <span class="fu">setNames</span><span class="op">(</span><span class="fu">factor</span><span class="op">(</span><span class="va">sc.metadata</span><span class="op">$</span><span class="va">Value</span><span class="op">)</span>, <span class="va">sc.metadata</span><span class="op">$</span><span class="va">Name</span><span class="op">)</span></span>
<span><span class="va">shared.cells</span> <span class="op">&lt;-</span> <span class="fu">intersect</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">sc.counts</span><span class="op">)</span>, <span class="fu">names</span><span class="op">(</span><span class="va">sc.cell.types</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in eval(expr, envir, enclos): object 'sc.counts' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.cell.types</span> <span class="op">&lt;-</span> <span class="va">sc.cell.types</span><span class="op">[</span><span class="va">shared.cells</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in eval(expr, envir, enclos): object 'shared.cells' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.counts</span> <span class="op">&lt;-</span> <span class="va">sc.counts</span><span class="op">[</span>, <span class="va">shared.cells</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in eval(expr, envir, enclos): object 'sc.counts' not found</code></pre>
</div>
</section><section id="deconvolution-with-rctd-1"><h2 class="section-heading">Deconvolution with RCTD<a class="anchor" aria-label="anchor" href="#deconvolution-with-rctd-1"></a>
</h2>
<hr class="half-width"><p>RCTD (Robust Cell Type Decomposition) is a powerful tool that uses
scRNA-seq reference data to deconvolute the spatial transcriptomics
data, thereby quantifying the cell types present in each spatial
spot.</p>
<p>First, we will create the reference object encapsulating the
scRNA-seq data.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">reference</span> <span class="op">&lt;-</span> <span class="fu">Reference</span><span class="op">(</span><span class="va">sc.counts</span>, <span class="va">sc.cell.types</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in eval(expr, envir, enclos): object 'sc.counts' not found</code></pre>
</div>
<p>Let’s write a wrapper function that performs RCTD deconvolution. This
will facilitate you running RCTD on other samples within this
dataset.</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">run.rctd</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">reference</span>, <span class="va">st.obj</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="co"># Get raw ST counts</span></span>
<span>  <span class="va">st.counts</span> <span class="op">&lt;-</span> <span class="fu">GetAssayData</span><span class="op">(</span><span class="va">st.obj</span>, assay<span class="op">=</span><span class="st">"Spatial"</span>, layer<span class="op">=</span><span class="st">"counts"</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Get the spot coordinates</span></span>
<span>  <span class="va">st.coords</span> <span class="op">&lt;-</span> <span class="va">st.obj</span><span class="op">[[</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>, <span class="fu">c</span><span class="op">(</span><span class="st">"array_col"</span>, <span class="st">"array_row"</span><span class="op">)</span><span class="op">]</span></span>
<span>  <span class="fu">colnames</span><span class="op">(</span><span class="va">st.coords</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">"x"</span>,<span class="st">"y"</span><span class="op">)</span></span>
<span>    </span>
<span>  <span class="co"># Create the RCTD 'puck', representing the ST data</span></span>
<span>  <span class="va">puck</span> <span class="op">&lt;-</span> <span class="fu">SpatialRNA</span><span class="op">(</span><span class="va">st.coords</span>, <span class="va">st.counts</span><span class="op">)</span></span>
<span></span>
<span>  <span class="va">myRCTD</span> <span class="op">&lt;-</span> <span class="fu">create.RCTD</span><span class="op">(</span><span class="va">puck</span>, <span class="va">reference</span>, max_cores <span class="op">=</span> <span class="fl">1</span>, keep_reference <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span>  <span class="co"># Run deconvolution -- note that we are using 'full' mode to devolve a spot into </span></span>
<span>  <span class="co"># (potentially) all available cell types.</span></span>
<span>  <span class="va">myRCTD</span> <span class="op">&lt;-</span> <span class="fu">suppressWarnings</span><span class="op">(</span><span class="fu">run.RCTD</span><span class="op">(</span><span class="va">myRCTD</span>, doublet_mode <span class="op">=</span> <span class="st">'full'</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">myRCTD</span></span>
<span><span class="op">}</span></span></code></pre>
</div>
</section><section id="running-deconvolution-on-brain-samples"><h2 class="section-heading">Running Deconvolution on Brain Samples<a class="anchor" aria-label="anchor" href="#running-deconvolution-on-brain-samples"></a>
</h2>
<hr class="half-width"><p>We apply the RCTD wrapper to our spatial transcriptomics data to
deconvolute the spots and quantify the cell types. This may take ~10
minutes. If you prefer, you can load the precomputed results
directly.</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">load.precomputed.results</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">rds.file</span> <span class="op">&lt;-</span> <span class="fu">paste0</span><span class="op">(</span><span class="st">"data/rctd-sample-1.rds"</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="va">load.precomputed.results</span> <span class="op">||</span> <span class="op">!</span><span class="fu">file.exists</span><span class="op">(</span><span class="va">rds.file</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">result_1</span> <span class="op">&lt;-</span> <span class="fu">run.rctd</span><span class="op">(</span><span class="va">reference</span>, <span class="va">filter_st</span><span class="op">)</span></span>
<span>  <span class="co"># The RCTD file is large. To save space, we will remove the reference counts.</span></span>
<span>  <span class="va">result_1</span> <span class="op">&lt;-</span> <span class="fu">remove.RCTD.reference.counts</span><span class="op">(</span><span class="va">result_1</span><span class="op">)</span></span>
<span>  <span class="fu">saveRDS</span><span class="op">(</span><span class="va">result_1</span>, <span class="va">rds.file</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>  <span class="va">result_1</span> <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span><span class="va">rds.file</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in readRDS(rds.file): unknown input format</code></pre>
</div>
</section><section id="interpreting-deconvolution-results"><h2 class="section-heading">Interpreting Deconvolution Results<a class="anchor" aria-label="anchor" href="#interpreting-deconvolution-results"></a>
</h2>
<hr class="half-width"><p>The deconvolution process outputs the proportion of different cell
types in each spatial spot. Let’s write a utility function to extract
these proportions from the RCTD output. This function is also defined in
code/spatial_utils.R.</p>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">format.rctd.output_</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">rctd</span>, <span class="va">normalize</span> <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">barcodes</span> <span class="op">&lt;-</span> <span class="fu">colnames</span><span class="op">(</span><span class="va">rctd</span><span class="op">@</span><span class="va">spatialRNA</span><span class="op">@</span><span class="va">counts</span><span class="op">)</span></span>
<span>  <span class="va">weights</span> <span class="op">&lt;-</span> <span class="va">rctd</span><span class="op">@</span><span class="va">results</span><span class="op">$</span><span class="va">weights</span></span>
<span>  <span class="kw">if</span><span class="op">(</span><span class="va">normalize</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">weights</span> <span class="op">&lt;-</span> <span class="fu">normalize_weights</span><span class="op">(</span><span class="va">weights</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="va">df</span> <span class="op">&lt;-</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="va">weights</span><span class="op">)</span></span>
<span>  <span class="va">df</span><span class="op">$</span><span class="va">x</span> <span class="op">&lt;-</span> <span class="va">rctd</span><span class="op">@</span><span class="va">spatialRNA</span><span class="op">@</span><span class="va">coords</span><span class="op">$</span><span class="va">x</span></span>
<span>  <span class="va">df</span><span class="op">$</span><span class="va">y</span> <span class="op">&lt;-</span> <span class="va">rctd</span><span class="op">@</span><span class="va">spatialRNA</span><span class="op">@</span><span class="va">coords</span><span class="op">$</span><span class="va">y</span></span>
<span>  <span class="va">df</span></span>
<span><span class="op">}</span></span></code></pre>
</div>
<p>And now let’s see the predicted proportions in our sample:</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">##props &lt;- format.rctd.output_(result_1, normalize = FALSE)</span></span>
<span><span class="co">##head(props)</span></span></code></pre>
</div>
<p>Notice that the proportions don’t sum exactly to one.</p>
<div class="codewrapper sourceCode" id="cb22">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">##head(rowSums(select(props, -c(x,y))))</span></span></code></pre>
</div>
<p>Let’s classify the spot according to the layer type with highest
proportion</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">##props$classification &lt;- apply(select(props, -c(x,y)), 1, function(row) names(row)[which.max(row)])</span></span></code></pre>
</div>
<p>Let’s add the deconvolution results to our Seurat object.</p>
<div class="codewrapper sourceCode" id="cb24">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filter_st</span> <span class="op">&lt;-</span> <span class="fu">AddMetaData</span><span class="op">(</span>object <span class="op">=</span> <span class="va">filter_st</span>, metadata <span class="op">=</span>  <span class="fu">select</span><span class="op">(</span><span class="va">props</span>, <span class="op">-</span><span class="fu">c</span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in eval(expr, envir, enclos): object 'filter_st' not found</code></pre>
</div>
<p>We can now visualize the predicted layer classifications and compare
them alongside the ground truth annotations that we saw previously.</p>
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">##g1 &lt;- SpatialDimPlotColorSafe(filter_st[, !is.na(filter_st[[]]$classification)], "classification")</span></span>
<span><span class="co">##g2 &lt;- SpatialDimPlotColorSafe(filter_st[, !is.na(filter_st[[]]$layer_guess)], "layer_guess")</span></span>
<span><span class="co">##g1 + g2</span></span></code></pre>
</div>
<p>To be more quantitative, we can compute a confusion matrix comparing
the predicted and observed layers.</p>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">##df &lt;- as.data.frame(table(filter_st[[]]$layer_guess, filter_st[[]]$classification))</span></span>
<span><span class="co">##colnames(df) &lt;- c("Annotation", "Prediction", "Freq")</span></span>
<span><span class="co">##df$Annotation &lt;- factor(df$Annotation)</span></span>
<span><span class="co">##df$Prediction &lt;- factor(df$Prediction)</span></span>
<span></span>
<span><span class="co">##g &lt;- ggplot(data = df, aes(x = Annotation, y = Prediction, fill = Freq)) + geom_tile()</span></span>
<span><span class="co">##g &lt;- g + theme(text = element_text(size = 20))</span></span>
<span><span class="co">##g</span></span></code></pre>
</div>
<p>Note that there is a fairly strong correlation between the predicted
and observed layers, particularly for the pairs Oligodendrocytes and WM
(White Matter), L4 and Layer 4, and L2-3 and Layer 3.</p>
</section><section id="summary"><h2 class="section-heading">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<hr class="half-width"><p>Incorporating deconvolution into the spatial transcriptomics workflow
enhances the ability to quantify the different cell types in each
spatial spot, providing a richer and more detailed understanding of the
tissue architecture. By following these steps, researchers can ensure
that their spatial transcriptomics data is accurately deconvoluted,
leading to more robust and insightful analyses.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul><li>Deconvolution enhances spatial transcriptomics by quantifying the
different cell types within spatial spots.</li>
<li>Integrating single-cell RNA-seq data with spatial transcriptomics
data is essential for accurate deconvolution.</li>
<li>The RCTD method is effective for quantifying the proportion of
different cell types in spatial transcriptomics data.</li>
</ul></div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/feature-selection-dimensionality-reduction-clustering.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/differential-expression-testing.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/feature-selection-dimensionality-reduction-clustering.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Feature Selection,
        </a>
        <a class="chapter-link float-end" href="../instructor/differential-expression-testing.html" rel="next">
          Next: Differential... 
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>
        
        <a href="https://github.com/smcclatchy/spatial-transcriptomics/edit/main/episodes/deconvolve-cell-types-in-a-spot.Rmd" class="external-link">Edit on GitHub</a>
        
	
        | <a href="https://github.com/smcclatchy/spatial-transcriptomics/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/smcclatchy/spatial-transcriptomics/" class="external-link">Source</a></p>
				<p><a href="https://github.com/smcclatchy/spatial-transcriptomics/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:susan.mcclatchy@jax.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">
        
        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>
        
        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/epiverse-trace/sandpaper/tree/patch-renv-github-bug" class="external-link">sandpaper (0.16.4)</a>, <a href="https://github.com/carpentries/pegboard/tree/a32a7836d4455f407c3cafe8ab95edc636e5e919" class="external-link">pegboard (0.7.5)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.2" class="external-link">varnish (1.0.2)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://smcclatchy.github.io/spatial-transcriptomics/instructor/deconvolve-cell-types-in-a-spot.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, The Carpentries, transcriptome, Visium, cancer",
  "name": "Deconvolution in Spatial Transcriptomics",
  "creativeWorkStatus": "active",
  "url": "https://smcclatchy.github.io/spatial-transcriptomics/instructor/deconvolve-cell-types-in-a-spot.html",
  "identifier": "https://smcclatchy.github.io/spatial-transcriptomics/instructor/deconvolve-cell-types-in-a-spot.html",
  "dateCreated": "2024-03-01",
  "dateModified": "2024-06-06",
  "datePublished": "2024-06-06"
}

  </script><script>
		feather.replace();
	</script><!-- Matomo
    2022-11-07: we have gotten a notification that we have an overage for our
    tracking and I'm pretty sure this has to do with Workbench usage.
    Considering that I am not _currently_ using this tracking because I do not
    yet know how to access the data, I am turning this off for now.
  <script>
    var _paq = window._paq = window._paq || [];
    /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
    _paq.push(["setDocumentTitle", document.domain + "/" + document.title]);
    _paq.push(["setDomains", ["*.preview.carpentries.org","*.datacarpentry.github.io","*.datacarpentry.org","*.librarycarpentry.github.io","*.librarycarpentry.org","*.swcarpentry.github.io", "*.carpentries.github.io"]]);
    _paq.push(["setDoNotTrack", true]);
    _paq.push(["disableCookies"]);
    _paq.push(['trackPageView']);
    _paq.push(['enableLinkTracking']);
    (function() {
          var u="https://carpentries.matomo.cloud/";
          _paq.push(['setTrackerUrl', u+'matomo.php']);
          _paq.push(['setSiteId', '1']);
          var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
          g.async=true; g.src='https://cdn.matomo.cloud/carpentries.matomo.cloud/matomo.js'; s.parentNode.insertBefore(g,s);
        })();
  </script>
  End Matomo Code --></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

