<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<title>Spatial Transcriptomics: All Images</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" type="text/css" href="../assets/styles.css">
<script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="manifest" href="../site.webmanifest">
<link rel="mask-icon" href="../safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" content="#ffffff">
</head>
<body>
    <header id="top" class="navbar navbar-expand-md navbar-light bg-white top-nav incubator"><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-6">
      <div class="large-logo">
        <img alt="Carpentries Incubator" src="../assets/images/incubator-logo.svg"><abbr class="badge badge-light" title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught." style="background-color: #FF4955; border-radius: 5px">
          <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link" style="color: #000">
            <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
            Pre-Alpha
          </a>
          <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
        </abbr>
        
      </div>
    </div>
    <div class="selector-container">
      
      
      <div class="dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
<li><button class="dropdown-item" type="button" onclick="window.location.href='../images.html';">Learner View</button></li>
        </ul>
</div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl navbar-light bg-white bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Carpentries Incubator" src="../assets/images/incubator-logo-sm.svg">
</div>
    <div class="lesson-title-md">
      Spatial Transcriptomics
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
<li class="nav-item">
          <span class="lesson-title">
            Spatial Transcriptomics
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown">
<hr>
<li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul>
</li>
      </ul>
</div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a class="btn btn-primary" href="../aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div>
<!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Spatial Transcriptomics
</div>

<aside class="col-md-12 lesson-progress"><div style="width: %" class="percentage">
    %
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: %" aria-valuenow="" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../images.html">Learner View</a>
                      </div>
                    </div>
                  </div>
<!--/div.accordion-item-->
                </div>
<!--/div.accordion-flush-->
              </div>
<!--div.sidenav-view-selector -->
            </div>
<!--/div.col -->
      
            <hr>
</div>
<!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="introduction.html">1. Spatially Resolved Transcriptomics in Life Sciences Research</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="data-and-study-design.html">2. Data and Study Design</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="data-preprocessing.html">3. Data Preprocessing</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="remove-low-quality-spots.html">4. Remove Low-quality Spots</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="apply-normalization-methods.html">5. Normalization in Spatial Transcriptomics</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="feature-selection-dimensionality-reduction-clustering.html">6. Feature Selection, Dimensionality Reduction, and Spot Clustering</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="deconvolve-cell-types-in-a-spot.html">7. Deconvolution in Spatial Transcriptomics</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="differential-expression-testing.html">8. Differential Expression Testing</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width">
<div class="accordion accordion-flush resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul>
<li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr>
<li><a class="dropdown-item" href="reference.html">Reference</a></li>
                    </ul>
</div>
                </div>
              </div>
            </div>
            <hr class="half-width resources">
<a href="../instructor/aio.html">See all in one page</a>
            

            <hr class="d-none d-sm-block d-md-none">
<div class="d-grid gap-1">
            
            </div>
          </div>
<!-- /div.accordion -->
        </div>
<!-- /div.sidebar-inner -->
      </nav>
</div>
<!-- /div.sidebar -->
  </div>
<!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-extra.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <main id="main-content" class="main-content"><div class="container lesson-content">
        
        
<section id="introduction"><h2 class="section-heading"><a href="introduction.html">Spatially Resolved Transcriptomics in Life Sciences Research</a></h2>
<hr class="half-width">
<h3 id="introduction-figure-1">Figure 1</h3>
<p aria-hidden="true">Image 1 of 1: ‘A cross-section of human skeletal muscle showing muscle cells and a nerve nearby. Stained with hematoxylin and eosin.’</p>
<figure><img src="https://upload.wikimedia.org/wikipedia/commons/c/c3/Skeletal_muscle_-_cross_section%2C_nerve_bundle.jpg" alt="A cross-section of human skeletal muscle showing muscle cells and a nerve nearby. Stained with hematoxylin and eosin." class="figure mx-auto d-block"><div class="figcaption">A cross-section of skeletal muscle tissue
showing muscle cells and a small nerve.</div>
</figure><hr>
<h3 id="introduction-figure-2">Figure 2</h3>
<p aria-hidden="true">Image 1 of 1: ‘alt text for accessibility purposes’</p>
<figure><img src="https://upload.wikimedia.org/wikipedia/commons/0/04/Notchccr.svg" alt="alt text for accessibility purposes" class="figure mx-auto d-block"><div class="figcaption">Signaling between adjacent cells. The Notch
protein functions as a receptor for ligands that activate or inhibit
such receptors. Receptor-ligand interactions ground cell signaling and
communication, often requiring close proximity between cells.</div>
</figure><hr>
<h3 id="introduction-figure-3">Figure 3</h3>
<p aria-hidden="true">Image 1 of 1: ‘a general schematic showing fluorescence in situ hybridization’</p>
<figure><img src="../fig/FISH_Overview.png" alt="a general schematic showing fluorescence in situ hybridization" class="figure mx-auto d-block"><div class="figcaption">Overview of fluorescence in situ hybridization
(FISH).</div>
</figure><hr>
<h3 id="introduction-figure-4">Figure 4</h3>
<p aria-hidden="true">Image 1 of 1: ‘alt text for accessibility purposes’</p>
<figure><img src="https://upload.wikimedia.org/wikipedia/commons/thumb/a/ad/MERFISH_Diagram.png/1024px-MERFISH_Diagram.png" alt="alt text for accessibility purposes" class="figure mx-auto d-block"><div class="figcaption">Schematic representation of multiplexed
error-robust FISH (MERFISH). Binary codes assigned to mRNA species of
interest, where “1” represents a short fluorescent DNA probe. b,
Consecutive hybridization rounds, bleaching in between is implied, but
not shown for clarity. At the end of the sixth round, it is possible to
tell different mRNAs apart due to the decoded combinations of “1” and
“0”.</div>
</figure><hr>
<h3 id="introduction-figure-5">Figure 5</h3>
<p aria-hidden="true">Image 1 of 1: ‘a general schematic showing in situ sequencing’</p>
<p><img src="../fig/ISS_Overview.png" alt="a general schematic showing in situ sequencing" class="figure"><a href="https://commons.wikimedia.org/wiki/File:Spatial_Transcriptomics_Overview.png" class="external-link">
Adapted from Spatial Transcriptomics Overview by SlifertheRyeDragon.</a>
Image created with Biorender.com. Public domain, via Wikimedia Commons
<a href="https://creativecommons.org/licenses/by-sa/4.0/" rel="license" class="external-link">CC
BY-SA 4.0 DEED</a></p>
<hr>
<h3 id="introduction-figure-6">Figure 6</h3>
<p aria-hidden="true">Image 1 of 1: ‘a general schematic showing laser-capture microdissection ’</p>
<p><img src="../fig/LCM_Overview.png" alt="a general schematic showing laser-capture microdissection " class="figure"><a href="https://commons.wikimedia.org/wiki/File:Spatial_Transcriptomics_Overview.png" class="external-link">
Adapted from Spatial Transcriptomics Overview by SlifertheRyeDragon.</a>
Image created with Biorender.com. Public domain, via Wikimedia Commons
<a href="https://creativecommons.org/licenses/by-sa/4.0/" rel="license" class="external-link">CC
BY-SA 4.0 DEED</a></p>
<hr>
<h3 id="introduction-figure-7">Figure 7</h3>
<p aria-hidden="true">Image 1 of 1: ‘a general schematic showing a microfluidics workflow with DBit-seq on formalin-fixed paraffin embedded (FFPE) tissue’</p>
<figure><img src="../fig/Workflow-of-DBiT-seq-on-FFPE-samples-a-Scheme-of-DBiT-seq-on-FFPE-samples.png" alt="a general schematic showing a microfluidics workflow with DBit-seq on formalin-fixed paraffin embedded (FFPE) tissue" class="figure mx-auto d-block"><div class="figcaption">Workflow schematic of DBiT-seq on FFPE
samples.</div>
</figure><hr>
<h3 id="introduction-figure-8">Figure 8</h3>
<p aria-hidden="true">Image 1 of 1: ‘FFPE workflow’</p>
<a href="https://www.researchgate.net/figure/Workflow-of-DBiT-seq-on-FFPE-samples-a-Scheme-of-DBiT-seq-on-FFPE-samples_fig2_346261659" class="external-link"><img src="https://www.researchgate.net/publication/346261659/figure/fig2/AS:961767006040117@1606314531966/Workflow-of-DBiT-seq-on-FFPE-samples-a-Scheme-of-DBiT-seq-on-FFPE-samples.png" alt="FFPE workflow" class="figure"></a><hr>
<h3 id="introduction-figure-9">Figure 9</h3>
<p aria-hidden="true">Image 1 of 1: ‘A graphic showing printed spots on a glass slide that are identified by a barcode and that contain primers to capture mRNA from the tissue laid on top of them’</p>
<figure><img src="https://upload.wikimedia.org/wikipedia/commons/1/14/Spatial_transcriptomics_ii.png" alt="A graphic showing printed spots on a glass slide that are identified by a barcode and that contain primers to capture mRNA from the tissue laid on top of them" class="figure mx-auto d-block"><div class="figcaption">A sequencing-based spatial transcriptomics
method using printed spots on a slide.</div>
</figure><hr>
<h3 id="introduction-figure-10">Figure 10</h3>
<p aria-hidden="true">Image 1 of 1: ‘a general schematic showing the Visium technology’</p>
<figure><img src="../fig/visium-technology.png" alt="a general schematic showing the Visium technology" class="figure mx-auto d-block"><div class="figcaption">Overview of Visium technology with fresh-frozen
(FF) or formalin-fixed paraffin embedded (FFPE) tissue. Source: <a href="https://www.10xgenomics.com/platforms/visium" class="external-link">10x Genomics
Visium</a>
</div>
</figure><hr></section><section id="data-and-study-design"><h2 class="section-heading"><a href="data-and-study-design.html">Data and Study Design</a></h2>
<hr class="half-width">
<h3 id="data-and-study-design-figure-1">Figure 1</h3>
<p aria-hidden="true">Image 1 of 1: ‘A graphic showing printed spots on a glass slide that are identified by a barcode and that contain oligonucleotides to capture messenger RNA from the tissue laid on top of them’</p>
<figure><img src="../fig/visium-slide.png" alt="A graphic showing printed spots on a glass slide that are identified by a barcode and that contain oligonucleotides to capture messenger RNA from the tissue laid on top of them" class="figure mx-auto d-block"><div class="figcaption">Visium spatial gene expression slide</div>
</figure><hr>
<h3 id="data-and-study-design-figure-2">Figure 2</h3>
<p aria-hidden="true">Image 1 of 1: ‘A graphic showing printed spots on a glass slide that are identified by a barcode and that contain primers to capture messenger RNA from the tissue laid on top of them’</p>
<figure><img src="../fig/Spatial_transcriptomics_ii_lower_third.png" alt="A graphic showing printed spots on a glass slide that are identified by a barcode and that contain primers to capture messenger RNA from the tissue laid on top of them" class="figure mx-auto d-block"><div class="figcaption">Sequencing data is mapped back to spots on the
slide and compared to an image of the tissue to localize
expression</div>
</figure><hr>
<h3 id="data-and-study-design-figure-3">Figure 3</h3>
<p aria-hidden="true">Image 1 of 1: ‘An example of spatial transcriptomics data showing genes in rows and barcodes (spots) in columns’</p>
<figure><img src="../fig/spatial-data.png" alt="An example of spatial transcriptomics data showing genes in rows and barcodes (spots) in columns" class="figure mx-auto d-block"><div class="figcaption">Spatial transcriptomics data include genes in
rows and barcodes in columns</div>
</figure><hr>
<h3 id="data-and-study-design-figure-4">Figure 4</h3>
<p aria-hidden="true">Image 1 of 1: ‘A Visium spatial transcriptomics workflow with fresh-frozen tissue’</p>
<p><img src="../fig/fresh-frozen-workflow.png" alt="A Visium spatial transcriptomics workflow with fresh-frozen tissue" class="figure">
Graphic from
<a href="https://www.10xgenomics.com/library/6f2b8a" class="external-link"><i>Grant
application resources for Visium products</i></a> at 10X Genomics</p>
<hr>
<h3 id="data-and-study-design-figure-5">Figure 5</h3>
<p aria-hidden="true">Image 1 of 1: ‘A human brain showing a section of dorsolateral prefrontal cortex extracted. A block of tissue containing six cortical layers and an underlying layer of white matter is excised from the section.’</p>
<figure><img src="../fig/tissue-block-with-layers.png" alt="A human brain showing a section of dorsolateral prefrontal cortex extracted. A block of tissue containing six cortical layers and an underlying layer of white matter is excised from the section." class="figure mx-auto d-block"><div class="figcaption">Tissue blocks were excised from human
dorsolateral prefrontal cortex. Tissue blocks include six cortical
layers and underlying white matter (wm).</div>
</figure><hr>
<h3 id="data-and-study-design-figure-6">Figure 6</h3>
<p aria-hidden="true">Image 1 of 1: ‘Three Visium slides showing four spatial capture areas each. Each slide contains directly adjacent serial tissue sections for one subject. The second pair of samples contains tissue sections that are 300 microns posterior to the first pair of samples.’</p>
<figure><img src="../fig/experimental-design.png" alt="Three Visium slides showing four spatial capture areas each. Each slide contains directly adjacent serial tissue sections for one subject. The second pair of samples contains tissue sections that are 300 microns posterior to the first pair of samples." class="figure mx-auto d-block"><div class="figcaption">Slides contain tissue samples from three
neurotypical adult subjects. Each slide contains two pairs of replicates
containing directly adjacent tissue sections 10 microns in size. The
second pair of replicates is located 300 microns posterior to the first
pair. A total of 12 samples were assayed with Visium.</div>
</figure><hr></section><section id="data-preprocessing"><h2 class="section-heading"><a href="data-preprocessing.html">Data Preprocessing</a></h2>
<hr class="half-width">
<h3 id="data-preprocessing-figure-1">Figure 1</h3>
<p aria-hidden="true">Image 1 of 1: ‘H &amp; E slide of sample 151673’</p>
<figure><img src="../fig/tissue_hires_image_151673.png" alt="H &amp; E slide of sample 151673" class="figure mx-auto d-block"><div class="figcaption">H &amp; E slide of sample 151673</div>
</figure><hr>
<h3 id="data-preprocessing-figure-2">Figure 2</h3>
<p aria-hidden="true">Image 1 of 1: ‘Diagram showing tissue and registration spots in corners’</p>
<figure><img src="../fig/tissue_diagram_with_registration_spots.png" alt="Diagram showing tissue and registration spots in corners" class="figure mx-auto d-block"><div class="figcaption">Slide Diagram</div>
</figure><hr>
<h3 id="data-preprocessing-figure-3">Figure 3</h3>
<p aria-hidden="true">Image 1 of 1: ‘Histology slide with tissue and background spots labelled’</p>
<figure style="text-align: center"><img src="../fig/data-preprocessing-rendered-unnamed-chunk-9-1.png" alt="Histology slide with tissue and background spots labelled" class="figure mx-auto d-block"><figcaption>
Spots identified in Tissue and Background
</figcaption></figure><hr>
<h3 id="data-preprocessing-figure-4">Figure 4</h3>
<p aria-hidden="true">Image 1 of 1: ‘Boxplot showing lower trancript counts in background area of slide’</p>
<figure style="text-align: center"><img src="../fig/data-preprocessing-rendered-unnamed-chunk-10-1.png" alt="Boxplot showing lower trancript counts in background area of slide" class="figure mx-auto d-block"><figcaption>
UMI Counts in Tissue and Background
</figcaption></figure><hr>
<h3 id="data-preprocessing-figure-5">Figure 5</h3>
<p aria-hidden="true">Image 1 of 1: ‘Boxplot showing lower numbers of genes in background area of slide’</p>
<figure style="text-align: center"><img src="../fig/data-preprocessing-rendered-unnamed-chunk-11-1.png" alt="Boxplot showing lower numbers of genes in background area of slide" class="figure mx-auto d-block"><figcaption>
Number of Genes in Tissue and Background
</figcaption></figure><hr>
<h3 id="data-preprocessing-figure-6">Figure 6</h3>
<figure><img src="../fig/data-preprocessing-rendered-unnamed-chunk-12-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="data-preprocessing-figure-7">Figure 7</h3>
<p aria-hidden="true">Image 1 of 1: ‘Figure showing UMI counts in each spot with varying intensity across the tissue’</p>
<figure style="text-align: center"><img src="../fig/data-preprocessing-rendered-unnamed-chunk-14-1.png" alt="Figure showing UMI counts in each spot with varying intensity across the tissue" class="figure mx-auto d-block"><figcaption>
UMI Counts in each Spot
</figcaption></figure><hr>
<h3 id="data-preprocessing-figure-8">Figure 8</h3>
<p aria-hidden="true">Image 1 of 1: ‘Figure showing number of genes detected in each spot with varying intensity across the tissue’</p>
<figure style="text-align: center"><img src="../fig/data-preprocessing-rendered-unnamed-chunk-15-1.png" alt="Figure showing number of genes detected in each spot with varying intensity across the tissue" class="figure mx-auto d-block"><figcaption>
Number of Genes in each Spot
</figcaption></figure><hr></section><section id="remove-low-quality-spots"><h2 class="section-heading"><a href="remove-low-quality-spots.html">Remove Low-quality Spots</a></h2>
<hr class="half-width">
<h3 id="remove-low-quality-spots-figure-1">Figure 1</h3>
<p aria-hidden="true">Image 1 of 1: ‘mito’</p>
<figure><img src="../fig/remove-low-quality-spots-rendered-unnamed-chunk-8-1.png" alt="mito" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="remove-low-quality-spots-figure-2">Figure 2</h3>
<p aria-hidden="true">Image 1 of 1: ‘mito’</p>
<figure><img src="../fig/remove-low-quality-spots-rendered-unnamed-chunk-9-1.png" alt="mito" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="remove-low-quality-spots-figure-3">Figure 3</h3>
<p aria-hidden="true">Image 1 of 1: ‘mito’</p>
<figure><img src="../fig/remove-low-quality-spots-rendered-unnamed-chunk-10-1.png" alt="mito" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="remove-low-quality-spots-figure-4">Figure 4</h3>
<figure><img src="../fig/remove-low-quality-spots-rendered-unnamed-chunk-13-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="remove-low-quality-spots-figure-5">Figure 5</h3>
<figure><img src="../fig/remove-low-quality-spots-rendered-unnamed-chunk-14-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr></section><section id="apply-normalization-methods"><h2 class="section-heading"><a href="apply-normalization-methods.html">Normalization in Spatial Transcriptomics</a></h2>
<hr class="half-width">
<h3 id="apply-normalization-methods-figure-1">Figure 1</h3>
<figure><img src="../fig/apply-normalization-methods-rendered-unnamed-chunk-2-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="apply-normalization-methods-figure-2">Figure 2</h3>
<figure><img src="../fig/apply-normalization-methods-rendered-unnamed-chunk-8-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="apply-normalization-methods-figure-3">Figure 3</h3>
<figure><img src="../fig/apply-normalization-methods-rendered-unnamed-chunk-9-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="apply-normalization-methods-figure-4">Figure 4</h3>
<figure><img src="../fig/apply-normalization-methods-rendered-unnamed-chunk-12-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="apply-normalization-methods-figure-5">Figure 5</h3>
<figure><img src="../fig/apply-normalization-methods-rendered-unnamed-chunk-13-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="apply-normalization-methods-figure-6">Figure 6</h3>
<figure><img src="../fig/apply-normalization-methods-rendered-unnamed-chunk-14-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="apply-normalization-methods-figure-7">Figure 7</h3>
<figure><img src="../fig/apply-normalization-methods-rendered-unnamed-chunk-14-2.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr></section><section id="feature-selection-dimensionality-reduction-clustering"><h2 class="section-heading"><a href="feature-selection-dimensionality-reduction-clustering.html">Feature Selection, Dimensionality Reduction, and Spot Clustering</a></h2>
<hr class="half-width">
<h3 id="feature-selection-dimensionality-reduction-clustering-figure-1">Figure 1</h3>
<figure><img src="../fig/feature-selection-dimensionality-reduction-clustering-rendered-unnamed-chunk-3-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="feature-selection-dimensionality-reduction-clustering-figure-2">Figure 2</h3>
<figure><img src="../fig/feature-selection-dimensionality-reduction-clustering-rendered-unnamed-chunk-7-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="feature-selection-dimensionality-reduction-clustering-figure-3">Figure 3</h3>
<figure><img src="../fig/feature-selection-dimensionality-reduction-clustering-rendered-unnamed-chunk-8-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="feature-selection-dimensionality-reduction-clustering-figure-4">Figure 4</h3>
<figure><img src="../fig/feature-selection-dimensionality-reduction-clustering-rendered-unnamed-chunk-9-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="feature-selection-dimensionality-reduction-clustering-figure-5">Figure 5</h3>
<p aria-hidden="true">Image 1 of 1: ‘Tissue Clustering showing the tissue layers at different resolutions and number of PCs’</p>
<figure><img src="../fig/tissue_cluster_resol.png" alt="Tissue Clustering showing the tissue layers at different resolutions and number of PCs" class="figure mx-auto d-block"><div class="figcaption">Tissue Clustering with Different Numbers of PCs
and Clustering Resolutions</div>
</figure><hr>
<h3 id="feature-selection-dimensionality-reduction-clustering-figure-6">Figure 6</h3>
<p aria-hidden="true">Image 1 of 1: ‘UMAP Clustering with Different Numbers of PCs and Clustering Resolutions’</p>
<figure><img src="../fig/umap_cluster_resol.png" alt="UMAP Clustering with Different Numbers of PCs and Clustering Resolutions" class="figure mx-auto d-block"><div class="figcaption">UMAP Clustering with Different Numbers of PCs
and Clustering Resolutions</div>
</figure><hr>
<h3 id="feature-selection-dimensionality-reduction-clustering-figure-7">Figure 7</h3>
<figure><img src="../fig/feature-selection-dimensionality-reduction-clustering-rendered-unnamed-chunk-12-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr></section><section id="deconvolve-cell-types-in-a-spot"><h2 class="section-heading"><a href="deconvolve-cell-types-in-a-spot.html">Deconvolution in Spatial Transcriptomics</a></h2>
<hr class="half-width"></section><section id="differential-expression-testing"><h2 class="section-heading"><a href="differential-expression-testing.html">Differential Expression Testing</a></h2>
<hr class="half-width">
<h3 id="differential-expression-testing-figure-1">Figure 1</h3>
<figure><img src="../fig/differential-expression-testing-rendered-layers-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr></section>
</div>
    </main>
</div>
<!-- END  : inst/pkgdown/templates/content-extra.html -->

      </div>
<!--/div.row-->
      		<footer class="row footer mx-md-3"><hr>
<div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>
        
        <a href="https://github.com/smcclatchy/spatial-transcriptomics/edit/main/README.md" class="external-link">Edit on GitHub</a>
        
	
        | <a href="https://github.com/smcclatchy/spatial-transcriptomics/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/smcclatchy/spatial-transcriptomics/" class="external-link">Source</a></p>
				<p><a href="https://github.com/smcclatchy/spatial-transcriptomics/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:susan.mcclatchy@jax.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">
        
        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>
        
        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/epiverse-trace/sandpaper/tree/patch-renv-github-bug" class="external-link">sandpaper (0.16.4)</a>, <a href="https://github.com/carpentries/pegboard/tree/a32a7836d4455f407c3cafe8ab95edc636e5e919" class="external-link">pegboard (0.7.5)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.2" class="external-link">varnish (1.0.2)</a></p>
			</div>
		</footer>
</div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://smcclatchy.github.io/spatial-transcriptomics/instructor/images.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, The Carpentries, transcriptome, Visium, cancer",
  "name": "All Images",
  "creativeWorkStatus": "active",
  "url": "https://smcclatchy.github.io/spatial-transcriptomics/instructor/images.html",
  "identifier": "https://smcclatchy.github.io/spatial-transcriptomics/instructor/images.html",
  "dateCreated": "2024-03-01",
  "dateModified": "2024-06-07",
  "datePublished": "2024-06-07"
}

  </script><script>
		feather.replace();
	</script><!-- Matomo
    2022-11-07: we have gotten a notification that we have an overage for our
    tracking and I'm pretty sure this has to do with Workbench usage.
    Considering that I am not _currently_ using this tracking because I do not
    yet know how to access the data, I am turning this off for now.
  <script>
    var _paq = window._paq = window._paq || [];
    /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
    _paq.push(["setDocumentTitle", document.domain + "/" + document.title]);
    _paq.push(["setDomains", ["*.preview.carpentries.org","*.datacarpentry.github.io","*.datacarpentry.org","*.librarycarpentry.github.io","*.librarycarpentry.org","*.swcarpentry.github.io", "*.carpentries.github.io"]]);
    _paq.push(["setDoNotTrack", true]);
    _paq.push(["disableCookies"]);
    _paq.push(['trackPageView']);
    _paq.push(['enableLinkTracking']);
    (function() {
          var u="https://carpentries.matomo.cloud/";
          _paq.push(['setTrackerUrl', u+'matomo.php']);
          _paq.push(['setSiteId', '1']);
          var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
          g.async=true; g.src='https://cdn.matomo.cloud/carpentries.matomo.cloud/matomo.js'; s.parentNode.insertBefore(g,s);
        })();
  </script>
  End Matomo Code -->
</body>
</html><!-- END:   inst/pkgdown/templates/layout.html-->

