---
title: 'Deconvolution in Spatial Transcriptomics'
teaching: 60
exercises: 10
---

:::::::::::::::::::::::::::::::::::::: questions 

- How can we bring single-cell resolution to multi-cellular spatial transcriptomics spots?
- What are different algorithmic approaches for doing so?

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: objectives

- Understand deconvolution.
- Perform deconvolution to quantify different cell types in spatial transcriptomics spots using scRNA-seq data.

::::::::::::::::::::::::::::::::::::::::::::::::

```{r echo=FALSE,warning=FALSE,message=FALSE}
suppressPackageStartupMessages(library(R.utils))
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(hdf5r))
suppressPackageStartupMessages(library(spacexr))
suppressPackageStartupMessages(library(Seurat))
suppressPackageStartupMessages(library(data.table))
suppressPackageStartupMessages(library(rcartocolor))

source("https://raw.githubusercontent.com/smcclatchy/spatial-transcriptomics/main/code/spatial_utils.R")

filter_st <- load_seurat_object(file_prefix = 'featselclust')
DefaultAssay(filter_st) <- "SCT"

```

## Deconvolution in Spatial Transcriptomics

Each spatial spot in an ST experiment generally contains multiple cells. For example, 
spots in the Visium assay are 55 microns in diameter, whereas a typical T cell has a diameter of ~10 microns.
As such, the expression read out from the spot mixes together the expression of the individual
cells encompassed by it. Deconvolution is the approach for unmixing this combined
expression signal. Most often, deconvolution methods predict the fraction of each spot's
expression derived from each particular cell type. Supervised methods deconvolve 
spot expression using cell type expression profiles (e.g., from scRNA-seq) or marker genes.
Unsupervised approaches instead infer the expression of the cell types first.

![Deconvolving spatial transcriptomics spots into their constituent cell types.](https://ars.els-cdn.com/content/image/1-s2.0-S200103702200558X-ga1_lrg.jpg){alt='alt text for accessibility purposes'}
<a href="https://www.sciencedirect.com/science/article/pii/S200103702200558X#f0015">Zhang et al,  Comput Struct Biotechnol J 21, 176â€“184 (2023)</a>
<a href="https://creativecommons.org/licenses/by-nc-nd/4.0/" rel="license">CC BY-NC-ND 4.0</a>

## Deconvolution with RCTD (Robust Cell Type Decomposition) 

We will deconvolve spots by applying Robust Cell Type Decomposition (RCTD; 
[Cable, D. M., et al., Nature Biotechnology, 2021](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8606190)),
which is implemented in the [spacexr](https://github.com/dmcable/spacexr) package.
The algorithm uses scRNA-seq data as a reference to deconvolve the 
spatial transcriptomics data, estimating the proportions of different cell types
in each spatial spot. RCTD models the observed gene expression in 
each spatial spot as a mixture of the gene expression profiles of different cell
types, where the estimated proportions are the coefficients of the expression profiles
in a linear model. Its procedure guarantees the <em>estimated</em> proportions are non-negative, but
not that they sum to one. RCTD can operate in two modes: 
in "doublet" mode it fits at most two cell types per spot, in "full" mode it 
fits potentially all cell types in the reference per spot, and in "multi" mode 
it again fits more than two cell types per spot by extending the "doublet" 
approach. Here, we will use "full" mode.

## Loading scRNA-seq reference data

First, we load the scRNA-seq data that will serve as a reference for
deconvolution. Individual cells are assumed to be annotated according to their type.
RCTD will use these cell type-specific expression profiles to deconvolve each spot's
expression.

```{r warning=FALSE,message=FALSE}
# Load scRNA-seq data
sc.counts <- fread("data/scRNA-seq/sc_counts.tsv.gz")
sc.counts <- as.data.frame(sc.counts)
rownames(sc.counts) <- sc.counts[,1]
sc.counts           <- as.matrix(sc.counts[,-1])

# Load cell type annotations
sc.metadata   <- read.delim("data/scRNA-seq/sc_cell_types.tsv")
sc.cell.types <- setNames(factor(sc.metadata$Value), sc.metadata$Name)
shared.cells  <- intersect(colnames(sc.counts), names(sc.cell.types))
sc.cell.types <- sc.cell.types[shared.cells]
sc.counts     <- sc.counts[, shared.cells]
```

Next, we will create the reference object encapsulating this scRNA-seq data.
We will use the [Reference](https://rdrr.io/github/dmcable/RCTD/man/Reference.html), which
will organize and store the scRNA-seq data for the next steps.

```{r warning=FALSE,message=FALSE}
sc_reference <- Reference(sc.counts, sc.cell.types)
```

## Applying RCTD for spot deconvolution

Let's write a wrapper function that performs RCTD deconvolution. This will 
facilitate running RCTD on other samples within this dataset.

```{r}
run.rctd <- function(reference, st.obj) {
  
  # Get raw ST counts
  st.counts <- GetAssayData(st.obj, assay = "Spatial", layer = "counts")
  
  # Get the spot coordinates
  st.coords           <- st.obj[[]][, c("array_col", "array_row")]
  colnames(st.coords) <- c("x","y")
    
  # Create the RCTD 'puck', representing the ST data
  puck   <- SpatialRNA(st.coords, st.counts)

  myRCTD <- create.RCTD(puck, reference, max_cores = 1, keep_reference = TRUE)

  # Run deconvolution -- note that we are using 'full' mode to devolve a spot into 
  # (potentially) all available cell types.
  myRCTD <- suppressWarnings(run.RCTD(myRCTD, doublet_mode = 'full'))
  
  myRCTD
}
```

## Running Deconvolution on Brain Samples

We apply the RCTD wrapper to our spatial transcriptomics data to deconvolve the 
spots and quantify the cell types. This may take ~10 minutes. If you prefer, 
you can load the precomputed results directly.

```{r warning=FALSE,message=FALSE}
# Change this variable to TRUE to load precomputed results, or FALSE to compute
# the results here.
load.precomputed.results <- TRUE

rds.file <- paste0("data/rctd-sample-1.rds")

if(!load.precomputed.results || !file.exists(rds.file)) {

  result_1 <- run.rctd(sc_reference, filter_st)
  # The RCTD file is large. To save space, we will remove the reference counts.
  result_1 <- remove.RCTD.reference.counts(result_1)
  saveRDS(result_1, rds.file)

} else {

  result_1 <- readRDS(rds.file)

}
```

## Interpreting Deconvolution Results

RCTD outputs the proportion of different cell types in each
spatial spot. These are held in the spatialRNA@counts slot. 
Let's see the proportions it predicts for this sample:

```{r}
props <- as.data.frame(result_1@results$weights)
head(props)
```

Notice that the proportions don't sum exactly to one:

```{r}
head(rowSums(props))
```

Let's classify the spot according to the layer type with highest proportion:

```{r}
props$classification <- apply(props, 1, function(row) names(row)[which.max(row)])
```

Let's add the deconvolution results to our Seurat object.

```{r}
filter_st <- AddMetaData(object = filter_st, metadata =  props)
```

We can now visualize the predicted layer classifications and compare them alongside
the authors' annotations that we saw previously.

```{r warning=FALSE,message=FALSE,fig.width=8}
SpatialDimPlotColorSafe(filter_st[, !is.na(filter_st[[]]$classification)], "classification")
```

```{r warning=FALSE,message=FALSE,fig.width=8}
SpatialDimPlotColorSafe(filter_st[, !is.na(filter_st[[]]$layer_guess)], "layer_guess")
```


To be more quantitative, we can compute a confusion matrix comparing the layers predicted by RCTD with
those annotated by the authors. 

```{r fig.width=6}
df            <- as.data.frame(table(filter_st[[]]$layer_guess, filter_st[[]]$classification))
colnames(df)  <- c("Annotation", "Prediction", "Freq")
df$Annotation <- factor(df$Annotation)
df$Prediction <- factor(df$Prediction)

ggplot(data = df, aes(x = Annotation, y = Prediction, fill = Freq)) + 
  geom_tile() +
  theme(text = element_text(size = 20), axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
```

Note that there is a fairly strong correlation between the predicted and observed layers,
particularly for the pairs Oligodendrocytes and WM (White Matter), L4 and Layer 4, and L2-3 and Layer 3.

## Summary

Incorporating deconvolution into the spatial transcriptomics workflow enhances the ability to quantify the different cell
types in each spatial spot, providing a richer and more detailed understanding of the tissue architecture. By following
these steps, researchers can ensure that their spatial transcriptomics data is accurately deconvolved, leading to more
robust and insightful analyses.

:::::::::::::::::::::::::::::::::: keypoints

- Deconvolution enhances spatial transcriptomics by quantifying the different cell types within spatial spots.
- Integrating scRNA-seq data with spatial transcriptomics data is essential for accurate deconvolution.
- The RCTD method is effective for quantifying the proportion of different cell types in spatial transcriptomics data.

:::::::::::::::::::::::::::::::::::::::::::::


```{r echo=FALSE,include=FALSE}
save_seurat_object(obj = filter_st, file_prefix = 'deconcelltypes')
```
