---
title: 'Differential Expression Testing'
teaching: 60
exercises: 20
---

:::::::::::::::::::::::::::::::::::::: questions 

- How can we access region-specific gene expression using differential expression?
- How can we access spatially varying gene expression using spatial statistics?
::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: objectives

- Identify differentially expressed genes across different layers defined by expert 
annotations.
- Utilize the Moran's I statistic to find spatially variable genes.
- Explore the relationship between layer-specific genes identified by differential expression and spatially varying genes identified by Moran's I.

::::::::::::::::::::::::::::::::::::::::::::::::

```{r setup, include=FALSE}
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(presto))
suppressPackageStartupMessages(library(Seurat))
suppressPackageStartupMessages(library(ComplexHeatmap))
suppressPackageStartupMessages(library(rcartocolor))
suppressPackageStartupMessages(library(data.table))
#suppressPackageStartupMessages(library(ggalluvial))

source("https://raw.githubusercontent.com/smcclatchy/spatial-transcriptomics/main/code/spatial_utils.R")

filter_st <- load_seurat_object(file_prefix = 'deconcelltypes')
DefaultAssay(filter_st) <- "SCT"
```

## Spot-level Differential Expression Across Annotated Regions

We will begin by performing differential expression across the annotated layers. 
We will use the source publication's spot annotation, which is stored in the
"layer_guess" column of the metadata. As a reminder, those look like:

```{r layers,warning=FALSE,message=FALSE}
SpatialDimPlotColorSafe(filter_st[, !is.na(filter_st[[]]$layer_guess)], "layer_guess") + 
  labs(fill = "Layer") 
```

We identify genes that are upregulated in each annotated brain region using the 
[FindAllMarkers](https://satijalab.org/seurat/reference/findallmarkers) function 
in Seurat. This performs a "one versus rest" comparison of a gene's expression 
in one region relative to that gene's expression in all other regions. The 
default test used here, the Wilcoxon Rank Sum test, 
will use an efficient implementation within the 
[presto](https://github.com/immunogenomics/presto) library, if installed. The 
speedup over the default implementation is substantial, and we highly recommend 
installing presto and using Seurat v5, which leverages it.

```{r layer-de}
Idents(filter_st)  <- "layer_guess"
de_genes           <- FindAllMarkers(filter_st, 
                                     assay    = "SCT",
                                     verbose  = FALSE,
                                     only.pos = TRUE, 
                                     min.pct  = 0.25, 
                                     logfc.threshold = 0.25)
```

The resulting table indicates DE p-values and adjusted p-values for each gene,
along with the percentage of spots in which the gene was detected (pct.1) 
in the corresponding cluster and the percentage of spots in which it was
detected in all <em>other</em> clusters (pct.2):

```{r layer-degs}
head(de_genes)
```

It is not uncommon to have pathologist annotations of regions. The Visium assay is performed
on a tissue that is also stained for hematoxylin and eosin (H&E) -- a routine practice in
pathology for diagnosing cancer, for example. A pathologist could manually annotate this H&E
image using a histology viewer, such as QuPath.

## Spot-level Differential Expression Across Clusters

In cases where we do not have expert annotations, we could perform the analysis above across
clusters. Indeed, this is often the first step to interpreting a cluster -- comparing its marker
genes to those of regions expected within the tissue. It is important to remember, however,
that these markers are for clusters of spots -- aggregations of cells -- not individual cells, as
would be the case in scRNA-seq. Those aggregations may be of cells with similar types, in which
case analysis may be similar to that of scRNA-seq, or of different cell types, in which case the
interpretation would be quite different than with scRNA-seq. Let's try this.

First, let's remind ourselves what the clusters look like:

```{r clusters,warning=FALSE,message=FALSE}
SpatialDimPlotColorSafe(filter_st, "seurat_clusters") + labs(fill = "Cluster") 
```

Now, we will use nearly identical code as above to perform the differential expression
analysis. We only change each cell's identity class with the 
[Idents](https://satijalab.github.io/seurat-object/reference/Idents.html) function,
to associate each cell with its cluster.

```{r layer-de-cluster}
Idents(filter_st)  <- "seurat_clusters"
de_genes_cluster   <- FindAllMarkers(filter_st, 
                                     assay    = "SCT",
                                     verbose  = FALSE,
                                     only.pos = TRUE, 
                                     min.pct  = 0.25, 
                                     logfc.threshold = 0.25)
```

Let's visualize the overlap between genes differentially expressed across annotated layers
and those differentially expressed across clusters. We will do so using a confusion matrix.

```{r de-comparison-confusion}
de_genes_all <- merge(subset(de_genes, p_val_adj < 0.05),
                      subset(de_genes_cluster, p_val_adj < 0.05),
                      by=c("gene"), all=TRUE, suffixes = c(".anno", ".cluster"))
layer.order <- c("WM", "Layer1", "Layer2", "Layer3", "Layer4", "Layer5", "Layer6")
df            <- as.data.frame(table(de_genes_all$cluster.anno, de_genes_all$cluster.cluster))
colnames(df)  <- c("de.anno", "de.cluster", "Freq")
df$de.anno <- factor(df$de.anno, levels = c("NA", layer.order))
df$de.cluster <- factor(df$de.cluster)

ggplot(data = df, aes(x = de.anno, y = de.cluster, fill = Freq)) +
  geom_tile() +
  theme(text = element_text(size = 20), axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  xlab("Annotation-based DE Genes") + ylab("Cluster-based DE Genes")
```

Let's visualize the overlap between genes differentially expressed across annotated layers
and those differentially expressed across clusters. We will do so using an alluvial diagram.

```{r de-comparison-alluvial}
if(FALSE) {
de_genes_all <- merge(subset(de_genes, p_val_adj < 0.05),
                      subset(de_genes_cluster, p_val_adj < 0.05),
                      by=c("gene"), all=TRUE, suffixes = c(".anno", ".cluster"))
layer.order <- c("WM", "Layer1", "Layer2", "Layer3", "Layer4", "Layer5", "Layer6")

de_genes_all %>% 
  mutate(cluster.anno = factor(cluster.anno, levels = c("NA", layer.order))) %>% 
  mutate(cluster.cluster = as.factor(cluster.cluster)) %>% 
  ggplot(
    aes(
      axis1 = cluster.anno, 
      axis2 = cluster.cluster,
      fill = cluster.anno
    )
  ) +
  geom_alluvium() +
  geom_stratum() +
  geom_text(stat = "stratum", aes(label = after_stat(stratum))) +
  scale_x_discrete(limits = c("Annotation DE Genes", "Cluster DE Genes"), 
                   expand = c(0.15, 0.05)) +
  labs(fill = "Layer")
}
```

The above plot shows only marginal overlap between the annotation-derived DE genes
on the x axis and the cluster-derived DE genes on the y axis. This might indicate that
the two approaches are complementary. In fact, they could be used in conjunction.
We could have performed clustering <em>within<\em> the annotated regions to assess
intra-layer heterogeneity.

## Moran's I Statistic

We next consider an alternative to the above approaches that both rely on defined 
regions -- be they defined from expert annotation or via clustering. 
Instead, we will apply the Moran's I statistic.
Moran's I is a measure used to assess spatial autocorrelation in data, 
indicating whether similar values of a feature (e.g., expression levels of
a gene) are clustered, dispersed, or random 
([Jackson, et al. 2010](https://ij-healthgeographics.biomedcentral.com/articles/10.1186/1476-072X-9-33)). 
These correspond to Moran's I values that are positive, negative, or near zero, 
respectively.

Here, we can apply it to detect genes whose expression patterns 
exhibit spatial structure, which may reflect region-specific, biological function.
That is, we anticipate that spatially variable genes will exhibit region-specific
expression. Let's check that hypothesis by first computing spatially variable genes
and then assessing whether they are differentially expressed across regions.

![Moran's I statistic quantifies spatial correlation. **Top Left:** Checkerboard pattern results in negative Moran's I, indicating anti-correlation. **Top Right:** Linear gradient shows a high positive Moran's I, reflecting a strong spatial gradient. **Bottom Left:** Random pattern leads to a Moran's I near zero, suggesting no significant spatial autocorrelation. **Bottom Right:** 'Ink blot' pattern demonstrates positive autocorrelation, indicative of a clustered or spreading pattern. Relationships are calculated using direct, equally weighted neighbors, normalized for each cell. ](https://upload.wikimedia.org/wikipedia/commons/f/f0/Moran%27s_I_example.png){alt="Moran's I statistic quantifies spatial correlation."}

Image by <a href="https://commons.wikimedia.org/wiki/File:Moran%27s_I_example.png">WikiNukalito</a>, <a href="https://creativecommons.org/licenses/by-sa/4.0">CC BY-SA 4.0</a>, via Wikimedia Commons.

### Spatial Differential Expression Using Moran's I

We identify the genes whose expression patterns exhibit clear spatial structure 
using Moran's I algorithm, as implemented in [FindSpatiallyVariableFeatures](https://satijalab.org/seurat/reference/findspatiallyvariablefeatures). 
We have selected the top 1,000 genes, which should
be sufficient to identify brain regions. The following will take several minutes to run.

```{r moran-i,warning=FALSE,message=FALSE}
svg <- 
  FindSpatiallyVariableFeatures(filter_st, 
                                assay            = "SCT", 
                                features         = VariableFeatures(filter_st)[1:1000], 
                                selection.method = "moransi")
```

FindSpatiallyVariableFeatures returns a Seurat object, populated with Moran's I-derived
results. Normally, we would use the 
[SpatiallyVariableFeatures](https://satijalab.github.io/seurat-object/reference/VariableFeatures.html) function to query those results. But, there is a bug in that
function as described [here](https://github.com/satijalab/seurat/issues/7422). So, instead
we will manually extract the top 100 ranked spatially variable genes, along with their 
Moran's I values and associated p-values:

```{r moran-genes}
# Get and sort 'MoransI_observed' values
morans_i_genes <- svg@assays[["SCT"]]@meta.features %>%
                    rownames_to_column("gene") %>%
                    arrange(desc(MoransI_observed)) %>%
                    slice_head(n = 100)
head(morans_i_genes)
```

## Correlation of Region-specific Differentially Expressed Genes and Spatially Variable Genes

### Heatmap of Differential Expression

As a sanity check of both the region-specific differential expression and the
Moran's I approaches, let's check out hypothesis that spatially variable genes
are likely to show regional differential expression. To do
this, we will plot a heatmap of the <em>differential expression</em> p-values 
for the top 100 spatially variable genes, organized by brain region.

```{r heatmap-de,fig.width=6}
# Merge the Moran's I values with the DE genes
df <- merge(morans_i_genes, de_genes, all.x = TRUE, by = "gene")
df <- subset(df, !is.na(cluster))

# We will plot the -log2 pvalues. Compute this and adjust for taking log of 0.
df$log_p_val_adj <- -log2(df$p_val_adj)
df$log_p_val_adj[is.infinite(df$log_p_val_adj)] <- 
  max(df$log_p_val_adj[!is.infinite(df$log_p_val_adj)])

# Create a matrix whose rows are the spatially variable genes (indicated by Moran's I),
# whose columns are the clusters, and whose entries are the adjusted DE pvalue for the
# corresponding gene and cluster.
p_val_adj_matrix <- df %>%
                       select(gene, cluster,log_p_val_adj) %>%
                       pivot_wider(names_from = cluster, values_from = log_p_val_adj, values_fill = 0) %>%
                       column_to_rownames("gene") %>%
                       as.matrix()

# Order the regions according to their spatial organization from inner to outer layers
p_val_adj_matrix <- p_val_adj_matrix[, layer.order]

# Create a heatmap of the DE p-values of spatially variable genes
Heatmap(p_val_adj_matrix,
        column_title      = "Heatmap of DE p-values of spatially variable genes",
        name              = "DE -log2(p-values)", # Title for the heatmap legend
        row_title         = "Spatially variable genes",
        cluster_rows      = TRUE, 
        cluster_columns   = FALSE,
        show_row_names    = FALSE, 
        show_column_names = TRUE,
        show_row_dend     = FALSE, 
        show_column_dend  = TRUE)

```

The heatmap visualization reveals a key finding of our analysis: genes displaying 
the highest Moran's I values show distinct expression patterns that align with 
specific brain regions identified through expert annotations. This observation 
underscores the spatial correlation of gene expression, highlighting its 
potential relevance in understanding regional brain functions and pathologies.

## Other Considerations

We here explored the very basics of differential expression analysis. We note here
just a few additional complications that may arise in your own analyses.
- Here we explored spot-level analysis within one sample. However, most studies
(including the published one that furnished the data we used here) include multiple
samples. This has some similarities to single cell-level DE analysis in scRNA-seq.
In the scRNA-seq setting, several studies, including one by [Squair, et al.](https://www.nature.com/articles/s41467-021-25960-2), have emphasized drawbacks of performing
DE on individual cells. Single-cell DE may be caused by technical variability across
single cells, biological variability that may not be of interest across conditions
studied (e.g., transcriptional bursting), or simply by the large number of individual
cells that inflate false positives. One solution is to perform "pseudobulking" of
the scRNA-seq and then to use conventional bulk RNA-seq DE tools (e.g., DESeq2) to
compare the pseudobulks. A Seurat 
[vignette](https://satijalab.org/seurat/articles/de_vignette) describing the 
approach for scRNA-seq should likewise be applicable for spatial transcriptomics.
However, we note the caveat that spots are not cells -- the former may have
considerably more intra- and inter-spot biological heterogeneity. Hence, pseudobulking
should be done with caution.
- Any comparison across samples will necessitate accounting for batch effects. In
bulk or single-cell RNA-seq studies, batch effects would typically be mitigated by
applying regression-based tools such as 
[Harmony](https://portals.broadinstitute.org/harmony/articles/quickstart.html) 
or by incorporating a batch factor in a linear modeling framework (e.g., of DESeq2).
Those approaches could be applied to spatial transcriptomics data, as well.
- Batch correction procedures, such as those just mentioned, may remove biological
variability along with technical variability. An alternative approach is to perform
analyses <em>within<\em> samples and then to perform meta-analysis across them.
One intriguing possibility is to compare rank-based statistics across samples, which
should be less susceptible to batch effects. Another is to cluster across samples
based on overlapping sets of genes within intra-cluster samples. Such an approach
has been applied in the "meta-programs" defined in cancer by [Gavish and
colleagues](https://pubmed.ncbi.nlm.nih.gov/37258682/).

There is unlikely to be one turn-key approach to addressing the above issues.
You will likely need to explore the data -- and that exploration should be
guided by some biological expectations that anchor your choices.

:::::::::::::::::::::::::::::::::: keypoints

- Differential expression testing pinpoints genes with significant expression 
variations across regions or clusters. 
- Moran's I statistic reveals spatial autocorrelation in gene 
expression, critical for examining spatially dependent biological activities.
- Moran's I algorithm effectively identifies genes expressed in anatomically 
distinct regions, as validated from the correlation analysis with the DE genes 
from the annotated regions.

:::::::::::::::::::::::::::::::::::::::::::::

```{r echo=FALSE,include=FALSE}
save_seurat_object(obj = filter_st, file_prefix = 'detesting')
```

