---
title: 'Remove Low-quality Spots'
teaching: 10
exercises: 2
---

:::::::::::::::::::::::::::::::::::::: questions 

- How do I remove low-quality spots?
- What kinds of problems produce low-quality spots?
- What happens if I skip quality control and proceed with analysis?

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: objectives

- Explain how to use markdown with the new lesson template
- Demonstrate how to include pieces of code, figures, and nested challenge blocks

::::::::::::::::::::::::::::::::::::::::::::::::

## Introduction

Spatial transcriptomics involves a complex process that may involves some
technical failures. If the processing of the entire slide fails, it should be
obvious due to a large number of transcripts appearing in spots outside of the
tissue or low transcript counts across the whole tissue.

However, there may also be variations in spot quality in a slide that has
largely high-quality spots. We will use two metrics to filter spots:

* Mitochondrial gene content, and
* Ribosomal gene content.


> DMG: Ask Brian for his thoughts too.

Maybe similar to the rationale in scRNAseq???

## Filter by Transcript and Gene Count

In the previous lesson, we looked a the number of transcripts expressed and
the number of genes detected in each spot. We plotted these values.



```{r}
hist(FetchData(filter_st, "nCount_Spatial")[,1],   main = 'Counts per Spot')
hist(FetchData(filter_st, "nFeature_Spatial")[,1], main = 'Genes per Spot')
```


## Identify Mitochondrial and Ribosomal Genes

We will search the gene symbols in the feature metadata to identify 
mitochondrial and ribosomal genes. We do not need to find all genes in these
categories, so we will search for genes with symbols that start with "MT" or
"RP".

The Seurat object is designed to be flexible and may contains several data types.
For example, it may contain both transcript and open chromatin peaks. In this 
analysis, the Seurat object only contains transcipt counts. The different types
of data are called "Layers" in Seurat and may be accessed using the 
[Layers](https://satijalab.github.io/seurat-object/reference/Layers.html) 
function.

```{r}
Layers(filter_st)
```

This tells us that the "filter_st" object only contains one data Layer called
"counts". We can access this using the 
[LayerData](https://satijalab.github.io/seurat-object/reference/Layers.html)
function using "counts" as an argument.

```{r}
counts <- LayerData(filter_st, 'counts')
head(counts)
```

> DMG: Add some explanation of what we're seeing, sparse matrix, etc.

The gene symbols are in the rownames of "counts". We will search for gene
symbols which start with "MT" to find mitochondrial genes.

```{r}
mito_pattern <- '^[Mm][Tt]-'
mito_genes   <- rownames(counts)[grep(mito_pattern, rownames(counts))]
mito_genes
```

We now have a set of mitochondrial genes. We will use these genes to estimate the 
percentage of transcript counts expressed by mitochondrial genes and add this 
to the Seurat object.

```{r}
filter_st[["percent.mt"]] = PercentageFeatureSet(filter_st, pattern = mito_pattern)
```

This syntax adds a new column to the spot metadata.

```{r}
head(filter_st@meta.data)
```

Next, we will get the ribosomal genes and add the percentage of ribosomal 
transcripts expressed to the spot metadata.

```{r}
ribo_pattern = '^RP[SL]'
ribo_genes   <- rownames(counts)[grep(ribo_pattern, rownames(counts))]
ribo_genes
```

```{r}
filter_st[["percent.ribo"]] = PercentageFeatureSet(filter_st, pattern = ribo_pattern)
```

:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::: instructor

There is no need to have students type out the figure titles and axis labels.

::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

```{r fig.height=6,fig.width=4}
layout(matrix(1:2, ncol = 1))
hist(FetchData(filter_st, "percent.ribo")[,1], main = '% Ribosomal Genes')
hist(FetchData(filter_st, "percent.mt")[,1],   main = "% Mitochondrial Genes")
```





:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::: instructor

Inline instructor notes can help inform instructors of timing challenges
associated with the lessons. They appear in the "Instructor View"

::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: challenge 

## Challenge 1: Can you do it?

What is the output of this command?

```r
paste("This", "new", "lesson", "looks", "good")
```

:::::::::::::::::::::::: solution 

## Output
 
```output
[1] "This new lesson looks good"
```

:::::::::::::::::::::::::::::::::


## Challenge 2: how do you nest solutions within challenge blocks?

:::::::::::::::::::::::: solution 

You can add a line with at least three colons and a `solution` tag.

:::::::::::::::::::::::::::::::::
::::::::::::::::::::::::::::::::::::::::::::::::

## Figures

You can include figures generated from R Markdown:

```{r pyramid, fig.alt = "pie chart illusion of a pyramid", fig.cap = "Sun arise each and every morning"}
pie(
  c(Sky = 78, "Sunny side of pyramid" = 17, "Shady side of pyramid" = 5), 
  init.angle = 315, 
  col = c("deepskyblue", "yellow", "yellow3"), 
  border = FALSE
)
```
Or you can use pandoc markdown for static figures with the following syntax:

`![optional caption that appears below the figure](figure url){alt='alt text for
accessibility purposes'}`

![You belong in The Carpentries!](https://raw.githubusercontent.com/carpentries/logo/master/Badge_Carpentries.svg){alt='Blue Carpentries hex person logo with no text.'}

## Math

One of our episodes contains $\LaTeX$ equations when describing how to create
dynamic reports with {knitr}, so we now use mathjax to describe this:

`$\alpha = \dfrac{1}{(1 - \beta)^2}$` becomes: $\alpha = \dfrac{1}{(1 - \beta)^2}$

Cool, right?

::::::::::::::::::::::::::::::::::::: keypoints 

- Use `.md` files for episodes when you want static content
- Use `.Rmd` files for episodes when you need to generate output
- Run `sandpaper::check_lesson()` to identify any issues with your lesson
- Run `sandpaper::build_lesson()` to preview your lesson locally

::::::::::::::::::::::::::::::::::::::::::::::::

